Combining F/RPKM values from public data sets
=============================================

We will join the data sets on ENSEMBL ID:s, losing a lot of data in the process; but joining on gene symbols or something else would lead to an even worse loss. 

```{r LoadPackagesAndData}
library(org.Hs.eg.db)
library(data.table)

hpa.fpkms <- read.delim("hpa_fpkms.txt")
altiso.fpkms <- read.delim("altiso_fpkms.txt")
gtex.fpkms <- read.delim("gtex_fpkms.txt")
atlas.fpkms <- read.delim("atlas_fpkms.txt")
```

The RNA-seq Atlas data set uses many different identifiers. Check how many unique values each identifier has.
 
```{r}
length(unique(atlas$entrez_gene_id)) 
length(unique(atlas$ensembl_gene_id))
length(unique(atlas$hgnc_symbol)) 
length(unique(atlas$transcripts)) 
```

Approach 1: Merge on ENSEMBL genes as given in RNA-seq Atlas. Note that there are repeated ENSG ID:s in RNA-seq Atlas, as opposed to the other data sets, so we need to do something about that. In this case, we just 

data.dt <- data.table(atlas.fpkms)
setkey(data.dt, ENSG_ID)
temp <- data.dt[, lapply(.SD, sum), by=ENSG_ID]
collapsed <- as.data.frame(temp)
atlas.fpkms.summed <- collapsed[,2:ncol(collapsed)] 
rownames(atlas.fpkms.summed) <- collapsed[,1]

atlas.fpkms.summed <- atlas.fpkms.summed[2:nrow(atlas.fpkms.summed),]

fpkms <- merge(fpkms, atlas.fpkms.summed, by.x="ENSG_ID", by.y=0)
gene_id <- fpkms[,1]
f <- fpkms[,2:ncol(fpkms)]
rownames(f) <- gene_id

# Try to map Entrez symbols to ENSEMBL to recover more ENSG IDs than already present in the table. Does it matter which direction?
# Take the names from "fpkms" and convert to Entrez, or vice versa?
m <- org.Hs.egENSEMBL
mapped_genes <- mappedkeys(m)
ensg.for.entrez <- as.list(m[mapped_genes])
remapped.ensg <- ensg.for.entrez[as.character(atlas$entrez_gene_id)]

atlas.fpkms$remapped_ensg <- as.character(remapped.ensg)

# And add expression values
data.dt <- data.table(atlas.fpkms[,2:ncol(atlas.fpkms)])
setkey(data.dt, remapped_ensg)
temp <- data.dt[, lapply(.SD, sum), by=remapped_ensg]
collapsed <- as.data.frame(temp)
atlas.fpkms.summed <- collapsed[,2:ncol(collapsed)] 
rownames(atlas.fpkms.summed) <- collapsed[,1]

# Many more recovered!!
fpkms <- merge(fpkms, atlas.fpkms.summed, by.x="ENSG_ID", by.y=0)
gene_id <- fpkms[,1]
f <- fpkms[,2:ncol(fpkms)]
rownames(f) <- gene_id

# 13537 common IDs, cool.

# Heatmap from Pearson correlation
pheatmap(cor(f))
# Heatmap from Spearman correlation
pheatmap(cor(f, method="spearman")) # Not so bad actually
# PCA 
p <- prcomp(t(f))
plot(p$x[,1],p$x[,2],pch=".")
text(p$x[,1],p$x[,2],labels=rownames(p$x),cex=0.75)

# log scale
library(limma)
log.fpkm <- voom(f)
pheatmap(cor(log.fpkm$E))
pheatmap(cor(log.fpkm$E, method="spearman"))

colors <- c(1,2,3,1,2,1,2,3,1,2,3)

p <- prcomp(t(log.fpkm$E)) # PC1 vs PC5 pretty good here!
p <- prcomp(t(f)) # No outstanding combo

par(mfrow=c(4,4))
for (i in 1:6){
	for(j in 1:6){
		if (i<j){ 
		plot(p$x[,i],p$x[,j],pch=20,col=colors,xlab=paste("PC", i),ylab=paste("PC", j))
		}
	}
}


