Comparing FPKMs for FASTQ files reprocessed with TopHat and Cufflinks 
========================================================

Preliminaries (libraries etc.)

```{r}
library(pheatmap)
library(calibrate)

f <- read.delim("fpkm_table_tophat.txt")
sampleinfo <- read.delim("sample_info_reprocessed.txt")
  
do.SVD = function(m, comp.1=1, comp.2=2){ # returns eig.cell
  s <- svd(m)
  ev <- s$d^2 / sum(s$d^2)
	return(s$u[,c(comp.1, comp.2)])
}

project.SVD <- function(m, eig.cell){
	return(t(m) %*% eig.cell)
}

plot.SVD <- function(m, comp.1=1, comp.2=2, groups=rep("blue", ncol(m)), title=""){
	eig <- do.SVD(m, comp.1, comp.2)
	proj <- project.SVD(m, eig)
	xminv <- min(proj[,1]) # - .2 * abs(min(proj[,1]))
	xmaxv <- max(proj[,1]) # + .2 * abs(max(proj[,1]))
	yminv <- min(proj[,2]) # - .2 * abs(min(proj[,2]))
	ymaxv <- max(proj[,2]) # + .2 * abs(max(proj[,2]))
	plot(proj,pch=20,col="white",xlim=c(xminv,xmaxv),ylim=c(yminv,ymaxv),xaxt='n',yaxt='n',xlab="PC1",ylab="PC2",main=title)
	
	points(proj, col=as.character(groups),pch=20) # , #pch=c(rep(15,3),rep(17,3),rep(19,3),rep(18,3),rep(20,2)), cex=2)
	textxy(proj[,1],proj[,2],labs=colnames(m))
}

loadings.SVD <- function(m, comp=1, gene.ids = rownames(m)){
	s <- svd(m)
	l <- s$u[,comp]
	names(l) <- gene.ids
	l.s <- l[order(l)]
	return(l.s)
}

plot.loadings.SVD <- function(m, comp=1, cutoff=0.1, gene.ids = rownames(m)){
	l <- loadings.SVD(m, comp, gene.ids)
	barplot(l[abs(l)>cutoff],las=2,main=paste("PC", comp, "cutoff", cutoff),cex.names=0.6)
}

plotPC <- function(matrix,a,b,desc,colors){
eig <- do.SVD(matrix, a, b)
proj <- project.SVD(matrix, eig)
xminv <- min(proj[,1]) - .2 * abs(min(proj[,1]))
xmaxv <- max(proj[,1]) + .2 * abs(max(proj[,1]))
yminv <- min(proj[,2]) - .2 * abs(min(proj[,2]))
ymaxv <- max(proj[,2]) + .2 * abs(max(proj[,2]))
plot(proj,pch=20,xlim=c(xminv,xmaxv),ylim=c(yminv,ymaxv),xaxt='n',yaxt='n',xlab=paste0("PC",a),ylab=paste("PC",b),col=colors,main=desc)
textxy(proj[,1],proj[,2],labs=rownames(proj))
}

```

First, we will restrict the data set to only include protein coding genes:

```{r}

library(biomaRt)

f_ids <- as.vector(f[,1])

#ensembl = useMart("ensembl", dataset = "hsapiens_gene_ensembl") #select the ensembl database

#gene_type <- getBM(attributes=c("ensembl_gene_id", "gene_biotype"), 
                  # filters = "ensembl_gene_id",
                  # values=f_ids,
                  # mart=ensembl)

#pc <- subset(gene_type[,1],gene_type[,2]=="protein_coding")

#f_pc <- f[match(pc,f[,1]),]

```

And let's remove all lines where FPKM is close to zero in all samples before we proceed with this version of the data set:

```{r}

#f_pc_nozero <- f_pc[-which(rowSums(f_pc[,3:16])<=0.01),]
#write.table(f_pc_nozero, file="cufflinks_fpkm_proteincoding_nozero.txt", quote=F)

f_pc_nozero <- read.table(file="cufflinks_fpkm_proteincoding_nozero.txt")

```

**Figure 4A**

Heatmap of Spearman correlations between reprocessed expression profiles (# genes = 22,360)

```{r}
pheatmap(cor(f_pc_nozero[,3:16], method="spearman"))
```

Alternatively, one could use Pearson correlation (not shown in paper):

```{r}
pheatmap(cor(f_pc_nozero[,3:16]))
```

Sometimes the linear (Pearson) correlation works better on log values.  (not shown in paper):

```{r}
pseudo <- 1
logs <- log2(f_pc_nozero[,3:16] + pseudo)
fpkms.log <- cbind(f_pc_nozero[,1:2],logs) 

pheatmap(cor(fpkms.log[,3:16]))
```

What if we drop the genes that have less than FPKM 1 on average? (not shown in paper):

```{r}
f.nolow <- f_pc_nozero[-which(rowMeans(f_pc_nozero[,3:16])<1),]
pheatmap(cor(log2(f.nolow[,3:16] + pseudo)))
```

Let's look at a few SVD plots. 

First we take a look at the "raw" FPKM values for PC 1&2:

**Figure 4B**
```{r}

colors <- c(2,1,3,2,1,3,2,1,3,2,1,3,2,1)
plotPC(f_pc_nozero[,3:16], 1, 2, "Reprocessed FPKM values \n SVD \n n=19524", colors=colors)

```

and PC 2&3:

**Figure 1D** (not included in the current manuscript version)

```{r}

plotPC(f_pc_nozero[,3:16], 2, 3, "Reprocessed FPKM values \n SVD \n n=19524", colors=colors)

```

We can plot all pairwise combinations of principal components 1 to 5. (not shown in paper)
Start with SVD on the "raw" F/RPKMs.

```{r}
colors <- c(2,1,3,2,1,3,2,1,3,2,1,3,2,1)

par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
  	if (i<j){ 
		plotPC(f_pc_nozero[,3:16],i,j,desc="",colors=colors)
		}
	}
}
```

Let's see how the PCA plots look for log2-FPKM values:

```{r}

plotPC(fpkms.log[,3:16], 1, 2, desc="Reprocessed F/RPKM values, log2 \n SVD \n n=19475", colors=colors)

plotPC(fpkms.log[,3:16], 2, 3, desc="Reprocessed FPKM values, log2 \n SVD \n n=19475", colors=colors)

```

Or all combinations for log2-FPKM and regular PCA (prcomp())
```{r}
colors <- c(2,1,3,2,1,3,2,1,3,2,1,3,2,1)

p <- prcomp(t(fpkms.log[,3:16]),scale.=T)

par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
    if (i<j){ 
    	plot(p$x[,i],p$x[,j],pch=20,col=colors,xlab=paste("PC", i),ylab=paste("PC", j))
		}
	}
}
```


Combat analysis for removal of batch effects (n=19475):

```{r:combat}
library(sva)

meta <- data.frame(study=c(rep("EoGE",3),rep("Atlas",3),rep("BodyMap",3),rep("HPA",3),rep("AltIso",2)),tissue=c("Brain","Heart","Kidney","Brain","Heart","Kidney","Brain","Heart","Kidney","Brain","Heart","Kidney","Brain","Heart"),prep=c(rep("poly-A",3),rep("rRNA-depl",3),rep("poly-A",8)),layout=c(rep("PE",3),rep("SE",3),rep("PE",6),rep("SE",2)))

batch <- meta$study
design <- model.matrix(~as.factor(tissue),data=meta)

combat <- ComBat(dat=fpkms.log[,3:16],batch=batch,mod=design,numCovs=NULL,par.prior=TRUE)

write.table(combat, file="reprocessed_rpkms_combat_log2.txt", quote=F)

```

Let's see how the correlation heatmap and PCA plots look after correction for batch effects with combat:

```{r}

pheatmap(cor(combat))
plotPC(combat,1,2,colors=colors,desc="Reprocessed F/RPKM values, ComBat on log2 values \n SVD \n n=19475")
plotPC(combat,2,3,colors=colors,desc="Reprocessed F/RPKM values, ComBat on log2 values \n SVD \n n=19475")

```

Anova analysis of different batch factors:

```{r:anova}
library(reshape)
m <- melt(f_pc_nozero)
colnames(m) <- c("ENSG","Gene","sample_ID","Cuff_FPKM")
meta <- data.frame(tissue=c("brain","heart","kidney","brain","heart","kidney","brain","heart","kidney","brain","heart","kidney","brain","heart"),study=c("EoGE","EoGE","EoGE","Atlas","Atlas","Atlas","BodyMap","BodyMap","BodyMap","HPA","HPA","HPA","AltIso","AltIso"),prep=c(rep("poly-A",3),rep("rRNA-depl",3),rep("poly-A",8)),layout=c(rep("PE",3),rep("SE",3),rep("PE",6),rep("SE",2)))
rownames(meta) <- colnames(f_pc_nozero)[3:16]
tissue <- rep(meta$tissue, each=nrow(f_pc_nozero))
study <- rep(meta$study, each=nrow(f_pc_nozero))
prep <- rep(meta$prep, each=nrow(f_pc_nozero))
layout <- rep(meta$layout, each=nrow(f_pc_nozero))
data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)

#subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(Cuff_FPKM ~ prep + layout + study + tissue, data=data)
a <- anova(fit)
maxval = 100
```

**Figure 4C**

```{r}
barplot(a$"F value"[-5],names.arg=rownames(a)[-5],main="Anova F score, Cufflinks FPKM",ylim=c(0,maxval))
```

**Figure 4D (?)**

ANOVA analyses on logged values:

```{r:anova-log}
m <- melt(fpkms.log[,3:ncol(fpkms.log)])
colnames(m) <- c("sample_ID","log2FPKM")

data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)
#subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(log2FPKM ~ prep + layout + study + tissue, data=data)
b <- anova(fit)

barplot(b$"F value"[-5],names.arg=rownames(b)[-5],main="Anova F score, log2-RPKM",ylim=c(0,3000))
print(b)
```

Finally, ANOVA on ComBat.

```{r:anova-combat}
m <- melt(combat)
colnames(m) <- c("sample_ID","combat")
data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)
#subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(combat ~ + prep + layout + study + tissue, data=data)
c <- anova(fit)

barplot(c$"F value"[-5],names.arg=rownames(c)[-5],main="Anova F score, ComBat/log2-FPKM",ylim=c(0,5000))
print(c)
```

**Figure X**


Let's have a look at the 100 most highly expressed genes in each sample and see how many of these genes that are shared between the studies


```{r:venn }

library(VennDiagram)

EoGE_b <- rownames(f_pc[order(f_pc$EoGE_brain,decreasing=T),][1:100,])
EoGE_h <- rownames(f_pc[order(f_pc$EoGE_heart,decreasing=T),][1:100,])
EoGE_k <- rownames(f_pc[order(f_pc$EoGE_kidney,decreasing=T),][1:100,])

Atlas_b <- rownames(f_pc[order(f_pc$Atlas_brain,decreasing=T),][1:100,])
Atlas_h <- rownames(f_pc[order(f_pc$Atlas_heart,decreasing=T),][1:100,])
Atlas_k <- rownames(f_pc[order(f_pc$Atlas_kidney,decreasing=T),][1:100,])

BodyMap_b <- rownames(f_pc[order(f_pc$BodyMap_brain,decreasing=T),][1:100,])
BodyMap_h <- rownames(f_pc[order(f_pc$BodyMap_heart,decreasing=T),][1:100,])
BodyMap_k <- rownames(f_pc[order(f_pc$BodyMap_kidney,decreasing=T),][1:100,])

HPA_b <- rownames(f_pc[order(f_pc$HPA_brain,decreasing=T),][1:100,])
HPA_h <- rownames(f_pc[order(f_pc$HPA_heart,decreasing=T),][1:100,])
HPA_k <- rownames(f_pc[order(f_pc$HPA_kidney,decreasing=T),][1:100,])

AltIso_b <- rownames(f_pc[order(f_pc$AltIso_brain,decreasing=T),][1:100,])
AltIso_h <- rownames(f_pc[order(f_pc$AltIso_heart,decreasing=T),][1:100,])

```

Let's start with the five brain samples:

```{r:venn brain}

draw.quintuple.venn(100, 100, 100, 100, 100, 
               length(intersect(EoGE_b,Atlas_b)),
               length(intersect(EoGE_b,BodyMap_b)),
               length(intersect(EoGE_b,HPA_b)),
               length(intersect(EoGE_b,AltIso_b)),
               length(intersect(Atlas_b,BodyMap_b)),
               length(intersect(Atlas_b,HPA_b)),
               length(intersect(Atlas_b,AltIso_b)),
               length(intersect(BodyMap_b,HPA_b)),
               length(intersect(BodyMap_b,AltIso_b)),
               length(intersect(HPA_b,AltIso_b)),
               length(intersect(intersect(EoGE_b,Atlas_b),BodyMap_b)),
               length(intersect(intersect(EoGE_b,Atlas_b),HPA_b)),
               length(intersect(intersect(EoGE_b,Atlas_b),AltIso_b)),
               length(intersect(intersect(EoGE_b,BodyMap_b),HPA_b)),
               length(intersect(intersect(EoGE_b,BodyMap_b),AltIso_b)),
               length(intersect(intersect(EoGE_b,HPA_b),AltIso_b)),
               length(intersect(intersect(Atlas_b,BodyMap_b),HPA_b)),
               length(intersect(intersect(Atlas_b,BodyMap_b),AltIso_b)),
               length(intersect(intersect(Atlas_b,HPA_b),AltIso_b)),
               length(intersect(intersect(BodyMap_b,HPA_b),AltIso_b)),
               length(intersect(intersect(EoGE_b,Atlas_b),intersect(BodyMap_b,HPA_b))),
               length(intersect(intersect(EoGE_b,Atlas_b),intersect(BodyMap_b,AltIso_b))),
               length(intersect(intersect(EoGE_b,Atlas_b),intersect(HPA_b,AltIso_b))),
               length(intersect(intersect(EoGE_b,BodyMap_b),intersect(HPA_b,AltIso_b))),
               length(intersect(intersect(Atlas_b,BodyMap_b),intersect(HPA_b,AltIso_b))),
               length(intersect(intersect(intersect(EoGE_b,Atlas_b),intersect(BodyMap_b,HPA_b)),AltIso_b)),
               category = c("EoGE","Atlas","BodyMap","HPA","AltIso"), lwd = rep(0, 5), lty = rep("solid", 5),
               fill = c("mistyrose","steelblue","lightgoldenrod","darkseagreen","lightblue")
)

```

and the five heart samples:

```{r:venn heart}
draw.quintuple.venn(100, 100, 100, 100, 100, 
               length(intersect(EoGE_h,Atlas_h)),
               length(intersect(EoGE_h,BodyMap_h)),
               length(intersect(EoGE_h,HPA_h)),
               length(intersect(EoGE_h,AltIso_h)),
               length(intersect(Atlas_h,BodyMap_h)),
               length(intersect(Atlas_h,HPA_h)),
               length(intersect(Atlas_h,AltIso_h)),
               length(intersect(BodyMap_h,HPA_h)),
               length(intersect(BodyMap_h,AltIso_h)),
               length(intersect(HPA_h,AltIso_h)),
               length(intersect(intersect(EoGE_h,Atlas_h),BodyMap_h)),
               length(intersect(intersect(EoGE_h,Atlas_h),HPA_h)),
               length(intersect(intersect(EoGE_h,Atlas_h),AltIso_h)),
               length(intersect(intersect(EoGE_h,BodyMap_h),HPA_h)),
               length(intersect(intersect(EoGE_h,BodyMap_h),AltIso_h)),
               length(intersect(intersect(EoGE_h,HPA_h),AltIso_h)),
               length(intersect(intersect(Atlas_h,BodyMap_h),HPA_h)),
               length(intersect(intersect(Atlas_h,BodyMap_h),AltIso_h)),
               length(intersect(intersect(Atlas_h,HPA_h),AltIso_h)),
               length(intersect(intersect(BodyMap_h,HPA_h),AltIso_h)),
               length(intersect(intersect(EoGE_h,Atlas_h),intersect(BodyMap_h,HPA_h))),
               length(intersect(intersect(EoGE_h,Atlas_h),intersect(BodyMap_h,AltIso_h))),
               length(intersect(intersect(EoGE_h,Atlas_h),intersect(HPA_h,AltIso_h))),
               length(intersect(intersect(EoGE_h,BodyMap_h),intersect(HPA_h,AltIso_h))),
               length(intersect(intersect(Atlas_h,BodyMap_h),intersect(HPA_h,AltIso_h))),
               length(intersect(intersect(intersect(EoGE_h,Atlas_h),intersect(BodyMap_h,HPA_h)),AltIso_h)),
               category = c("EoGE","Atlas","BodyMap","HPA","AltIso"), lwd = rep(0, 5), lty = rep("solid", 5),
               fill = c("mistyrose","steelblue","lightgoldenrod","darkseagreen","lightblue")
)



```

...and the four kidney samples:

```{r:venn kidney}

draw.quad.venn(100, 100, 100, 100, 
               length(intersect(EoGE_k,Atlas_k)),
               length(intersect(EoGE_k,BodyMap_k)),
               length(intersect(EoGE_k,HPA_k)),
               length(intersect(Atlas_k,BodyMap_k)),
               length(intersect(Atlas_k,HPA_k)),
               length(intersect(BodyMap_k,HPA_k)),
               length(intersect(intersect(EoGE_k,Atlas_k),BodyMap_k)),
               length(intersect(intersect(EoGE_k,Atlas_k),HPA_k)),
               length(intersect(intersect(EoGE_k,BodyMap_k),HPA_k)),
               length(intersect(intersect(Atlas_k,BodyMap_k),HPA_k)),
               length(intersect(intersect(EoGE_k,Atlas_k),intersect(BodyMap_k,HPA_k))),
               category = c("EoGE","Atlas","BodyMap","HPA"), lwd = rep(0, 4), lty = rep("solid", 4),
               fill = c("mistyrose","steelblue","lightgoldenrod","darkseagreen")
)

```

Let's have a look again at the 100 most highly expressed genes in each sample and see how many of these genes that are shared between the studies, but this time looking at the values after the ComBat run:

```{r:combatVenn}

combat_data <- cbind(fpkms.log[,1:2],combat)

EoGE_bc <- rownames(combat_data[order(combat_data$EoGE_brain,decreasing=T),][1:100,])
EoGE_hc <- rownames(combat_data[order(combat_data$EoGE_heart,decreasing=T),][1:100,])
EoGE_kc <- rownames(combat_data[order(combat_data$EoGE_kidney,decreasing=T),][1:100,])

Atlas_bc <- rownames(combat_data[order(combat_data$Atlas_brain,decreasing=T),][1:100,])
Atlas_hc <- rownames(combat_data[order(combat_data$Atlas_heart,decreasing=T),][1:100,])
Atlas_kc <- rownames(combat_data[order(combat_data$Atlas_kidney,decreasing=T),][1:100,])

BodyMap_bc <- rownames(combat_data[order(combat_data$BodyMap_brain,decreasing=T),][1:100,])
BodyMap_hc <- rownames(combat_data[order(combat_data$BodyMap_heart,decreasing=T),][1:100,])
BodyMap_kc <- rownames(combat_data[order(combat_data$BodyMap_kidney,decreasing=T),][1:100,])

HPA_bc <- rownames(combat_data[order(combat_data$HPA_brain,decreasing=T),][1:100,])
HPA_hc <- rownames(combat_data[order(combat_data$HPA_heart,decreasing=T),][1:100,])
HPA_kc <- rownames(combat_data[order(combat_data$HPA_kidney,decreasing=T),][1:100,])

AltIso_bc <- rownames(combat_data[order(combat_data$AltIso_brain,decreasing=T),][1:100,])
AltIso_hc <- rownames(combat_data[order(combat_data$AltIso_heart,decreasing=T),][1:100,])


```

First the five brain samples:

```{r:combatVennBrain}

draw.quintuple.venn(100, 100, 100, 100, 100, 
               length(intersect(EoGE_bc,Atlas_bc)),
               length(intersect(EoGE_bc,BodyMap_bc)),
               length(intersect(EoGE_bc,HPA_bc)),
               length(intersect(EoGE_bc,AltIso_bc)),
               length(intersect(Atlas_bc,BodyMap_bc)),
               length(intersect(Atlas_bc,HPA_bc)),
               length(intersect(Atlas_bc,AltIso_bc)),
               length(intersect(BodyMap_bc,HPA_bc)),
               length(intersect(BodyMap_bc,AltIso_bc)),
               length(intersect(HPA_bc,AltIso_bc)),
               length(intersect(intersect(EoGE_bc,Atlas_bc),BodyMap_bc)),
               length(intersect(intersect(EoGE_bc,Atlas_bc),HPA_bc)),
               length(intersect(intersect(EoGE_bc,Atlas_bc),AltIso_bc)),
               length(intersect(intersect(EoGE_bc,BodyMap_bc),HPA_bc)),
               length(intersect(intersect(EoGE_bc,BodyMap_bc),AltIso_bc)),
               length(intersect(intersect(EoGE_bc,HPA_bc),AltIso_bc)),
               length(intersect(intersect(Atlas_bc,BodyMap_bc),HPA_bc)),
               length(intersect(intersect(Atlas_bc,BodyMap_bc),AltIso_bc)),
               length(intersect(intersect(Atlas_bc,HPA_bc),AltIso_bc)),
               length(intersect(intersect(BodyMap_bc,HPA_bc),AltIso_bc)),
               length(intersect(intersect(EoGE_bc,Atlas_bc),intersect(BodyMap_bc,HPA_bc))),
               length(intersect(intersect(EoGE_bc,Atlas_bc),intersect(BodyMap_bc,AltIso_bc))),
               length(intersect(intersect(EoGE_bc,Atlas_bc),intersect(HPA_bc,AltIso_bc))),
               length(intersect(intersect(EoGE_bc,BodyMap_bc),intersect(HPA_bc,AltIso_bc))),
               length(intersect(intersect(Atlas_bc,BodyMap_bc),intersect(HPA_bc,AltIso_bc))),
               length(intersect(intersect(intersect(EoGE_bc,Atlas_bc),intersect(BodyMap_bc,HPA_bc)),AltIso_bc)),
               category = c("EoGE","Atlas","BodyMap","HPA","AltIso"), lwd = rep(0, 5), lty = rep("solid", 5),
               fill = c("mistyrose","steelblue","lightgoldenrod","darkseagreen","lightblue")
)

```

And for the five heart samples:

```{r:combatVennHeart}

draw.quintuple.venn(100, 100, 100, 100, 100, 
               length(intersect(EoGE_hc,Atlas_hc)),
               length(intersect(EoGE_hc,BodyMap_hc)),
               length(intersect(EoGE_hc,HPA_hc)),
               length(intersect(EoGE_hc,AltIso_hc)),
               length(intersect(Atlas_hc,BodyMap_hc)),
               length(intersect(Atlas_hc,HPA_hc)),
               length(intersect(Atlas_hc,AltIso_hc)),
               length(intersect(BodyMap_hc,HPA_hc)),
               length(intersect(BodyMap_hc,AltIso_hc)),
               length(intersect(HPA_hc,AltIso_hc)),
               length(intersect(intersect(EoGE_hc,Atlas_hc),BodyMap_hc)),
               length(intersect(intersect(EoGE_hc,Atlas_hc),HPA_hc)),
               length(intersect(intersect(EoGE_hc,Atlas_hc),AltIso_hc)),
               length(intersect(intersect(EoGE_hc,BodyMap_hc),HPA_hc)),
               length(intersect(intersect(EoGE_hc,BodyMap_hc),AltIso_hc)),
               length(intersect(intersect(EoGE_hc,HPA_hc),AltIso_hc)),
               length(intersect(intersect(Atlas_hc,BodyMap_hc),HPA_hc)),
               length(intersect(intersect(Atlas_hc,BodyMap_hc),AltIso_hc)),
               length(intersect(intersect(Atlas_hc,HPA_hc),AltIso_hc)),
               length(intersect(intersect(BodyMap_hc,HPA_hc),AltIso_hc)),
               length(intersect(intersect(EoGE_hc,Atlas_hc),intersect(BodyMap_hc,HPA_hc))),
               length(intersect(intersect(EoGE_hc,Atlas_hc),intersect(BodyMap_hc,AltIso_hc))),
               length(intersect(intersect(EoGE_hc,Atlas_hc),intersect(HPA_hc,AltIso_hc))),
               length(intersect(intersect(EoGE_hc,BodyMap_hc),intersect(HPA_hc,AltIso_hc))),
               length(intersect(intersect(Atlas_hc,BodyMap_hc),intersect(HPA_hc,AltIso_hc))),
               length(intersect(intersect(intersect(EoGE_hc,Atlas_hc),intersect(BodyMap_hc,HPA_hc)),AltIso_hc)),
               category = c("EoGE","Atlas","BodyMap","HPA","AltIso"), lwd = rep(0, 5), lty = rep("solid", 5),
               fill = c("mistyrose","steelblue","lightgoldenrod","darkseagreen","lightblue")
)

```

And for the four kidney samples:

```{r:combatVennKidney}

draw.quad.venn(100, 100, 100, 100, 
               length(intersect(EoGE_kc,Atlas_kc)),
               length(intersect(EoGE_kc,BodyMap_kc)),
               length(intersect(EoGE_kc,HPA_kc)),
               length(intersect(Atlas_kc,BodyMap_kc)),
               length(intersect(Atlas_kc,HPA_kc)),
               length(intersect(BodyMap_kc,HPA_kc)),
               length(intersect(intersect(EoGE_kc,Atlas_kc),BodyMap_kc)),
               length(intersect(intersect(EoGE_kc,Atlas_kc),HPA_kc)),
               length(intersect(intersect(EoGE_kc,BodyMap_kc),HPA_kc)),
               length(intersect(intersect(Atlas_kc,BodyMap_kc),HPA_kc)),
               length(intersect(intersect(EoGE_kc,Atlas_kc),intersect(BodyMap_kc,HPA_kc))),
               category = c("EoGE","Atlas","BodyMap","HPA"), lwd = rep(0, 4), lty = rep("solid", 4),
               fill = c("mistyrose","steelblue","lightgoldenrod","darkseagreen")
)

```{r:PC-correlation}
print_PCA_SVD_corrs <- function(data){
pca <- prcomp(t(data[,]))

#var.percent <- ((pca$sdev)^2)/sum(pca$sdev^2) *100
#var.percent <- (pca$d^2 / sum(pca$d^2) ) *100
#barplot(var.percent[1:5], xlab="PC", ylab="Percent Variance",names.arg=1:length(var.percent[1:5]), las=1,ylim=c(0,max(var.percent[1:5])+10), col="gray")

rot <- pca$r
x <- pca$x
#plot(pca)
#summary(pca)
#screeplot(pca,type=c("lines"))
# Test correlations between number of seq'd reads and PCs 1-4 from prcomp
pval.nraw.pc1 <- cor.test(x[,1], sampleinfo$NumberRaw,method="spearman")$p.value
pval.nraw.pc2 <- cor.test(x[,2], sampleinfo$NumberRaw,method="spearman")$p.value
pval.nraw.pc3 <- cor.test(x[,3], sampleinfo$NumberRaw,method="spearman")$p.value
pval.nraw.pc4 <- cor.test(x[,4], sampleinfo$NumberRaw,method="spearman")$p.value

cat(sprintf("Number_of_rawreads~PCAs: PCA1=\"%f\"PCA2=\"%f\"PCA3=\"%f\"PCA4=\"%f\n", pval.nraw.pc1,pval.nraw.pc2,pval.nraw.pc3,pval.nraw.pc4))

# Thus no significant correlations between no of seq'd reads and PCs 1-4

pval.nmapped.pc1 <- cor.test(x[,1], sampleinfo$Numbermapped,method="spearman")$p.value
pval.nmapped.pc2 <- cor.test(x[,2], sampleinfo$Numbermapped,method="spearman")$p.value
pval.nmapped.pc3 <- cor.test(x[,3], sampleinfo$Numbermapped,method="spearman")$p.value
pval.nmapped.pc4 <- cor.test(x[,4], sampleinfo$Numbermapped,method="spearman")$p.value

cat(sprintf("Number_of_mappedreads~PCAs: PCA1=\"%f\"PCA2=\"%f\"PCA3=\"%f\"PCA4=\"%f\n", pval.nmapped.pc1,pval.nmapped.pc2,pval.nmapped.pc3,pval.nmapped.pc4))

# For tissue, use kruskal.test which handles ordinal variables 
pval.tissue.pc1<-kruskal.test(x[,1], sampleinfo$Tissue)$p.value
pval.tissue.pc2<-kruskal.test(x[,2], sampleinfo$Tissue)$p.value
pval.tissue.pc3<-kruskal.test(x[,3], sampleinfo$Tissue)$p.value
pval.tissue.pc4<-kruskal.test(x[,4], sampleinfo$Tissue)$p.value

cat(sprintf("Tissues~PCAs: PCA1=\"%f\"PCA2=\"%f\"PCA3=\"%f\"\n", pval.tissue.pc1,pval.tissue.pc2,pval.tissue.pc3,pval.tissue.pc4))

# Library prep 
pval.prep.pc1<-kruskal.test(x[,1], sampleinfo$Preparation)$p.value
pval.prep.pc2<-kruskal.test(x[,2], sampleinfo$Preparation)$p.value
pval.prep.pc3<-kruskal.test(x[,3], sampleinfo$Preparation)$p.value
pval.prep.pc4<-kruskal.test(x[,4], sampleinfo$Preparation)$p.value

cat(sprintf("LibPrep~PCAs: PCA1=\"%f\"PCA2=\"%f\"PCA3=\"%f\"\n", pval.prep.pc1,pval.prep.pc2,pval.prep.pc3,pval.prep.pc4))

# Study  
pval.study.pc1<-kruskal.test(x[,1], sampleinfo$Study)$p.value
pval.study.pc2<-kruskal.test(x[,2], sampleinfo$Study)$p.value
pval.study.pc3<-kruskal.test(x[,3], sampleinfo$Study)$p.value
pval.study.pc4<-kruskal.test(x[,4], sampleinfo$Study)$p.value

cat(sprintf("Study~PCAs: PCA1=\"%f\"PCA2=\"%f\"PCA3=\"%f\"\n", pval.study.pc1,pval.study.pc2,pval.study.pc3,pval.study.pc4))

# Layout
pval.layout.pc1<-kruskal.test(x[,1], sampleinfo$readlength)$p.value
pval.layout.pc2<-kruskal.test(x[,2], sampleinfo$readlength)$p.value
pval.layout.pc3<-kruskal.test(x[,3], sampleinfo$readlength)$p.value
pval.layout.pc4<-kruskal.test(x[,4], sampleinfo$readlengt)$p.value

cat(sprintf("ReadType~PCAs: PCA1=\"%f\"PCA2=\"%f\"PCA3=\"%f\"\n", pval.layout.pc1,pval.layout.pc2,pval.layout.pc3,pval.layout.pc4))

# To test:
# same for logged values
# same for ComBat
# svd() without mean subtraction instead

#svd.12 <- do.SVD(f.nozero, comp.1=1, comp.2=2)
#svd.scores.12 <- project.SVD(f.nozero, svd.12)
svd.12 <- do.SVD(data, comp.1=1, comp.2=2)
svd.scores.12 <- project.SVD(data, svd.12)
# Number of raw reads
pval.nraw.svd1 <- cor.test(svd.scores.12[,1], sampleinfo$NumberRaw,method="spearman")$p.value
pval.nraw.svd2 <- cor.test(svd.scores.12[,2], sampleinfo$NumberRaw,method="spearman")$p.value
cat(sprintf("RawReads~SVDcomps: SVD-PC1=\"%f\"SVD-PC2=\"%f\"\n", pval.nraw.svd1,pval.nraw.svd2))
# Number of mapped reads
pval.nmapped.svd1 <- cor.test(svd.scores.12[,1], sampleinfo$Numbermapped,method="spearman")$p.value
pval.nmapped.svd2 <- cor.test(svd.scores.12[,2], sampleinfo$Numbermapped,method="spearman")$p.value
cat(sprintf("MappedReads~SVDcomps: SVD-PC1=\"%f\"SVD-PC2=\"%f\"\n", pval.nmapped.svd1,pval.nmapped.svd2))
# Layout / read length
pval.layout.svd1 <- kruskal.test(svd.scores.12[,1], sampleinfo$Readtype)$p.value
pval.layout.svd2 <- kruskal.test(svd.scores.12[,2], sampleinfo$Readtype)$p.value
cat(sprintf("Layout~SVDcomps: SVD-PC1=\"%f\"SVD-PC2=\"%f\"\n", pval.layout.svd1,pval.layout.svd2))
# Prep
pval.prep.svd1 <- kruskal.test(svd.scores.12[,1], sampleinfo$Preparation)$p.value
pval.prep.svd2 <- kruskal.test(svd.scores.12[,2], sampleinfo$Preparation)$p.value
cat(sprintf("Prep~SVDcomps: SVD-PC1=\"%f\"SVD-PC2=\"%f\"\n", pval.prep.svd1,pval.prep.svd2))
# Study
pval.study.svd1 <- kruskal.test(svd.scores.12[,1], sampleinfo$Study)$p.value
pval.study.svd2 <- kruskal.test(svd.scores.12[,2], sampleinfo$Study)$p.value
cat(sprintf("Study~SVDcomps: SVD-PC1=\"%f\"SVD-PC2=\"%f\"\n", pval.study.svd1,pval.study.svd2))
# Tissue
pval.tissue.svd1 <- kruskal.test(svd.scores.12[,1], sampleinfo$Tissue)$p.value
pval.tissue.svd2 <- kruskal.test(svd.scores.12[,2], sampleinfo$Tissue)$p.value
cat(sprintf("Tissue~SVDcomps: SVD-PC1=\"%f\"SVD-PC2=\"%f\"\n", pval.tissue.svd1,pval.tissue.svd2))
}
```

Test raw values, logged values and ComBat-processed values.
```{r:testPCcorrelations}
print_PCA_SVD_corrs(f_pc_nozero[,3:16])
print_PCA_SVD_corrs(fpkms.log[,3:16])
print_PCA_SVD_corrs(combat)
```

