<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>Downloading the F/RPKM data</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: rgb(88, 72, 246)
   }

   pre .number {
     color: rgb(0, 0, 205);
   }

   pre .comment {
     color: rgb(76, 136, 107);
   }

   pre .keyword {
     color: rgb(0, 0, 255);
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: rgb(3, 106, 7);
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>




</head>

<body>
<h2>Downloading the F/RPKM data</h2>

<p>Prepare by defining functions etc.</p>

<pre><code class="r">library(edgeR)
</code></pre>

<pre><code>## Loading required package: limma
</code></pre>

<pre><code class="r">library(pheatmap)
library(gplots)
</code></pre>

<pre><code>## KernSmooth 2.23 loaded
## Copyright M. P. Wand 1997-2009
## 
## Attaching package: &#39;gplots&#39;
## 
## The following object is masked from &#39;package:stats&#39;:
## 
##     lowess
</code></pre>

<pre><code class="r">library(ops)
</code></pre>

<pre><code>## 
## Attaching package: &#39;ops&#39;
## 
## The following object is masked from &#39;package:stats&#39;:
## 
##     filter
</code></pre>

<pre><code class="r">library(calibrate)
</code></pre>

<pre><code>## Loading required package: MASS
</code></pre>

<pre><code class="r">normalize.voom &lt;- function(counts){
  require(limma)
  return(voom(counts)$E)
}

cpm.tmm &lt;- function(counts, groups=NA){
    require(edgeR)
    if(is.na(groups)){
        d&lt;-DGEList(counts=counts)
    }
    else{
        d&lt;-DGEList(counts=counts, group=groups)
    }
    d &lt;- calcNormFactors(d, method=&quot;TMM&quot;) 
    return(cpm(d, normalized.lib.sizes=TRUE))
}

log2.cpm &lt;- function(x){
  x.pseud &lt;- x + 0.5
  libsizes = colSums(x)
  libsize.pseud &lt;- libsizes + 1.0
  new.mtx &lt;- log2(1e6 * mapply(&quot;/&quot;,as.data.frame(x.pseud),libsize.pseud))
  rownames(new.mtx) &lt;- rownames(x)
  return(as.matrix(new.mtx))
}

do.SVD = function(m, comp.1=1, comp.2=2){ # returns eig.cell
  s &lt;- svd(m)
    ev &lt;- s$d^2 / sum(s$d^2)
    return(s$u[,c(comp.1, comp.2)])
}

project.SVD &lt;- function(m, eig.cell){
    return(t(m) %*% eig.cell)
}

plot.SVD &lt;- function(m, comp.1=1, comp.2=2, groups=rep(&quot;blue&quot;, ncol(m)), title=&quot;&quot;){
    eig &lt;- do.SVD(m, comp.1, comp.2)
    proj &lt;- project.SVD(m, eig)
    xminv &lt;- min(proj[,1]) # - .2 * abs(min(proj[,1]))
    xmaxv &lt;- max(proj[,1]) # + .2 * abs(max(proj[,1]))
    yminv &lt;- min(proj[,2]) # - .2 * abs(min(proj[,2]))
    ymaxv &lt;- max(proj[,2]) # + .2 * abs(max(proj[,2]))
    plot(proj,pch=20,col=&quot;white&quot;,xlim=c(xminv,xmaxv),ylim=c(yminv,ymaxv),xaxt=&#39;n&#39;,yaxt=&#39;n&#39;,xlab=&quot;PC1&quot;,ylab=&quot;PC2&quot;,main=title)

    points(proj, col=as.character(groups),pch=20) # , #pch=c(rep(15,3),rep(17,3),rep(19,3),rep(18,3),rep(20,2)), cex=2)
    textxy(proj[,1],proj[,2],labs=colnames(m))
}

loadings.SVD &lt;- function(m, comp=1, gene.ids = rownames(m)){
    s &lt;- svd(m)
    l &lt;- s$u[,comp]
    names(l) &lt;- gene.ids
    l.s &lt;- l[order(l)]
    return(l.s)
}

plot.loadings.SVD &lt;- function(m, comp=1, cutoff=0.1, gene.ids = rownames(m)){
    l &lt;- loadings.SVD(m, comp, gene.ids)
    barplot(l[abs(l)&gt;cutoff],las=2,main=paste(&quot;PC&quot;, comp, &quot;cutoff&quot;, cutoff),cex.names=0.6)
}

plotPC &lt;- function(matrix,a,b,desc,colors){
eig &lt;- do.SVD(matrix, a, b)
proj &lt;- project.SVD(matrix, eig)
xminv &lt;- min(proj[,1]) - .2 * abs(min(proj[,1]))
xmaxv &lt;- max(proj[,1]) + .2 * abs(max(proj[,1]))
yminv &lt;- min(proj[,2]) - .2 * abs(min(proj[,2]))
ymaxv &lt;- max(proj[,2]) + .2 * abs(max(proj[,2]))
plot(proj,pch=20,xlim=c(xminv,xmaxv),ylim=c(yminv,ymaxv),xaxt=&#39;n&#39;,yaxt=&#39;n&#39;,xlab=paste0(&quot;PC&quot;,a),ylab=paste(&quot;PC&quot;,b),col=colors,main=desc)
textxy(proj[,1],proj[,2],labs=rownames(proj))
}
</code></pre>

<p>Here, we download data from various public sources and extract the brain, heart and kidney samples.</p>

<p>&ldquo;HPA&rdquo;: Human Protein Atlas</p>

<pre><code class="r">#temp &lt;- tempfile()
#download.file(url=&quot;http://www.proteinatlas.org/download/rna.csv.zip&quot;,destfile=temp)
#hpa &lt;- read.csv(unz(temp, &quot;rna.csv&quot;))
#unlink(temp)

#hpa.heart &lt;- hpa[hpa$Sample==&quot;heart muscle&quot;, c(&quot;Gene&quot;, &quot;Value&quot;)]
#hpa.brain &lt;- hpa[hpa$Sample==&quot;cerebral cortex&quot;, c(&quot;Gene&quot;, &quot;Value&quot;)]
#hpa.kidney &lt;- hpa[hpa$Sample==&quot;kidney&quot;, c(&quot;Gene&quot;, &quot;Value&quot;)]

#hpa.fpkms &lt;- merge(hpa.heart, hpa.brain, by=&quot;Gene&quot;)
#hpa.fpkms &lt;- merge(hpa.fpkms, hpa.kidney, by=&quot;Gene&quot;)
#colnames(hpa.fpkms) &lt;- c(&quot;ENSG_ID&quot;, &quot;HPA_heart&quot;, &quot;HPA_brain&quot;, &quot;HPA_kidney&quot;)
</code></pre>

<p>Check if the identifiers are unique and write table to file.</p>

<pre><code class="r">#length(hpa.fpkms[,1])
#length(unique(hpa.fpkms[,1]))

#write.table(hpa.fpkms,file=&quot;hpa_fpkms.txt&quot;,quote=F,sep=&quot;\t&quot;)
</code></pre>

<p>&ldquo;Altiso&rdquo;: Alternative isoform regulation in human tissue transcriptomes</p>

<pre><code class="r">#temp &lt;- tempfile()
#download.file(url=&quot;http://genes.mit.edu/burgelab/Supplementary/wang_sandberg08/hg18.ensGene.CEs.rpkm.txt&quot;,destfile=temp)
#altiso &lt;- read.delim(temp, sep=&quot;\t&quot;)
#unlink(temp)
</code></pre>

<p>There is no kidney sample here, so just use heart + brain</p>

<pre><code class="r">#altiso.fpkms &lt;- altiso[,c(&quot;X.Gene&quot;,&quot;heart&quot;,&quot;brain&quot;)]
#colnames(altiso.fpkms) &lt;- c(&quot;ENSG_ID&quot;, &quot;AltIso_heart&quot;, &quot;AltIso_brain&quot;)
</code></pre>

<p>Check uniqueness of IDs.</p>

<pre><code class="r">#length(altiso.fpkms[,1])
#length(unique(altiso.fpkms[,1]))

#write.table(altiso.fpkms,file=&quot;altiso_fpkms.txt&quot;,quote=F,sep=&quot;\t&quot;)
</code></pre>

<p>&ldquo;GTEx&rdquo;: Genotype-Tissue Expression</p>

<p>This is a big download: 337.8 Mb (as of 2014-02-04)
We also add some code to randomly select one sample from each tissue type; there are many biological replicates in this data set.</p>

<pre><code class="r">#temp &lt;- tempfile()
#download.file(url=&quot;http://www.broadinstitute.org/gtex/rest/file/download?portalFileId=119363&amp;forDownload=true&quot;,destfile=temp)
#header_lines &lt;- readLines(temp, n=2)
#gtex &lt;- read.delim(temp, skip=2, sep=&quot;\t&quot;)
#unlink(temp)

#write.table(gtex, file=&quot;gtex_all.txt&quot;,     quote=F, sep=&quot;\t&quot;)

#download.file(url=&quot;http://www.broadinstitute.org/gtex/rest/file/download?portalFileId=119273&amp;forDownload=true&quot;,destfile=&quot;GTEx_description.txt&quot;)

#metadata &lt;- read.delim(&quot;GTEx_description.txt&quot;, sep=&quot;\t&quot;)
</code></pre>

<p>The metadata table seems to contain entries that are not in the RPKM table.</p>

<pre><code class="r">#samp.id &lt;- gsub(&#39;-&#39;,&#39;.&#39;,metadata$SAMPID)
#eligible.samples &lt;- which(samp.id %in% colnames(gtex))
#metadata &lt;- metadata[eligible.samples,]
</code></pre>

<p>Select random heart, kidney and brain samples.</p>

<pre><code class="r">#random.heart &lt;- sample(which(metadata$SMTS==&quot;Heart&quot;), size=1)
#random.heart.samplename &lt;- gsub(&#39;-&#39;,&#39;.&#39;,metadata[random.heart, &quot;SAMPID&quot;])
#gtex.heart.fpkm &lt;- as.numeric(gtex[,random.heart.samplename])

#random.brain &lt;- sample(which(metadata$SMTS==&quot;Brain&quot;), size=1)
#random.brain.samplename &lt;- gsub(&#39;-&#39;,&#39;.&#39;,metadata[random.brain, &quot;SAMPID&quot;])
#gtex.brain.fpkm &lt;- as.numeric(gtex[,random.brain.samplename])

#random.kidney &lt;- sample(which(metadata$SMTS==&quot;Kidney&quot;), size=1)
#random.kidney.samplename &lt;- gsub(&#39;-&#39;,&#39;.&#39;,metadata[random.kidney, &quot;SAMPID&quot;])
#gtex.kidney.fpkm &lt;- as.numeric(gtex[,random.kidney.samplename])
</code></pre>

<p>Get gene IDs on same format as the other data sets by removing the part after the dot; check ID uniqueness and write to file.</p>

<pre><code class="r">#gtex.names &lt;- gtex[,&quot;Name&quot;]
#temp_list &lt;- strsplit(as.character(gtex.names), split=&quot;\\.&quot;)
#gtex.names.nodot &lt;- unlist(temp_list)[2*(1:length(gtex.names))-1]

#gtex.fpkms &lt;- data.frame(ENSG_ID=gtex.names.nodot, GTEx_heart=gtex.heart.fpkm, GTEx_brain=gtex.brain.fpkm,GTEx_kidney=gtex.kidney.fpkm)

#length(gtex.fpkms[,1])
#length(unique(gtex.fpkms[,1]))

#write.table(gtex.fpkms,file=&quot;gtex_fpkms.txt&quot;,quote=F,sep=&quot;\t&quot;)
</code></pre>

<p><em>RNA-seq Atlas</em></p>

<pre><code class="r">#temp &lt;- tempfile()
#download.file(url=&quot;http://medicalgenomics.org/rna_seq_atlas/download?download_revision1=1&quot;,destfile=temp)
#atlas &lt;- read.delim(temp, sep=&quot;\t&quot;)
#unlink(temp)

#atlas.fpkms &lt;- atlas[,c(&quot;ensembl_gene_id&quot;,&quot;heart&quot;,&quot;hypothalamus&quot;,&quot;kidney&quot;)]
#colnames(atlas.fpkms) &lt;- c(&quot;ENSG_ID&quot;,&quot;Atlas_heart&quot;,&quot;Atlas_brain&quot;,&quot;Atlas_kidney&quot;)
#write.table(atlas.fpkms,file=&quot;atlas_fpkms.txt&quot;,quote=F,sep=&quot;\t&quot;)
</code></pre>

<h2>Combining F/RPKM values from public data sets</h2>

<p>We will join the data sets on ENSEMBL ID:s, losing a lot of data in the process - but joining on gene symbols or something else would lead to an even worse loss. </p>

<pre><code class="r">library(org.Hs.eg.db) # for transferring gene identifiers
</code></pre>

<pre><code>## Error: there is no package called &#39;org.Hs.eg.db&#39;
</code></pre>

<pre><code class="r">library(data.table) # for collapsing transcript RPKMs
</code></pre>

<pre><code>## Error: there is no package called &#39;data.table&#39;
</code></pre>

<pre><code class="r">library(pheatmap) # for nicer visualization
library(edgeR) # for TMM normalization

#hpa.fpkms &lt;- read.delim(&quot;hpa_fpkms.txt&quot;)
#altiso.fpkms &lt;- read.delim(&quot;altiso_fpkms.txt&quot;)
#gtex.fpkms &lt;- read.delim(&quot;gtex_fpkms.txt&quot;)
#atlas.fpkms &lt;- read.delim(&quot;atlas_fpkms.txt&quot;)
</code></pre>

<p>The RNA-seq Atlas data set uses many different identifiers, while the other all use ENSG as the primary identifier</p>

<p>Approach 1: Merge on ENSEMBL genes (ENSG) as given in RNA-seq Atlas. Note that there are repeated ENSG ID:s in RNA-seq Atlas, as opposed to the other data sets, so we need to do something about that. In this case, we just sum the transcripts that belong to each ENSG gene. We use data.table for this.</p>

<pre><code class="r">#data.dt &lt;- data.table(atlas.fpkms)
#setkey(data.dt, ENSG_ID)
#temp &lt;- data.dt[, lapply(.SD, sum), by=ENSG_ID]
#collapsed &lt;- as.data.frame(temp)
#atlas.fpkms.summed &lt;- collapsed[,2:ncol(collapsed)] 
#rownames(atlas.fpkms.summed) &lt;- collapsed[,1]

#atlas.fpkms.summed &lt;- atlas.fpkms.summed[2:nrow(atlas.fpkms.summed),]
</code></pre>

<p>Finally, combine all the data sets into a data frame.</p>

<pre><code class="r">#fpkms &lt;- merge(hpa.fpkms, altiso.fpkms, by=&quot;ENSG_ID&quot;)
#fpkms &lt;- merge(fpkms, gtex.fpkms, by=&quot;ENSG_ID&quot;)
#fpkms &lt;- merge(fpkms, atlas.fpkms.summed, by.x=&quot;ENSG_ID&quot;, by.y=0)
#gene_id &lt;- fpkms[,1]
#f &lt;- fpkms[,2:ncol(fpkms)]
#rownames(f) &lt;- gene_id
</code></pre>

<p>Check how many ENSG IDs we have left.</p>

<pre><code class="r">#dim(f)
</code></pre>

<p>Approach 2: Try to map Entrez symbols to ENSEMBL to recover more ENSG IDs than already present in the table. </p>

<pre><code class="r">#m &lt;- org.Hs.egENSEMBL
#mapped_genes &lt;- mappedkeys(m)
#ensg.for.entrez &lt;- as.list(m[mapped_genes])
#remapped.ensg &lt;- ensg.for.entrez[as.character(atlas$entrez_gene_id)]

#atlas.fpkms$remapped_ensg &lt;- as.character(remapped.ensg)

# And add expression values
#data.dt &lt;- data.table(atlas.fpkms[,2:ncol(atlas.fpkms)])
#setkey(data.dt, remapped_ensg)
#temp &lt;- data.dt[, lapply(.SD, sum), by=remapped_ensg]
#collapsed &lt;- as.data.frame(temp)
#atlas.fpkms.summed &lt;- collapsed[,2:ncol(collapsed)] 
#rownames(atlas.fpkms.summed) &lt;- collapsed[,1]
</code></pre>

<p>Combine data sets again</p>

<pre><code class="r">#fpkms &lt;- merge(hpa.fpkms, altiso.fpkms, by=&quot;ENSG_ID&quot;)
#fpkms &lt;- merge(fpkms, gtex.fpkms, by=&quot;ENSG_ID&quot;)
#fpkms &lt;- merge(fpkms, atlas.fpkms.summed, by.x=&quot;ENSG_ID&quot;, by.y=0)
#gene_id &lt;- fpkms[,1]
#f &lt;- fpkms[,2:ncol(fpkms)]
#rownames(f) &lt;- gene_id
#write.table(f, file = &#39;published_rpkms.txt&#39;, quote=F)
</code></pre>

<p>Check how many ENSG IDs we have left.</p>

<pre><code class="r">f &lt;- read.delim(&quot;published_rpkms.txt&quot;,sep=&quot; &quot;)
</code></pre>

<pre><code>## Warning: cannot open file &#39;published_rpkms.txt&#39;: No such file or directory
</code></pre>

<pre><code>## Error: cannot open the connection
</code></pre>

<pre><code class="r">#dim(f)
</code></pre>

<p>This looks much better. 
Let&#39;s remove all lines where FPKM=0 in all samples before we proceed with this version of the data set:</p>

<pre><code class="r">f.nozero &lt;- f[-which(rowSums(f[,])==0),]
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<p>Start by a few correlation heat maps:</p>

<p><strong>Figure 1B</strong></p>

<p>Heatmap of Spearman correlations between published expression profiles (# genes = 13,333)</p>

<pre><code class="r">pheatmap(cor(f.nozero, method=&quot;spearman&quot;)) 
</code></pre>

<pre><code>## Error: object &#39;f.nozero&#39; not found
</code></pre>

<p>The brain samples are in a separate cluster, whereas the heart and kidney ones are intermixed.</p>

<p>Alternatively, one could use Pearson correlation (not shown in paper):</p>

<pre><code class="r">pheatmap(cor(f.nozero))
</code></pre>

<pre><code>## Error: object &#39;f.nozero&#39; not found
</code></pre>

<p>Sometimes the linear (Pearson) correlation works better on log values.  (not shown in paper):</p>

<pre><code class="r">pseudo &lt;- 1
fpkms.log &lt;- log2(f.nozero + pseudo)
</code></pre>

<pre><code>## Error: object &#39;f.nozero&#39; not found
</code></pre>

<pre><code class="r">pheatmap(cor(fpkms.log))
</code></pre>

<pre><code>## Error: object &#39;fpkms.log&#39; not found
</code></pre>

<pre><code class="r">write.table(fpkms.log, file=&quot;published_rpkms_log2.txt&quot;, quote=F)
</code></pre>

<pre><code>## Error: object &#39;fpkms.log&#39; not found
</code></pre>

<p>What if we drop the genes that have less than FPKM 1 on average? (not shown in paper):</p>

<pre><code class="r">f.nolow &lt;- f.nozero[-which(rowMeans(f.nozero)&lt;1),]
</code></pre>

<pre><code>## Error: object &#39;f.nozero&#39; not found
</code></pre>

<pre><code class="r">#pheatmap(cor(log2(f.nolow+pseudo)))
fpkms.log.nolow &lt;- log2(f.nolow + pseudo)
</code></pre>

<pre><code>## Error: object &#39;f.nolow&#39; not found
</code></pre>

<pre><code class="r">pheatmap(cor(fpkms.log.nolow))
</code></pre>

<pre><code>## Error: object &#39;fpkms.log.nolow&#39; not found
</code></pre>

<p>Try Anova on a &ldquo;melted&rdquo; expression matrix with some metadata:</p>

<pre><code class="r">library(reshape)
m &lt;- melt(f.nolow)
</code></pre>

<pre><code>## Error: object &#39;f.nolow&#39; not found
</code></pre>

<pre><code class="r">colnames(m) &lt;- c(&quot;sample_ID&quot;,&quot;RPKM&quot;)
</code></pre>

<pre><code>## Error: object &#39;m&#39; not found
</code></pre>

<pre><code class="r">meta &lt;- data.frame(tissue=c(&quot;heart&quot;,&quot;brain&quot;,&quot;kidney&quot;,&quot;heart&quot;,&quot;brain&quot;,&quot;heart&quot;,&quot;brain&quot;,&quot;kidney&quot;,&quot;heart&quot;,&quot;brain&quot;,&quot;kidney&quot;),study=c(&quot;HPA&quot;,&quot;HPA&quot;,&quot;HPA&quot;,&quot;AltIso&quot;,&quot;AltIso&quot;,&quot;GTex&quot;,&quot;GTex&quot;,&quot;GTex&quot;,&quot;Atlas&quot;,&quot;Atlas&quot;,&quot;Atlas&quot;),prep=c(rep(&quot;poly-A&quot;,8),rep(&quot;rRNA-depl&quot;,3)),layout=c(rep(&quot;PE&quot;,3),rep(&quot;SE&quot;,2),rep(&quot;PE&quot;,3),rep(&quot;SE&quot;,3)))
rownames(meta) &lt;- colnames(f)
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">tissue &lt;- rep(meta$tissue, each=nrow(f))
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">study &lt;- rep(meta$study, each=nrow(f))
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">prep &lt;- rep(meta$prep, each=nrow(f))
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">layout &lt;- rep(meta$layout, each=nrow(f))
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">data &lt;- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)
</code></pre>

<pre><code>## Error: object &#39;m&#39; not found
</code></pre>

<pre><code class="r">#subset &lt;- data[sample(1:nrow(data), 1000),]
fit &lt;- lm(RPKM ~ prep + layout + study + tissue, data=data)
</code></pre>

<pre><code>## Error: &#39;data&#39; argument is of the wrong type
</code></pre>

<pre><code class="r">a &lt;- anova(fit)
</code></pre>

<pre><code>## Error: object &#39;fit&#39; not found
</code></pre>

<pre><code class="r">maxval = 3200
</code></pre>

<p><strong>Figure 1X</strong></p>

<pre><code class="r">barplot(a$&quot;F value&quot;[-5],names.arg=rownames(a)[-5],main=&quot;Anova F score, Raw RPKM&quot;,ylim=c(0,maxval))
</code></pre>

<pre><code>## Error: object &#39;a&#39; not found
</code></pre>

<p>Let&#39;s look at a few SVD plots. </p>

<p><strong>Figure 1C</strong></p>

<pre><code class="r">colors &lt;- c(1,2,3,1,2,1,2,3,1,2,3)
plotPC(f.nozero, 1, 2, &quot;Published FPKM values \n SVD \n n=13333&quot;, colors=colors)
</code></pre>

<pre><code>## Error: object &#39;f.nozero&#39; not found
</code></pre>

<p>The heart samples are clearly separating into their own group.</p>

<p><strong>Figure 1D</strong> (not included in the current manuscript version)</p>

<pre><code class="r">plotPC(f.nozero, 2, 3, &quot;Published FPKM values \n SVD \n n=13333&quot;, colors=colors)
</code></pre>

<pre><code>## Error: object &#39;f.nozero&#39; not found
</code></pre>

<p>We can plot all pairwise combinations of principal components 1 to 5. (not shown in paper)
Start with SVD on the &ldquo;raw&rdquo; F/RPKMs.</p>

<pre><code class="r">colors &lt;- c(1,2,3,1,2,1,2,3,1,2,3)

par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
        if (i&lt;j){ 
        plotPC(f.nozero,i,j,desc=&quot;&quot;,colors=colors)
        }
    }
}
</code></pre>

<pre><code>## Error: object &#39;f.nozero&#39; not found
</code></pre>

<p><strong>Code for figure 2</strong></p>

<p>Figure 2 deals with log-transformed F/RPKM values from published data sets. </p>

<p>First we remove all rows where FPKM is zero in all samples and then add a pseudo count 1</p>

<p>PCA on log2-FPKM values:</p>

<p><strong>Figure 2A</strong></p>

<pre><code class="r">pseudo &lt;- 1
fpkms.log &lt;- log2(f.nozero + pseudo)
</code></pre>

<pre><code>## Error: object &#39;f.nozero&#39; not found
</code></pre>

<pre><code class="r">plotPC(fpkms.log, 1, 2, desc=&quot;Published FPKM values, log2 \n SVD \n n=13333&quot;, colors=colors)
</code></pre>

<pre><code>## Error: object &#39;fpkms.log&#39; not found
</code></pre>

<p><strong>Figure 2B</strong></p>

<pre><code class="r">plotPC(fpkms.log, 2, 3, desc=&quot;Published FPKM values, log2 \n SVD \n n=13333&quot;, colors=colors)
</code></pre>

<pre><code>## Error: object &#39;fpkms.log&#39; not found
</code></pre>

<p>Alternatively, with PCA (prcomp()) the plot would have looked like this (all combinations of top 5 PCs):</p>

<pre><code class="r">colors &lt;- c(1,2,3,1,2,1,2,3,1,2,3)

p &lt;- prcomp(t(fpkms.log))
</code></pre>

<pre><code>## Error: object &#39;fpkms.log&#39; not found
</code></pre>

<pre><code class="r">par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
        if (i&lt;j){ 
        plot(p$x[,i],p$x[,j],pch=20,col=colors,xlab=paste(&quot;PC&quot;, i),ylab=paste(&quot;PC&quot;, j))
        }
    }
}
</code></pre>

<pre><code>## Error: object &#39;p&#39; not found
</code></pre>

<p><strong>Figure 2C</strong></p>

<p>Let&#39;s have a look at the 100 most highly expressed genes in each sample and see how many of these genes that are shared between the studies</p>

<pre><code class="r">library(VennDiagram)
</code></pre>

<pre><code>## Loading required package: grid
</code></pre>

<pre><code class="r">HPA_b &lt;- rownames(f[order(f$HPA_brain,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">HPA_h &lt;- rownames(f[order(f$HPA_heart,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">HPA_k &lt;- rownames(f[order(f$HPA_kidney,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">AltIso_b &lt;- rownames(f[order(f$AltIso_brain,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">AltIso_h &lt;- rownames(f[order(f$AltIso_heart,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">GTEx_b &lt;- rownames(f[order(f$GTEx_brain,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">GTEx_h &lt;- rownames(f[order(f$GTEx_heart,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">GTEx_k &lt;- rownames(f[order(f$GTEx_kidney,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">Atlas_b &lt;- rownames(f[order(f$Atlas_brain,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">Atlas_h &lt;- rownames(f[order(f$Atlas_heart,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<pre><code class="r">Atlas_k &lt;- rownames(f[order(f$Atlas_kidney,decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;f&#39; not found
</code></pre>

<p>Let&#39;s start with the four brain samples:</p>

<pre><code class="r">b_Ids &lt;- list(HPA_b,AltIso_b,GTEx_b,Atlas_b)
</code></pre>

<pre><code>## Error: object &#39;HPA_b&#39; not found
</code></pre>

<pre><code class="r">draw.quad.venn(100, 100, 100, 100, 
               length(intersect(HPA_b,AltIso_b)),
               length(intersect(HPA_b,GTEx_b)),
               length(intersect(HPA_b,Atlas_b)),
               length(intersect(AltIso_b,GTEx_b)),
               length(intersect(AltIso_b,Atlas_b)),
               length(intersect(GTEx_b,Atlas_b)),
               length(intersect(intersect(HPA_b,AltIso_b),GTEx_b)),
               length(intersect(intersect(HPA_b,AltIso_b),Atlas_b)),
               length(intersect(intersect(HPA_b,GTEx_b),Atlas_b)),
               length(intersect(intersect(AltIso_b,GTEx_b),Atlas_b)),
               length(intersect(intersect(HPA_b,AltIso_b),intersect(Atlas_b,GTEx_b))),
               category = c(&quot;HPA&quot;,&quot;AltIso&quot;,&quot;GTEx&quot;,&quot;Atlas&quot;), lwd = rep(0, 4), lty = rep(&quot;solid&quot;, 4),
               fill = c(&quot;mistyrose&quot;,&quot;steelblue&quot;,&quot;lightgoldenrod&quot;,&quot;darkseagreen&quot;)
)
</code></pre>

<pre><code>## Error: object &#39;GTEx_b&#39; not found
</code></pre>

<p>Let&#39;s investigate the heart samples:</p>

<pre><code class="r">h_Ids &lt;- list(HPA_h,AltIso_h,GTEx_h,Atlas_h)
</code></pre>

<pre><code>## Error: object &#39;HPA_h&#39; not found
</code></pre>

<pre><code class="r">draw.quad.venn(100, 100, 100, 100, 
               length(intersect(HPA_h,AltIso_h)),
               length(intersect(HPA_h,GTEx_h)),
               length(intersect(HPA_h,Atlas_h)),
               length(intersect(AltIso_h,GTEx_h)),
               length(intersect(AltIso_h,Atlas_h)),
               length(intersect(GTEx_h,Atlas_h)),
               length(intersect(intersect(HPA_h,AltIso_h),GTEx_h)),
               length(intersect(intersect(HPA_h,AltIso_h),Atlas_h)),
               length(intersect(intersect(HPA_h,GTEx_h),Atlas_h)),
               length(intersect(intersect(AltIso_h,GTEx_h),Atlas_h)),
               length(intersect(intersect(HPA_h,AltIso_h),intersect(Atlas_h,GTEx_h))),
               category = c(&quot;HPA&quot;,&quot;AltIso&quot;,&quot;GTEx&quot;,&quot;Atlas&quot;), lwd = rep(0, 4), lty = rep(&quot;solid&quot;, 4),
               fill = c(&quot;mistyrose&quot;,&quot;steelblue&quot;,&quot;lightgoldenrod&quot;,&quot;darkseagreen&quot;)
)
</code></pre>

<pre><code>## Error: object &#39;GTEx_h&#39; not found
</code></pre>

<p>&hellip;and the three kidney samples:</p>

<pre><code class="r">k_Ids &lt;- list(HPA_k,GTEx_k,Atlas_k)
</code></pre>

<pre><code>## Error: object &#39;HPA_k&#39; not found
</code></pre>

<pre><code class="r">draw.triple.venn(100, 100, 100, 
                 length(intersect(HPA_k,GTEx_k)),
                 length(intersect(GTEx_k,Atlas_k)),
                 length(intersect(HPA_k,Atlas_k)),
                 length(intersect(intersect(HPA_k,GTEx_k),Atlas_k)),
                 category = c(&quot;HPA&quot;,&quot;GTEx&quot;,&quot;Atlas&quot;), lwd = rep(0, 3), lty = rep(&quot;solid&quot;, 3),
                 fill = c(&quot;steelblue&quot;,&quot;lightgoldenrod&quot;,&quot;darkseagreen&quot;)
)
</code></pre>

<pre><code>## Error: object &#39;GTEx_k&#39; not found
</code></pre>

<p><strong>Figure 2D</strong></p>

<pre><code class="r">m &lt;- melt(f.log)
</code></pre>

<pre><code>## Error: object &#39;f.log&#39; not found
</code></pre>

<pre><code class="r">colnames(m) &lt;- c(&quot;gene_ID&quot;,&quot;sample_ID&quot;,&quot;log2RPKM&quot;)
</code></pre>

<pre><code>## Error: object &#39;m&#39; not found
</code></pre>

<pre><code class="r">data &lt;- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)
</code></pre>

<pre><code>## Error: object &#39;m&#39; not found
</code></pre>

<pre><code class="r">#subset &lt;- data[sample(1:nrow(data), 1000),]
fit &lt;- lm(log2RPKM ~ + prep + layout + study + tissue, data=data)
</code></pre>

<pre><code>## Error: &#39;data&#39; argument is of the wrong type
</code></pre>

<pre><code class="r">b &lt;- anova(fit)
</code></pre>

<pre><code>## Error: object &#39;fit&#39; not found
</code></pre>

<pre><code class="r">barplot(b$&quot;F value&quot;[-5],names.arg=rownames(b)[-5],main=&quot;Anova F score, log2-RPKM&quot;,ylim=c(0,maxval))
</code></pre>

<pre><code>## Error: object &#39;b&#39; not found
</code></pre>

<pre><code class="r">print(b)
</code></pre>

<pre><code>## Error: object &#39;b&#39; not found
</code></pre>

<p>ANOVA for TMM scaled values. (no figure)</p>

<pre><code class="r">m &lt;- melt(f.tmm)
</code></pre>

<pre><code>## Error: object &#39;f.tmm&#39; not found
</code></pre>

<pre><code class="r">colnames(m) &lt;- c(&quot;gene_ID&quot;,&quot;sample_ID&quot;,&quot;TMM_RPKM&quot;)
</code></pre>

<pre><code>## Error: object &#39;m&#39; not found
</code></pre>

<pre><code class="r">data &lt;- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)
</code></pre>

<pre><code>## Error: object &#39;m&#39; not found
</code></pre>

<pre><code class="r">#subset &lt;- data[sample(1:nrow(data), 1000),]
fit &lt;- lm(TMM_RPKM ~ + prep + layout + study + tissue, data=data)
</code></pre>

<pre><code>## Error: &#39;data&#39; argument is of the wrong type
</code></pre>

<pre><code class="r">a.tmm &lt;- anova(fit)
</code></pre>

<pre><code>## Error: object &#39;fit&#39; not found
</code></pre>

<pre><code class="r">barplot(a.tmm$&quot;F value&quot;[-5],names.arg=rownames(a.tmm)[-5],main=&quot;Anova F score, TMM&quot;,ylim=c(0,maxval))
</code></pre>

<pre><code>## Error: object &#39;a.tmm&#39; not found
</code></pre>

<pre><code class="r">print(a.tmm)
</code></pre>

<pre><code>## Error: object &#39;a.tmm&#39; not found
</code></pre>

<p><strong>Code for figure 3</strong></p>

<p>Figure 3 is about ComBat.</p>

<p>Combat analysis is performed on logged values (n=13333)</p>

<pre><code class="r">library(sva)
</code></pre>

<pre><code>## Loading required package: corpcor
## Loading required package: mgcv
## Loading required package: nlme
## This is mgcv 1.7-29. For overview type &#39;help(&quot;mgcv-package&quot;)&#39;.
</code></pre>

<pre><code class="r">meta &lt;- data.frame(study=c(rep(&quot;HPA&quot;,3),rep(&quot;AltIso&quot;,2),rep(&quot;GTex&quot;,3),rep(&quot;Atlas&quot;,3)),tissue=c(&quot;Heart&quot;,&quot;Brain&quot;,&quot;Kidney&quot;,&quot;Heart&quot;,&quot;Brain&quot;,&quot;Heart&quot;,&quot;Brain&quot;,&quot;Kidney&quot;,&quot;Heart&quot;,&quot;Brain&quot;,&quot;Kidney&quot;))

batch &lt;- meta$study
design &lt;- model.matrix(~as.factor(tissue),data=meta)

combat &lt;- ComBat(dat=fpkms.log,batch=batch,mod=design,numCovs=NULL,par.prior=TRUE)
</code></pre>

<pre><code>## Found 4 batches
## Found 2  categorical covariate(s)
</code></pre>

<pre><code>## Error: object &#39;fpkms.log&#39; not found
</code></pre>

<pre><code class="r">write.table(combat, file=&quot;published_rpkms_combat_log2.txt&quot;, quote=F)
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">pheatmap(cor(combat))
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">plotPC(combat,1,2,colors=colors,desc=&quot;Published F/RPKM values, ComBat adjusted log2 TMM values \n SVD \n n=13333&quot;)
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">plotPC(combat,2,3,colors=colors,desc=&quot;Published F/RPKM values, ComBat adjusted log2 TMM values \n SVD \n n=13333&quot;)
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
        if (i&lt;j){ 
      eig.cell &lt;- s$u[,c(i, j)]
      proj &lt;- t(combat) %*% eig.cell
          plot(proj[,1],proj[,2],pch=20,col=colors,xlab=paste(&quot;PC&quot;, i),ylab=paste(&quot;PC&quot;, j))
        }
    }
}
</code></pre>

<pre><code>## Error: object of type &#39;closure&#39; is not subsettable
</code></pre>

<p>Let&#39;s have a look again at the 100 most highly expressed genes in each sample and see how many of these genes that are shared between the studies, but this time looking at the values after the ComBat run:</p>

<pre><code class="r">HPA_b_c &lt;- rownames(combat[order(combat[,1],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">HPA_h_c &lt;- rownames(combat[order(combat[,2],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">HPA_k_c &lt;- rownames(combat[order(combat[,3],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">AltIso_b_c &lt;- rownames(combat[order(combat[,4],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">AltIso_h_c &lt;- rownames(combat[order(combat[,5],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">GTEx_b_c &lt;- rownames(combat[order(combat[,6],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">GTEx_h_c &lt;- rownames(combat[order(combat[,7],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">GTEx_k_c &lt;- rownames(combat[order(combat[,8],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">Atlas_b_c &lt;- rownames(combat[order(combat[,9],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">Atlas_h_c &lt;- rownames(combat[order(combat[,10],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">Atlas_k_c &lt;- rownames(combat[order(combat[,11],decreasing=T),][1:100,])
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<p>Brain:</p>

<pre><code class="r">library(gridExtra)
r &lt;- rectGrob(gp=gpar(fill=&quot;white&quot;,lwd=0))
top &lt;- arrangeGrob(r, r, r, r, nrow=2)
grid.arrange(top, r, ncol=1, main = &quot;\nBrain Combat&quot;, heights=c(2, 1))
</code></pre>

<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAfgAAAH4CAYAAACmKP9/AAAEJGlDQ1BJQ0MgUHJvZmlsZQAAOBGFVd9v21QUPolvUqQWPyBYR4eKxa9VU1u5GxqtxgZJk6XtShal6dgqJOQ6N4mpGwfb6baqT3uBNwb8AUDZAw9IPCENBmJ72fbAtElThyqqSUh76MQPISbtBVXhu3ZiJ1PEXPX6yznfOec7517bRD1fabWaGVWIlquunc8klZOnFpSeTYrSs9RLA9Sr6U4tkcvNEi7BFffO6+EdigjL7ZHu/k72I796i9zRiSJPwG4VHX0Z+AxRzNRrtksUvwf7+Gm3BtzzHPDTNgQCqwKXfZwSeNHHJz1OIT8JjtAq6xWtCLwGPLzYZi+3YV8DGMiT4VVuG7oiZpGzrZJhcs/hL49xtzH/Dy6bdfTsXYNY+5yluWO4D4neK/ZUvok/17X0HPBLsF+vuUlhfwX4j/rSfAJ4H1H0qZJ9dN7nR19frRTeBt4Fe9FwpwtN+2p1MXscGLHR9SXrmMgjONd1ZxKzpBeA71b4tNhj6JGoyFNp4GHgwUp9qplfmnFW5oTdy7NamcwCI49kv6fN5IAHgD+0rbyoBc3SOjczohbyS1drbq6pQdqumllRC/0ymTtej8gpbbuVwpQfyw66dqEZyxZKxtHpJn+tZnpnEdrYBbueF9qQn93S7HQGGHnYP7w6L+YGHNtd1FJitqPAR+hERCNOFi1i1alKO6RQnjKUxL1GNjwlMsiEhcPLYTEiT9ISbN15OY/jx4SMshe9LaJRpTvHr3C/ybFYP1PZAfwfYrPsMBtnE6SwN9ib7AhLwTrBDgUKcm06FSrTfSj187xPdVQWOk5Q8vxAfSiIUc7Z7xr6zY/+hpqwSyv0I0/QMTRb7RMgBxNodTfSPqdraz/sDjzKBrv4zu2+a2t0/HHzjd2Lbcc2sG7GtsL42K+xLfxtUgI7YHqKlqHK8HbCCXgjHT1cAdMlDetv4FnQ2lLasaOl6vmB0CMmwT/IPszSueHQqv6i/qluqF+oF9TfO2qEGTumJH0qfSv9KH0nfS/9TIp0Wboi/SRdlb6RLgU5u++9nyXYe69fYRPdil1o1WufNSdTTsp75BfllPy8/LI8G7AUuV8ek6fkvfDsCfbNDP0dvRh0CrNqTbV7LfEEGDQPJQadBtfGVMWEq3QWWdufk6ZSNsjG2PQjp3ZcnOWWing6noonSInvi0/Ex+IzAreevPhe+CawpgP1/pMTMDo64G0sTCXIM+KdOnFWRfQKdJvQzV1+Bt8OokmrdtY2yhVX2a+qrykJfMq4Ml3VR4cVzTQVz+UoNne4vcKLoyS+gyKO6EHe+75Fdt0Mbe5bRIf/wjvrVmhbqBN97RD1vxrahvBOfOYzoosH9bq94uejSOQGkVM6sN/7HelL4t10t9F4gPdVzydEOx83Gv+uNxo7XyL/FtFl8z9ZAHF4bBsrEwAAHadJREFUeAHt22uobWXZBuBX81Sah8yyTMwTZQpiZAgWCCqKP8QD/VHxAP0IFcTISiiJkjwgSGCKiJmlgqAY6I8iECTCzDDLFDpoWlmpmVpqpqlf74C12O72tvzcBt33NWG311pzrTHXfT2P3XOOMfdGL//zNtwIECBAgACBKIGNo9IIQ4AAAQIECCwCCt4iECBAgACBQAEFHzhUkQgQIECAgIK3AwQIECBAIFBAwQcOVSQCBAgQIKDg7QABAgQIEAgUUPCBQxWJAAECBAgoeDtAgAABAgQCBRR84FBFIkCAAAECCt4OECBAgACBQAEFHzhUkQgQIECAgIK3AwQIECBAIFBAwQcOVSQCBAgQIKDg7QABAgQIEAgUUPCBQxWJAAECBAgoeDtAoEDgb3/723j++ecLkopIgMCKgIJfkfA3gQ0s8Jvf/GZstNFGY+utt17+bL/99uOQQw4ZP//5z1/zI33mM58Zn/vc517zz33nO98ZBx544PjABz4w3v/+94+jjz563Hvvva/5OOv7gR/+8IfLsdd3//q+Pn+vW2+9dX13+zoBAhtAQMFvAESHIPBqAk888cT4y1/+Mh599NGl4E888cRX+/Z13nf22WePs846a533re+L3/rWt8Z8rEsuuWT8+te/Hvfff/847LDDxrHHHjuefvrp9f3Yf+Xrl19++Xj44Yf/K4/lQQi0Cij41snL/V8XeNOb3jQ++MEPrp4q/8lPfjJOPvnkccQRRyyvrp955plx9dVXj7322mtstdVWy/feeeedy+955ZVXjquuumr5+OCDDx5f//rXxx577DHe9a53jUsvvXSdWWaJfvGLXxz77bffcv88m/CJT3xinH/++eMf//jH8rXbbrttHHPMMeNtb3vbOOqoo8Yf//jH5ev33HPPOOmkk8app5465pmHww8/fDzwwAPjoIMOGjvvvPO4+OKLVx9znvo/5ZRTxnbbbTcOPfTQ1eJ+9tlnl8fbaaedluN/7GMfG3/961/H1772tfHd7353zLMS11xzzepxfECAwIYVUPAb1tPRCPyLwIUXXjguuOCC8dnPfnYpzPPOO2/5nnldfBbchz/84XHRRReN3//+9+O0004b11133fjtb387PvShD62elp+v/v/0pz8tPzdfic/vueWWW8YVV1wxzjzzzPHUU0/9y+P+6Ec/Wp4krH3HLPJtt912eVV/5JFHjvnnpz/96Xjzm9+8lPr8/ueee25885vfHO9973vHXXfdNR555JFxwAEHjM9//vPj5ptvHueee+74+9//vhx6/j7zScA89f/ud797fPzjH1++/pWvfGU5a/DjH/943H777ctjXH/99eP4449fniicc845Y5a+GwECb4zAJm/MYR2VAIEVgbvvvnu5Fv/kk08up8bnK/f5injetthii/GFL3xh+Xiexr/jjjvG3nvvvRT2vG7+gx/8YLlv7f+Zp+vnNfX5Z5tttlmeHMy/V27zVfW8NDCv/6/vNk/h77PPPstZhPk9X/rSl8aee+65XEqYn8/C//SnPz0/HPOswS9+8Yvl7/n529/+9qWw58ebbrrpkmGedZhPYubvPS8BHHfcccux3/GOd4z5ZGYee54h2HzzzZef2XLLLZeP5zHcCBDY8AIKfsObOiKBVwjMV9vz9Py8/e53vxu77bbb6ivX97znPavf+9a3vnXMV7jzzyz7eQr+pZdeWr1/zQ/e+c53rn46i/KFF15Y/Xx+sNlmmy2F+qtf/Wq8733ve8V982u77LLLeOihh5ZX5St3zsebr8TnmYR5m6/GV25vectblicTK5/P47/44otj4403Xp6QzHKft3l5Yf4+8xjzZ84444wxLwPssMMOy++4//77rxzC3wQIvMECTtG/wcAOT2BNgVnoszhX3sm+Uvzze77xjW+MG264Ydx4443jD3/4w/KmupdffnnNH1/9eF5P/3e3ee3929/+9iu+bZ7Kn6f+Z+nOV+H33Xff6v3zMf/85z+PXXfddfnamr/b6jet44OVJwTzrgcffHDM9xLsuOOOy/X3eW1/Xs//2c9+tlyKWF+edRzWlwgQeJ0CCv51AvpxAv9OYJbm/DOvY1922WXLKfB1vZKd3zNPY8/T5rMI5xvp1n5l/u8ea8375yn3eR19HmdeL5/vWp+vqOc1//nP9eY76r/3ve8tTzbmmYJ5PX9eHljzVP+ax1vfx/P9Ad///veX33m+EfAjH/nIcmng8ccfXz6ep+jnmYv5xrqVPPNV/rxk4UaAwBsn4BT9G2fryAQWgVlw8zavac/T4Ndee+3yCne+2l3zNq9Zzzfd7bvvvksRnnDCCUspznej/39uu++++1LwX/7yl5c34s1X/fMd+1/96leXw80nGfONbrPw56n5Wew33XTTa36oec39U5/61PIEZpNNNlk9xnyfwCc/+cnln+nNJyzz3+D/8pe/XI7/0Y9+dJx++ulLyc837rkRILDhBTb653946z4HuOEfyxEJEPgPBOYr3/lPzub17Q11m296m2+Gm29wW/s2/8ncPHU/S/713B577LHlWvuax5j/9zLzzMsBa9/mO/Xn7/SfXgpY++d9ToDAqwso+Ff3cS8BAgQIEPifFNhwLxH+J+P7pQkQIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBcQMGXL4D4BAgQIJApoOAz5yoVAQIECJQLKPjyBRCfAAECBDIFFHzmXKUiQIAAgXIBBV++AOITIECAQKaAgs+cq1QECBAgUC6g4MsXQHwCBAgQyBRQ8JlzlYoAAQIEygUUfPkCiE+AAAECmQIKPnOuUhEgQIBAuYCCL18A8QkQIEAgU0DBZ85VKgIECBAoF1Dw5QsgPgECBAhkCij4zLlKRYAAAQLlAgq+fAHEJ0CAAIFMAQWfOVepCBAgQKBc4P8A4oFAea3o/PQAAAAASUVORK5CYII=" alt="plot of chunk :combatVennBrain"/> </p>

<pre><code class="r">draw.quad.venn(100, 100, 100, 100, 
               length(intersect(HPA_b_c,AltIso_b_c)),
               length(intersect(HPA_b_c,GTEx_b_c)),
               length(intersect(HPA_b_c,Atlas_b_c)),
               length(intersect(AltIso_b_c,GTEx_b_c)),
               length(intersect(AltIso_b_c,Atlas_b_c)),
               length(intersect(GTEx_b_c,Atlas_b_c)),
               length(intersect(intersect(HPA_b_c,AltIso_b_c),GTEx_b_c)),
               length(intersect(intersect(HPA_b_c,AltIso_b_c),Atlas_b_c)),
               length(intersect(intersect(HPA_b_c,GTEx_b_c),Atlas_b_c)),
               length(intersect(intersect(AltIso_b_c,GTEx_b_c),Atlas_b_c)),
               length(intersect(intersect(HPA_b_c,AltIso_b_c),intersect(Atlas_b_c,GTEx_b_c))),
               category = c(&quot;HPA&quot;,&quot;AltIso&quot;,&quot;GTEx&quot;,&quot;Atlas&quot;), lwd = rep(0, 4), lty = rep(&quot;solid&quot;, 4),
               fill = c(&quot;mistyrose&quot;,&quot;steelblue&quot;,&quot;lightgoldenrod&quot;,&quot;darkseagreen&quot;)
)
</code></pre>

<pre><code>## Error: object &#39;GTEx_b_c&#39; not found
</code></pre>

<p>Heart:</p>

<pre><code class="r">r &lt;- rectGrob(gp=gpar(fill=&quot;white&quot;,lwd=0))
top &lt;- arrangeGrob(r, r, r, r, nrow=2)
grid.arrange(top, r, ncol=1, main = &quot;\nHeart Combat&quot;, heights=c(2, 1))
</code></pre>

<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAfgAAAH4CAYAAACmKP9/AAAEJGlDQ1BJQ0MgUHJvZmlsZQAAOBGFVd9v21QUPolvUqQWPyBYR4eKxa9VU1u5GxqtxgZJk6XtShal6dgqJOQ6N4mpGwfb6baqT3uBNwb8AUDZAw9IPCENBmJ72fbAtElThyqqSUh76MQPISbtBVXhu3ZiJ1PEXPX6yznfOec7517bRD1fabWaGVWIlquunc8klZOnFpSeTYrSs9RLA9Sr6U4tkcvNEi7BFffO6+EdigjL7ZHu/k72I796i9zRiSJPwG4VHX0Z+AxRzNRrtksUvwf7+Gm3BtzzHPDTNgQCqwKXfZwSeNHHJz1OIT8JjtAq6xWtCLwGPLzYZi+3YV8DGMiT4VVuG7oiZpGzrZJhcs/hL49xtzH/Dy6bdfTsXYNY+5yluWO4D4neK/ZUvok/17X0HPBLsF+vuUlhfwX4j/rSfAJ4H1H0qZJ9dN7nR19frRTeBt4Fe9FwpwtN+2p1MXscGLHR9SXrmMgjONd1ZxKzpBeA71b4tNhj6JGoyFNp4GHgwUp9qplfmnFW5oTdy7NamcwCI49kv6fN5IAHgD+0rbyoBc3SOjczohbyS1drbq6pQdqumllRC/0ymTtej8gpbbuVwpQfyw66dqEZyxZKxtHpJn+tZnpnEdrYBbueF9qQn93S7HQGGHnYP7w6L+YGHNtd1FJitqPAR+hERCNOFi1i1alKO6RQnjKUxL1GNjwlMsiEhcPLYTEiT9ISbN15OY/jx4SMshe9LaJRpTvHr3C/ybFYP1PZAfwfYrPsMBtnE6SwN9ib7AhLwTrBDgUKcm06FSrTfSj187xPdVQWOk5Q8vxAfSiIUc7Z7xr6zY/+hpqwSyv0I0/QMTRb7RMgBxNodTfSPqdraz/sDjzKBrv4zu2+a2t0/HHzjd2Lbcc2sG7GtsL42K+xLfxtUgI7YHqKlqHK8HbCCXgjHT1cAdMlDetv4FnQ2lLasaOl6vmB0CMmwT/IPszSueHQqv6i/qluqF+oF9TfO2qEGTumJH0qfSv9KH0nfS/9TIp0Wboi/SRdlb6RLgU5u++9nyXYe69fYRPdil1o1WufNSdTTsp75BfllPy8/LI8G7AUuV8ek6fkvfDsCfbNDP0dvRh0CrNqTbV7LfEEGDQPJQadBtfGVMWEq3QWWdufk6ZSNsjG2PQjp3ZcnOWWing6noonSInvi0/Ex+IzAreevPhe+CawpgP1/pMTMDo64G0sTCXIM+KdOnFWRfQKdJvQzV1+Bt8OokmrdtY2yhVX2a+qrykJfMq4Ml3VR4cVzTQVz+UoNne4vcKLoyS+gyKO6EHe+75Fdt0Mbe5bRIf/wjvrVmhbqBN97RD1vxrahvBOfOYzoosH9bq94uejSOQGkVM6sN/7HelL4t10t9F4gPdVzydEOx83Gv+uNxo7XyL/FtFl8z9ZAHF4bBsrEwAAHxtJREFUeAHt22morXXZBvC/ZpTabIpYYEkTCQ2ElRM0FzbQQAOiZpAUEYY5gZGWUpEFQhplEYFpfVATCQIJhQYbPkSJGpZ9UczZckiNSp93/Rec/Xrs8Gp22fWy/S04evZea1/7Xr97H65nPevZ2y2r23AjQIAAAQIENpXA9pvq2XgyBAgQIECAwFpAwftBIECAAAECm1BAwW/CpXpKBAgQIEBAwfsZIECAAAECm1BAwW/CpXpKBAgQIEBAwfsZIECAAAECm1BAwW/CpXpKBAgQIEBAwfsZIECAAAECm1BAwW/CpXpKBAgQIEBAwfsZIECAAAECm1BAwW/CpXpKBAgQIEBAwfsZIECAAAECm1BAwW/CpXpKBAgQIEBAwfsZIECAAAECm1BAwW/CpXpKBAgQIEBAwfsZIPAYEbjjjjseI8/U0yRAYAooeD8HBB4lgbe97W3jK1/5ylbp119//dhuu+3G3/72t60+/598cNFFF41LLrlkmxHLsozPfOYz40UvetF45StfOV74wheOT3ziE+Pvf//7Nh//SD55zDHHjJNOOunf/tL5Nck5/u0BfAGBTS6g4Df5gj29zS9w5plnjj/96U/bfKIf+tCHxi9+8Yvx85//fPz+978fv/3tb8cf//jHcdxxx23z8f+tT953333j5JNPHvfff/9/61v6PgQecwIK/jG3ck/4/5PAj3/84/HSl750PO1pTxvvfve7x6233roe75577hkf/ehHx7Oe9azxjGc8Y7z3ve8dd9111/q+L37xi+OUU04Zz372s8cb3vCG8aMf/Wgcf/zx4+yzz97qqd15553rz51zzjnrjHnnjjvuOL71rW+Nt7zlLevHzoL96le/Ol7+8pevv9dnP/vZjdL90pe+NOafAw44YOy2227jC1/4wvjBD34w9tprr/GqV71qfeCw5Rv+4Q9/GK94xSvGrrvuOo499tgxC3zefve7343Xvva146lPferYc889x2mnnbb+/Ac+8IH1/+dz3/Kc15/wHwIEcgKrU3huBAg8CgJvfetbl9e85jXLUUcdtfHniCOOWFb/epd77713ufnmm5cnP/nJy3e+851ldep+Ofzww5dPfvKT60k+//nPL6vyXm666ablqquuWl7wghcs3/zmN9f3zcfssssuywUXXLD89Kc/XVZvBSxf+9rXltVp/62exeq0/fKc5zxnq889+IPTTz992XvvvZdf/epXy6WXXro8//nP3/g+q6Jenv70py+//OUvl5/85CfL9ttvv7zxjW9crr766uVzn/vc8v73v38dd/TRRy9PetKTlgsvvHC57LLLltXbAMvqIGJ93+rAYTn11FOXv/71r8v555+/PO5xj1tuu+225S9/+cva4YYbblhWBxkPHsvHBAgEBLyCzx0rSSLwLwKrf6PrV8TzlfKWP1se9P3vf3+synW84x3vGDvvvPP41Kc+NX74wx+u7z744IPHWWedtX7l/MQnPnGsinfceOONW750vP3tbx/vfOc716+uH//4x6+//glPeMLG/fMvqwOI8ZSnPGWrzz34g+9+97vjwx/+8Pr9+f3222+sDjLG6oBj42Fztvlq/cADD1yfMTjssMPG8573vLE6eBmrg4KNx73uda9bP4+XvOQlY85+3nnnre/7xje+MVYHJGPOtjrYWJ9BuOWWW8bqwGZ9/zxzMa9JcCNAIC+wQz5SIgECWwTmafcjjzxyy4djXmQ3T5HP23XXXTcuv/zy9YVvGw9Y/WW+n756pbu+GG6ewp+nvf/xj3+MffbZZ+Nh8/T8Q93mwcN8v30eZDywROfH8/34eeHdNddcM/bdd9+NqPn3b3/72xsf77HHHht/n6f350V68zYL+5///OfGfa9+9as3/j5P1c+Dl3mbZT4PDlZnIcbLXvay9an7eaDjRoDAoy/gFfyjb+w7ENimwLyqfb5qXp2m3vjz61//esxSne+/z/fe5wHAFVdcsX6FPYt5y20eADzUbZbx/Jqf/exnWz10vrp+/etfP1ZvE4xnPvOZ48orr9y4f36/+R77ltvD+T7zsQ+8yG++77777ruPP//5z+M973nPWJ3CXx/YXHzxxet5Hvg8tnwf/ydAIC+g4POmEgk8LIF5gdw8zf2b3/xm/fh5kdy8+G2+wl29T71xcdt8pT8vpJuv4rd1m6f3b7/99n+5a566//KXvzwOPfTQsXpvfH3/vKL+xBNPHPNiuvmK/M1vfvP43ve+N+bvyM9CPvfcc8f+++//L1kP9Yk53+p99vUcq/fax0EHHbT+eH7dfJ7zbYb5feavB87nMQ8c5lkAv5v/ULLuJ/DIBZyif+R2vpLAfyQwC3Z1Md36FPa8Wn5+/PWvf31dfvNK9Pne9RlnnLF+1fuud71rrC5u2+b3m6fAP/7xj6/L9dOf/vRWj/nYxz42rr322vUr6Xm6fJ7uf9/73jfmr8/N2wknnDAOOeSQ8dznPnfssMMO401vetP6c1uFPIwP5un+eQp+HmjMswMf+chH1gX+wQ9+cP1bAquLAseLX/ziMU/lzyvu59Xz8+r6+VbD/NW9+XaCGwECWYHtVqfL/ve8XzZbGgECD0Ng/krZLMZZgg+8zX+a85X8PI3+ULf5yni+Yv+/TqnPV+jztP+2bvNX6ubXz4OMR3qbr8zvvvvu9a/8PTBjfm5eA7DTTjs98NPrv8/75hkINwIE8gIKPm8qkQABAgQI1AW8B19fgQEIECBAgEBeQMHnTSUSIECAAIG6gIKvr8AABAgQIEAgL6Dg86YSCRAgQIBAXUDB11dgAAIECBAgkBdQ8HlTiQQIECBAoC6g4OsrMAABAgQIEMgLKPi8qUQCBAgQIFAXUPD1FRiAAAECBAjkBRR83lQiAQIECBCoCyj4+goMQIAAAQIE8gIKPm8qkQABAgQI1AUUfH0FBiBAgAABAnkBBZ83lUiAAAECBOoCCr6+AgMQIECAAIG8gILPm0okQIAAAQJ1AQVfX4EBCBAgQIBAXkDB500lEiBAgACBuoCCr6/AAAQIECBAIC+g4POmEgkQIECAQF1AwddXYAACBAgQIJAXUPB5U4kECBAgQKAuoODrKzAAAQIECBDICyj4vKlEAgQIECBQF1Dw9RUYgAABAgQI5AUUfN5UIgECBAgQqAso+PoKDECAAAECBPICCj5vKpEAAQIECNQFFHx9BQYgQIAAAQJ5AQWfN5VIgAABAgTqAgq+vgIDECBAgACBvICCz5tKJECAAAECdQEFX1+BAQgQIECAQF5AwedNJRIgQIAAgbqAgq+vwAAECBAgQCAvoODzphIJECBAgEBdQMHXV2AAAgQIECCQF1DweVOJBAgQIECgLqDg6yswAAECBAgQyAso+LypRAIECBAgUBdQ8PUVGIAAAQIECOQFFHzeVCIBAgQIEKgLKPj6CgxAgAABAgTyAgo+byqRAAECBAjUBRR8fQUGIECAAAECeQEFnzeVSIAAAQIE6gIKvr4CAxAgQIAAgbyAgs+bSiRAgAABAnUBBV9fgQEIECBAgEBeQMHnTSUSIECAAIG6gIKvr8AABAgQIEAgL6Dg86YSCRAgQIBAXUDB11dgAAIECBAgkBdQ8HlTiQQIECBAoC6g4OsrMAABAgQIEMgLKPi8qUQCBAgQIFAXUPD1FRiAAAECBAjkBRR83lQiAQIECBCoCyj4+goMQIAAAQIE8gIKPm8qkQABAgQI1AUUfH0FBiBAgAABAnkBBZ83lUiAAAECBOoCCr6+AgMQIECAAIG8gILPm0okQIAAAQJ1AQVfX4EBCBAgQIBAXkDB500lEiBAgACBuoCCr6/AAAQIECBAIC+g4POmEgkQIECAQF1AwddXYAACBAgQIJAXUPB5U4kECBAgQKAuoODrKzAAAQIECBDICyj4vKlEAgQIECBQF1Dw9RUYgAABAgQI5AUUfN5UIgECBAgQqAso+PoKDECAAAECBPICCj5vKpEAAQIECNQFFHx9BQYgQIAAAQJ5AQWfN5VIgAABAgTqAgq+vgIDECBAgACBvICCz5tKJECAAAECdQEFX1+BAQgQIECAQF5AwedNJRIgQIAAgbqAgq+vwAAECBAgQCAvoODzphIJECBAgEBdQMHXV2AAAgQIECCQF1DweVOJBAgQIECgLqDg6yswAAECBAgQyAso+LypRAIECBAgUBdQ8PUVGIAAAQIECOQFFHzeVCIBAgQIEKgLKPj6CgxAgAABAgTyAgo+byqRAAECBAjUBRR8fQUGIECAAAECeQEFnzeVSIAAAQIE6gIKvr4CAxAgQIAAgbyAgs+bSiRAgAABAnUBBV9fgQEIECBAgEBeQMHnTSUSIECAAIG6gIKvr8AABAgQIEAgL6Dg86YSCRAgQIBAXUDB11dgAAIECBAgkBdQ8HlTiQQIECBAoC6g4OsrMAABAgQIEMgLKPi8qUQCBAgQIFAXUPD1FRiAAAECBAjkBRR83lQiAQIECBCoCyj4+goMQIAAAQIE8gIKPm8qkQABAgQI1AUUfH0FBiBAgAABAnkBBZ83lUiAAAECBOoCCr6+AgMQIECAAIG8gILPm0okQIAAAQJ1AQVfX4EBCBAgQIBAXkDB500lEiBAgACBuoCCr6/AAAQIECBAIC+g4POmEgkQIECAQF1AwddXYAACBAgQIJAXUPB5U4kECBAgQKAuoODrKzAAAQIECBDICyj4vKlEAgQIECBQF1Dw9RUYgAABAgQI5AUUfN5UIgECBAgQqAso+PoKDECAAAECBPICCj5vKpEAAQIECNQFFHx9BQYgQIAAAQJ5AQWfN5VIgAABAgTqAgq+vgIDECBAgACBvICCz5tKJECAAAECdQEFX1+BAQgQIECAQF5AwedNJRIgQIAAgbqAgq+vwAAECBAgQCAvoODzphIJECBAgEBdQMHXV2AAAgQIECCQF1DweVOJBAgQIECgLqDg6yswAAECBAgQyAso+LypRAIECBAgUBdQ8PUVGIAAAQIECOQFFHzeVCIBAgQIEKgLKPj6CgxAgAABAgTyAgo+byqRAAECBAjUBRR8fQUGIECAAAECeQEFnzeVSIAAAQIE6gIKvr4CAxAgQIAAgbyAgs+bSiRAgAABAnUBBV9fgQEIECBAgEBeQMHnTSUSIECAAIG6gIKvr8AABAgQIEAgL6Dg86YSCRAgQIBAXUDB11dgAAIECBAgkBdQ8HlTiQQIECBAoC6g4OsrMAABAgQIEMgLKPi8qUQCBAgQIFAXUPD1FRiAAAECBAjkBRR83lQiAQIECBCoCyj4+goMQIAAAQIE8gIKPm8qkQABAgQI1AUUfH0FBiBAgAABAnkBBZ83lUiAAAECBOoCCr6+AgMQIECAAIG8gILPm0okQIAAAQJ1AQVfX4EBCBAgQIBAXkDB500lEiBAgACBuoCCr6/AAAQIECBAIC+g4POmEgkQIECAQF1AwddXYAACBAgQIJAXUPB5U4kECBAgQKAuoODrKzAAAQIECBDICyj4vKlEAgQIECBQF1Dw9RUYgAABAgQI5AUUfN5UIgECBAgQqAso+PoKDECAAAECBPICCj5vKpEAAQIECNQFFHx9BQYgQIAAAQJ5AQWfN5VIgAABAgTqAgq+vgIDECBAgACBvICCz5tKJECAAAECdQEFX1+BAQgQIECAQF5AwedNJRIgQIAAgbqAgq+vwAAECBAgQCAvoODzphIJECBAgEBdQMHXV2AAAgQIECCQF1DweVOJBAgQIECgLqDg6yswAAECBAgQyAso+LypRAIECBAgUBdQ8PUVGIAAAQIECOQFFHzeVCIBAgQIEKgLKPj6CgxAgAABAgTyAgo+byqRAAECBAjUBRR8fQUGIECAAAECeQEFnzeVSIAAAQIE6gIKvr4CAxAgQIAAgbyAgs+bSiRAgAABAnUBBV9fgQEIECBAgEBeQMHnTSUSIECAAIG6gIKvr8AABAgQIEAgL6Dg86YSCRAgQIBAXUDB11dgAAIECBAgkBdQ8HlTiQQIECBAoC6g4OsrMAABAgQIEMgLKPi8qUQCBAgQIFAXUPD1FRiAAAECBAjkBRR83lQiAQIECBCoCyj4+goMQIAAAQIE8gIKPm8qkQABAgQI1AUUfH0FBiBAgAABAnkBBZ83lUiAAAECBOoCCr6+AgMQIECAAIG8gILPm0okQIAAAQJ1AQVfX4EBCBAgQIBAXkDB500lEiBAgACBuoCCr6/AAAQIECBAIC+g4POmEgkQIECAQF1AwddXYAACBAgQIJAXUPB5U4kECBAgQKAuoODrKzAAAQIECBDICyj4vKlEAgQIECBQF1Dw9RUYgAABAgQI5AUUfN5UIgECBAgQqAso+PoKDECAAAECBPICCj5vKpEAAQIECNQFFHx9BQYgQIAAAQJ5AQWfN5VIgAABAgTqAgq+vgIDECBAgACBvICCz5tKJECAAAECdQEFX1+BAQgQIECAQF5AwedNJRIgQIAAgbqAgq+vwAAECBAgQCAvoODzphIJECBAgEBdQMHXV2AAAgQIECCQF1DweVOJBAgQIECgLqDg6yswAAECBAgQyAso+LypRAIECBAgUBdQ8PUVGIAAAQIECOQFFHzeVCIBAgQIEKgLKPj6CgxAgAABAgTyAgo+byqRAAECBAjUBRR8fQUGIECAAAECeQEFnzeVSIAAAQIE6gIKvr4CAxAgQIAAgbyAgs+bSiRAgAABAnUBBV9fgQEIECBAgEBeQMHnTSUSIECAAIG6gIKvr8AABAgQIEAgL6Dg86YSCRAgQIBAXUDB11dgAAIECBAgkBdQ8HlTiQQIECBAoC6g4OsrMAABAgQIEMgLKPi8qUQCBAgQIFAXUPD1FRiAAAECBAjkBRR83lQiAQIECBCoCyj4+goMQIAAAQIE8gIKPm8qkQABAgQI1AUUfH0FBiBAgAABAnkBBZ83lUiAAAECBOoCCr6+AgMQIECAAIG8gILPm0okQIAAAQJ1AQVfX4EBCBAgQIBAXkDB500lEiBAgACBuoCCr6/AAAQIECBAIC+g4POmEgkQIECAQF1AwddXYAACBAgQIJAXUPB5U4kECBAgQKAuoODrKzAAAQIECBDICyj4vKlEAgQIECBQF1Dw9RUYgAABAgQI5AUUfN5UIgECBAgQqAso+PoKDECAAAECBPICCj5vKpEAAQIECNQFFHx9BQYgQIAAAQJ5AQWfN5VIgAABAgTqAgq+vgIDECBAgACBvICCz5tKJECAAAECdQEFX1+BAQgQIECAQF5AwedNJRIgQIAAgbqAgq+vwAAECBAgQCAvoODzphIJECBAgEBdQMHXV2AAAgQIECCQF1DweVOJBAgQIECgLqDg6yswAAECBAgQyAso+LypRAIECBAgUBdQ8PUVGIAAAQIECOQFFHzeVCIBAgQIEKgLKPj6CgxAgAABAgTyAgo+byqRAAECBAjUBRR8fQUGIECAAAECeQEFnzeVSIAAAQIE6gIKvr4CAxAgQIAAgbyAgs+bSiRAgAABAnUBBV9fgQEIECBAgEBeQMHnTSUSIECAAIG6gIKvr8AABAgQIEAgL6Dg86YSCRAgQIBAXUDB11dgAAIECBAgkBdQ8HlTiQQIECBAoC6g4OsrMAABAgQIEMgLKPi8qUQCBAgQIFAXUPD1FRiAAAECBAjkBRR83lQiAQIECBCoCyj4+goMQIAAAQIE8gIKPm8qkQABAgQI1AUUfH0FBiBAgAABAnkBBZ83lUiAAAECBOoCCr6+AgMQIECAAIG8gILPm0okQIAAAQJ1AQVfX4EBCBAgQIBAXkDB500lEiBAgACBuoCCr6/AAAQIECBAIC+g4POmEgkQIECAQF1AwddXYAACBAgQIJAXUPB5U4kECBAgQKAuoODrKzAAAQIECBDICyj4vKlEAgQIECBQF1Dw9RUYgAABAgQI5AUUfN5UIgECBAgQqAso+PoKDECAAAECBPICCj5vKpEAAQIECNQFFHx9BQYgQIAAAQJ5AQWfN5VIgAABAgTqAgq+vgIDECBAgACBvICCz5tKJECAAAECdQEFX1+BAQgQIECAQF5AwedNJRIgQIAAgbqAgq+vwAAECBAgQCAvoODzphIJECBAgEBdQMHXV2AAAgQIECCQF1DweVOJBAgQIECgLqDg6yswAAECBAgQyAso+LypRAIECBAgUBdQ8PUVGIAAAQIECOQFFHzeVCIBAgQIEKgLKPj6CgxAgAABAgTyAgo+byqRAAECBAjUBRR8fQUGIECAAAECeQEFnzeVSIAAAQIE6gIKvr4CAxAgQIAAgbyAgs+bSiRAgAABAnUBBV9fgQEIECBAgEBeQMHnTSUSIECAAIG6gIKvr8AABAgQIEAgL6Dg86YSCRAgQIBAXUDB11dgAAIECBAgkBdQ8HlTiQQIECBAoC6g4OsrMAABAgQIEMgLKPi8qUQCBAgQIFAXUPD1FRiAAAECBAjkBRR83lQiAQIECBCoCyj4+goMQIAAAQIE8gIKPm8qkQABAgQI1AUUfH0FBiBAgAABAnkBBZ83lUiAAAECBOoCCr6+AgMQIECAAIG8gILPm0okQIAAAQJ1AQVfX4EBCBAgQIBAXkDB500lEiBAgACBuoCCr6/AAAQIECBAIC+g4POmEgkQIECAQF1AwddXYAACBAgQIJAXUPB5U4kECBAgQKAuoODrKzAAAQIECBDICyj4vKlEAgQIECBQF1Dw9RUYgAABAgQI5AUUfN5UIgECBAgQqAso+PoKDECAAAECBPICCj5vKpEAAQIECNQFFHx9BQYgQIAAAQJ5AQWfN5VIgAABAgTqAgq+vgIDECBAgACBvICCz5tKJECAAAECdQEFX1+BAQgQIECAQF5AwedNJRIgQIAAgbqAgq+vwAAECBAgQCAvoODzphIJECBAgEBdQMHXV2AAAgQIECCQF1DweVOJBAgQIECgLqDg6yswAAECBAgQyAso+LypRAIECBAgUBdQ8PUVGIAAAQIECOQFFHzeVCIBAgQIEKgLKPj6CgxAgAABAgTyAgo+byqRAAECBAjUBRR8fQUGIECAAAECeQEFnzeVSIAAAQIE6gIKvr4CAxAgQIAAgbyAgs+bSiRAgAABAnUBBV9fgQEIECBAgEBeQMHnTSUSIECAAIG6gIKvr8AABAgQIEAgL6Dg86YSCRAgQIBAXUDB11dgAAIECBAgkBdQ8HlTiQQIECBAoC6g4OsrMAABAgQIEMgLKPi8qUQCBAgQIFAXUPD1FRiAAAECBAjkBRR83lQiAQIECBCoCyj4+goMQIAAAQIE8gIKPm8qkQABAgQI1AUUfH0FBiBAgAABAnkBBZ83lUiAAAECBOoCCr6+AgMQIECAAIG8gILPm0okQIAAAQJ1AQVfX4EBCBAgQIBAXkDB500lEiBAgACBuoCCr6/AAAQIECBAIC+g4POmEgkQIECAQF1AwddXYAACBAgQIJAXUPB5U4kECBAgQKAuoODrKzAAAQIECBDICyj4vKlEAgQIECBQF1Dw9RUYgAABAgQI5AUUfN5UIgECBAgQqAso+PoKDECAAAECBPICCj5vKpEAAQIECNQFFHx9BQYgQIAAAQJ5AQWfN5VIgAABAgTqAgq+vgIDECBAgACBvICCz5tKJECAAAECdQEFX1+BAQgQIECAQF5AwedNJRIgQIAAgbqAgq+vwAAECBAgQCAvoODzphIJECBAgEBdQMHXV2AAAgQIECCQF1DweVOJBAgQIECgLqDg6yswAAECBAgQyAso+LypRAIECBAgUBdQ8PUVGIAAAQIECOQFFHzeVCIBAgQIEKgLKPj6CgxAgAABAgTyAgo+byqRAAECBAjUBRR8fQUGIECAAAECeQEFnzeVSIAAAQIE6gIKvr4CAxAgQIAAgbyAgs+bSiRAgAABAnUBBV9fgQEIECBAgEBeQMHnTSUSIECAAIG6gIKvr8AABAgQIEAgL6Dg86YSCRAgQIBAXUDB11dgAAIECBAgkBdQ8HlTiQQIECBAoC6g4OsrMAABAgQIEMgLKPi8qUQCBAgQIFAXUPD1FRiAAAECBAjkBRR83lQiAQIECBCoCyj4+goMQIAAAQIE8gIKPm8qkQABAgQI1AUUfH0FBiBAgAABAnkBBZ83lUiAAAECBOoCCr6+AgMQIECAAIG8gILPm0okQIAAAQJ1AQVfX4EBCBAgQIBAXkDB500lEiBAgACBuoCCr6/AAAQIECBAIC+g4POmEgkQIECAQF1AwddXYAACBAgQIJAXUPB5U4kECBAgQKAuoODrKzAAAQIECBDICyj4vKlEAgQIECBQF1Dw9RUYgAABAgQI5AUUfN5UIgECBAgQqAso+PoKDECAAAECBPICCj5vKpEAAQIECNQFFHx9BQYgQIAAAQJ5AQWfN5VIgAABAgTqAgq+vgIDECBAgACBvICCz5tKJECAAAECdQEFX1+BAQgQIECAQF5AwedNJRIgQIAAgbqAgq+vwAAECBAgQCAvoODzphIJECBAgEBdQMHXV2AAAgQIECCQF1DweVOJBAgQIECgLqDg6yswAAECBAgQyAso+LypRAIECBAgUBdQ8PUVGIAAAQIECOQFFHzeVCIBAgQIEKgLKPj6CgxAgAABAgTyAgo+byqRAAECBAjUBf4HcZhrVUWVYHAAAAAASUVORK5CYII=" alt="plot of chunk :combatVennHeart"/> </p>

<pre><code class="r">draw.quad.venn(100, 100, 100, 100, 
               length(intersect(HPA_h_c,AltIso_h_c)),
               length(intersect(HPA_h_c,GTEx_h_c)),
               length(intersect(HPA_h_c,Atlas_h_c)),
               length(intersect(AltIso_h_c,GTEx_h_c)),
               length(intersect(AltIso_h_c,Atlas_h_c)),
               length(intersect(GTEx_h_c,Atlas_h_c)),
               length(intersect(intersect(HPA_h_c,AltIso_h_c),GTEx_h_c)),
               length(intersect(intersect(HPA_h_c,AltIso_h_c),Atlas_h_c)),
               length(intersect(intersect(HPA_h_c,GTEx_h_c),Atlas_h_c)),
               length(intersect(intersect(AltIso_h_c,GTEx_h_c),Atlas_h_c)),
               length(intersect(intersect(HPA_h_c,AltIso_h_c),intersect(Atlas_h_c,GTEx_h_c))),
               category = c(&quot;HPA&quot;,&quot;AltIso&quot;,&quot;GTEx&quot;,&quot;Atlas&quot;), lwd = rep(0, 4), lty = rep(&quot;solid&quot;, 4),
               fill = c(&quot;mistyrose&quot;,&quot;steelblue&quot;,&quot;lightgoldenrod&quot;,&quot;darkseagreen&quot;)

)
</code></pre>

<pre><code>## Error: object &#39;GTEx_h_c&#39; not found
</code></pre>

<p>Kidney:</p>

<pre><code class="r">r &lt;- rectGrob(gp=gpar(fill=&quot;white&quot;,lwd=0))
top &lt;- arrangeGrob(r, r, r, r, nrow=2)
grid.arrange(top, r, ncol=1, main = &quot;\n\nKidney Combat&quot;, heights=c(2, 1))
</code></pre>

<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAfgAAAH4CAYAAACmKP9/AAAEJGlDQ1BJQ0MgUHJvZmlsZQAAOBGFVd9v21QUPolvUqQWPyBYR4eKxa9VU1u5GxqtxgZJk6XtShal6dgqJOQ6N4mpGwfb6baqT3uBNwb8AUDZAw9IPCENBmJ72fbAtElThyqqSUh76MQPISbtBVXhu3ZiJ1PEXPX6yznfOec7517bRD1fabWaGVWIlquunc8klZOnFpSeTYrSs9RLA9Sr6U4tkcvNEi7BFffO6+EdigjL7ZHu/k72I796i9zRiSJPwG4VHX0Z+AxRzNRrtksUvwf7+Gm3BtzzHPDTNgQCqwKXfZwSeNHHJz1OIT8JjtAq6xWtCLwGPLzYZi+3YV8DGMiT4VVuG7oiZpGzrZJhcs/hL49xtzH/Dy6bdfTsXYNY+5yluWO4D4neK/ZUvok/17X0HPBLsF+vuUlhfwX4j/rSfAJ4H1H0qZJ9dN7nR19frRTeBt4Fe9FwpwtN+2p1MXscGLHR9SXrmMgjONd1ZxKzpBeA71b4tNhj6JGoyFNp4GHgwUp9qplfmnFW5oTdy7NamcwCI49kv6fN5IAHgD+0rbyoBc3SOjczohbyS1drbq6pQdqumllRC/0ymTtej8gpbbuVwpQfyw66dqEZyxZKxtHpJn+tZnpnEdrYBbueF9qQn93S7HQGGHnYP7w6L+YGHNtd1FJitqPAR+hERCNOFi1i1alKO6RQnjKUxL1GNjwlMsiEhcPLYTEiT9ISbN15OY/jx4SMshe9LaJRpTvHr3C/ybFYP1PZAfwfYrPsMBtnE6SwN9ib7AhLwTrBDgUKcm06FSrTfSj187xPdVQWOk5Q8vxAfSiIUc7Z7xr6zY/+hpqwSyv0I0/QMTRb7RMgBxNodTfSPqdraz/sDjzKBrv4zu2+a2t0/HHzjd2Lbcc2sG7GtsL42K+xLfxtUgI7YHqKlqHK8HbCCXgjHT1cAdMlDetv4FnQ2lLasaOl6vmB0CMmwT/IPszSueHQqv6i/qluqF+oF9TfO2qEGTumJH0qfSv9KH0nfS/9TIp0Wboi/SRdlb6RLgU5u++9nyXYe69fYRPdil1o1WufNSdTTsp75BfllPy8/LI8G7AUuV8ek6fkvfDsCfbNDP0dvRh0CrNqTbV7LfEEGDQPJQadBtfGVMWEq3QWWdufk6ZSNsjG2PQjp3ZcnOWWing6noonSInvi0/Ex+IzAreevPhe+CawpgP1/pMTMDo64G0sTCXIM+KdOnFWRfQKdJvQzV1+Bt8OokmrdtY2yhVX2a+qrykJfMq4Ml3VR4cVzTQVz+UoNne4vcKLoyS+gyKO6EHe+75Fdt0Mbe5bRIf/wjvrVmhbqBN97RD1vxrahvBOfOYzoosH9bq94uejSOQGkVM6sN/7HelL4t10t9F4gPdVzydEOx83Gv+uNxo7XyL/FtFl8z9ZAHF4bBsrEwAAH7lJREFUeAHt23vM/vX8B/B3J80KU4RUqDmbMAwlf2ROtbWYOa+izcawNCWasJo/mDZGzPlQqJjTxtgaOlDNkBiLTYhVJJmcS+/Pdt373rHvr77P78+eu3pcW33vw3U/79f1eF3b8/p8Pte9w8233IYbAQIECBAgsFYCO67Vo/FgCBAgQIAAgUVAwXsiECBAgACBNRRQ8Gu4VA+JAAECBAgoeM8BAgQIECCwhgIKfg2X6iERIECAAAEF7zlAgAABAgTWUEDBr+FSPSQCBAgQIKDgPQcIECBAgMAaCij4NVyqh0SAAAECBBS85wABAgQIEFhDAQW/hkv1kAgQIECAgIL3HCBAgAABAmsooODXcKkeEgECBAgQUPCeAwQIECBAYA0FFPwaLtVDIkCAAAECCt5zgAABAgQIrKGAgl/DpXpIBAgQIEBAwXsOECBAgACBNRRQ8Gu4VA+JAAECBAgoeM8BAgQIECCwhgIKfg2X6iERIECAAAEF7zlAgAABAgTWUEDBr+FSPSQCBAgQIKDgPQcIECBAgMAaCij4NVyqh0SAAAECBBS85wABAgQIEFhDAQW/hkv1kAgQIECAgIL3HCBAgAABAmsooODXcKkeEgECBAgQUPCeAwQIECBAYA0FFPwaLtVDIkCAAAECCt5zgAABAgQIrKGAgl/DpXpIBAgQIEBAwXsOELiDCvzxj3+8gz5yD5vAHUNAwd8x9uxR/g8E3vGOd4xnP/vZm37T8ccfPx7ykIeMX//61+PEE08cJ5988qbvrz65973vPX7+85+vPv1/+/fmm28eb37zm5eZHv/4x48HP/jB4zWvec34+9//vt1+5/3ud79x+eWX36686fPud7/7dv2MOxMgsHUBBb91H98lsM0Cr33ta8d55503zj///LHPPvuMk046abzuda/b5rzt8YPHHHPM+Pa3vz0uuuii8dOf/nR8//vfHz/72c/GCSecsD3itzljGn3ta1/b5p/3gwQI/KeAgv9PE18hEAvMcr/kkkvGN77xjXHPe95zyfvQhz40PvKRjywf//KXvxzPfOYzxz3ucY/lCPqmm25avv7DH/5wHHXUUcvX9txzz/GoRz1qXHbZZcv35tH3qaeeurxYuO997ztOO+20Mb/2t7/9bTz60Y9ezhIsd7zlf69//evHWWedtfp0+feGG24Yn/zkJ8eZZ5459thjj+Vrd77zncec6xnPeMby+ZzjPe95z5I3f8db3vKWsZrt0EMPHR/4wAfGAx7wgOUMwHyRMB/nXnvtNV7wgheMP/3pTxu/72Mf+9jYe++9x4Me9KDx5S9/edPXH/rQh47dd999POYxjxmXXnrp+M1vfjPmmY5p9eIXv3jjvj4gQCATUPCZn58msElgFu5xxx23nG7+zGc+M+52t7ttfP+aa64Zv/vd75bPX/KSl4wHPvCB4wc/+MH461//Oq699trl6/PjWcKzgOdp7ic+8YnjDW94w/K9T3ziE8v3vvSlL43Pf/7z41Of+tTyImLXXXcd+++///jc5z633O+f//zn+PCHPzwOOeSQ5fPV/7773e+Offfdd3lRsfra/HdeHlgV/Hvf+95xxhlnjPe///3jnHPOWV4MzKx5m5cQzj333OWMxGGHHTae+tSnLkU9zwLMU+xf//rXl/vN/11wwQXjm9/85jjllFPG8573vOXxXXHFFeOVr3zl8sLjV7/61XjsYx+7XLKYv/+tb33rOOigg8b73ve+jQwfECCQCSj4zM9PE9gk8NWvfnVceOGF4+CDD94o5k13uOWT6667bnzrW98ab3zjG8c8Sn7Tm9606S53vetdl2K8z33uM57//OePX/ziF8v351HxPMV+wAEHLNfOX/rSl45Z9vM27/fZz352+XgeCc9r6/OywJa3+QJjZm/tNo/6jz322DGvzz/pSU8aRx999JgvLFa3V7/61UvuLO35QmJecphH6k95ylPGd77zndXdliKfL2Be9KIXLUf88yj+Xve617j44ouXswM77rjjeNjDHjZ++9vfjvnxbrvtNnbZZZflBcNGiA8IEIgEFHzE54cJbBaYZfeVr3xlfPCDH1yOdlcFvOW95jXvWb6z8OZtlvz8b3Wbp7xXt1l8s0jn7aqrrhpvf/vbl/KeBT4//t73vrd87/DDD1/OBlx99dXLkfcs4FvfHv7why/X2+dZhi1v8/Of/OQny5euvPLK5azB6vvzDMI8hb66zcc3b/PU/vz4Lne5y/L5ne50p4055xee8IQnLF+f/5tH6jNj3nee1Zizzzcezhckq9P/G3f2AQEC201AwW83SkEExjjwwAPHvHY+j7Lnu9Vf/vKXjz/84Q+baPbbb7+l8K6//vrl63/+85/HPLpe3XbYYYfVh5v+fdzjHjfe9ra3LUe988h3nvJeXWefhTtL/otf/OJyzfu5z33upp+dn8xinWU+T59veZun3ef19b/85S/L6fsf/ehHG9+e7wmYp/9Xt5133nn14Vb/3fJFwY9//OPlMsDHP/7x5UXPLPY5/zz6v/WLja2G+iYBArdLQMHfLi53JnDbBeYb0OZp9le96lWbfmhec55HsPOU+7x9+tOfHv/4xz823ee/fXLEEUcsb9KbLxhmMc43pJ1++ukbd52n6d/5zncu2auzAxvfvOWDeQp8/infvP4/r/3P23xH/bxEMN9MN18kPP3pT1+u7c+/kZ+XEuZ1+Hlt/PbeVu8HmGcY5nsJ5vX6mTdP2z/iEY9Y5v/oRz+68bjnmQp/l397ld2fwNYFFPzWfXyXwDYL7LTTTss71Odp6S984Qubcs4+++zljXjzaH4W3fz3/7o961nPWo6E73//+y9F+a9//Wv52/rVzz3taU8b8xT9fzs9v7rPK17xiuV6/XOe85zlDYCz7I888sjl2v68z3xD33zT3uqd8vNv2ldv8ltl3JZ/55sG54uYJz/5yeNd73rXmDkvfOELlzfjzbMc83LBPHU/j+RvvPHG8chHPnLMI/35VwNuBAhsH4EdbjkS2HxBbvvkSiFA4DYI/P73v19O6d+Gu27cZZ7Sn7d51LvlbZ5in0fI87T63e9+9y2/9V8/nkfUqz+Xu/Ud5p/UzSP+eVS/rbd5RD5/fl6f3/I2H/Ocb765bsvbvB4//+Qv+Z1b5vmYwB1dQMHf0Z8BHv9aCMw/rZvvdp/vTN/ytP1aPDgPggCBbRK4be+Y2aZoP0SAwP9KYL7zfp5qf9nLXva/+pV+DwEC5QKO4MsXZDwCBAgQILAtApsvgm1Lgp8hQIAAAQIE6gQUfN1KDESAAAECBHIBBZ8bSiBAgAABAnUCCr5uJQYiQIAAAQK5gILPDSUQIECAAIE6AQVftxIDESBAgACBXEDB54YSCBAgQIBAnYCCr1uJgQgQIECAQC6g4HNDCQQIECBAoE5AwdetxEAECBAgQCAXUPC5oQQCBAgQIFAnoODrVmIgAgQIECCQCyj43FACAQIECBCoE1DwdSsxEAECBAgQyAUUfG4ogQABAgQI1Ako+LqVGIgAAQIECOQCCj43lECAAAECBOoEFHzdSgxEgAABAgRyAQWfG0ogQIAAAQJ1Agq+biUGIkCAAAECuYCCzw0lECBAgACBOgEFX7cSAxEgQIAAgVxAweeGEggQIECAQJ2Agq9biYEIECBAgEAuoOBzQwkECBAgQKBOQMHXrcRABAgQIEAgF1DwuaEEAgQIECBQJ6Dg61ZiIAIECBAgkAso+NxQAgECBAgQqBNQ8HUrMRABAgQIEMgFFHxuKIEAAQIECNQJKPi6lRiIAAECBAjkAgo+N5RAgAABAgTqBBR83UoMRIAAAQIEcgEFnxtKIECAAAECdQIKvm4lBiJAgAABArmAgs8NJRAgQIAAgToBBV+3EgMRIECAAIFcQMHnhhIIECBAgECdgIKvW4mBCBAgQIBALqDgc0MJBAgQIECgTkDB163EQAQIECBAIBdQ8LmhBAIECBAgUCeg4OtWYiACBAgQIJALKPjcUAIBAgQIEKgTUPB1KzEQAQIECBDIBRR8biiBAAECBAjUCSj4upUYiAABAgQI5AIKPjeUQIAAAQIE6gQUfN1KDESAAAECBHIBBZ8bSiBAgAABAnUCCr5uJQYiQIAAAQK5gILPDSUQIECAAIE6AQVftxIDESBAgACBXEDB54YSCBAgQIBAnYCCr1uJgQgQIECAQC6g4HNDCQQIECBAoE5AwdetxEAECBAgQCAXUPC5oQQCBAgQIFAnoODrVmIgAgQIECCQCyj43FACAQIECBCoE1DwdSsxEAECBAgQyAUUfG4ogQABAgQI1Ako+LqVGIgAAQIECOQCCj43lECAAAECBOoEFHzdSgxEgAABAgRyAQWfG0ogQIAAAQJ1Agq+biUGIkCAAAECuYCCzw0lECBAgACBOgEFX7cSAxEgQIAAgVxAweeGEggQIECAQJ2Agq9biYEIECBAgEAuoOBzQwkECBAgQKBOQMHXrcRABAgQIEAgF1DwuaEEAgQIECBQJ6Dg61ZiIAIECBAgkAso+NxQAgECBAgQqBNQ8HUrMRABAgQIEMgFFHxuKIEAAQIECNQJKPi6lRiIAAECBAjkAgo+N5RAgAABAgTqBBR83UoMRIAAAQIEcgEFnxtKIECAAAECdQIKvm4lBiJAgAABArmAgs8NJRAgQIAAgToBBV+3EgMRIECAAIFcQMHnhhIIECBAgECdgIKvW4mBCBAgQIBALqDgc0MJBAgQIECgTkDB163EQAQIECBAIBdQ8LmhBAIECBAgUCeg4OtWYiACBAgQIJALKPjcUAIBAgQIEKgTUPB1KzEQAQIECBDIBRR8biiBAAECBAjUCSj4upUYiAABAgQI5AIKPjeUQIAAAQIE6gQUfN1KDESAAAECBHIBBZ8bSiBAgAABAnUCCr5uJQYiQIAAAQK5gILPDSUQIECAAIE6AQVftxIDESBAgACBXEDB54YSCBAgQIBAnYCCr1uJgQgQIECAQC6g4HNDCQQIECBAoE5AwdetxEAECBAgQCAXUPC5oQQCBAgQIFAnoODrVmIgAgQIECCQCyj43FACAQIECBCoE1DwdSsxEAECBAgQyAUUfG4ogQABAgQI1Ako+LqVGIgAAQIECOQCCj43lECAAAECBOoEFHzdSgxEgAABAgRyAQWfG0ogQIAAAQJ1Agq+biUGIkCAAAECuYCCzw0lECBAgACBOgEFX7cSAxEgQIAAgVxAweeGEggQIECAQJ2Agq9biYEIECBAgEAuoOBzQwkECBAgQKBOQMHXrcRABAgQIEAgF1DwuaEEAgQIECBQJ6Dg61ZiIAIECBAgkAso+NxQAgECBAgQqBNQ8HUrMRABAgQIEMgFFHxuKIEAAQIECNQJKPi6lRiIAAECBAjkAgo+N5RAgAABAgTqBBR83UoMRIAAAQIEcgEFnxtKIECAAAECdQIKvm4lBiJAgAABArmAgs8NJRAgQIAAgToBBV+3EgMRIECAAIFcQMHnhhIIECBAgECdgIKvW4mBCBAgQIBALqDgc0MJBAgQIECgTkDB163EQAQIECBAIBdQ8LmhBAIECBAgUCeg4OtWYiACBAgQIJALKPjcUAIBAgQIEKgTUPB1KzEQAQIECBDIBRR8biiBAAECBAjUCSj4upUYiAABAgQI5AIKPjeUQIAAAQIE6gQUfN1KDESAAAECBHIBBZ8bSiBAgAABAnUCCr5uJQYiQIAAAQK5gILPDSUQIECAAIE6AQVftxIDESBAgACBXEDB54YSCBAgQIBAnYCCr1uJgQgQIECAQC6g4HNDCQQIECBAoE5AwdetxEAECBAgQCAXUPC5oQQCBAgQIFAnoODrVmIgAgQIECCQCyj43FACAQIECBCoE1DwdSsxEAECBAgQyAUUfG4ogQABAgQI1Ako+LqVGIgAAQIECOQCCj43lECAAAECBOoEFHzdSgxEgAABAgRyAQWfG0ogQIAAAQJ1Agq+biUGIkCAAAECuYCCzw0lECBAgACBOgEFX7cSAxEgQIAAgVxAweeGEggQIECAQJ2Agq9biYEIECBAgEAuoOBzQwkECBAgQKBOQMHXrcRABAgQIEAgF1DwuaEEAgQIECBQJ6Dg61ZiIAIECBAgkAso+NxQAgECBAgQqBNQ8HUrMRABAgQIEMgFFHxuKIEAAQIECNQJKPi6lRiIAAECBAjkAgo+N5RAgAABAgTqBBR83UoMRIAAAQIEcgEFnxtKIECAAAECdQIKvm4lBiJAgAABArmAgs8NJRAgQIAAgToBBV+3EgMRIECAAIFcQMHnhhIIECBAgECdgIKvW4mBCBAgQIBALqDgc0MJBAgQIECgTkDB163EQAQIECBAIBdQ8LmhBAIECBAgUCeg4OtWYiACBAgQIJALKPjcUAIBAgQIEKgTUPB1KzEQAQIECBDIBRR8biiBAAECBAjUCSj4upUYiAABAgQI5AIKPjeUQIAAAQIE6gQUfN1KDESAAAECBHIBBZ8bSiBAgAABAnUCCr5uJQYiQIAAAQK5gILPDSUQIECAAIE6AQVftxIDESBAgACBXEDB54YSCBAgQIBAnYCCr1uJgQgQIECAQC6g4HNDCQQIECBAoE5AwdetxEAECBAgQCAXUPC5oQQCBAgQIFAnoODrVmIgAgQIECCQCyj43FACAQIECBCoE1DwdSsxEAECBAgQyAUUfG4ogQABAgQI1Ako+LqVGIgAAQIECOQCCj43lECAAAECBOoEFHzdSgxEgAABAgRyAQWfG0ogQIAAAQJ1Agq+biUGIkCAAAECuYCCzw0lECBAgACBOgEFX7cSAxEgQIAAgVxAweeGEggQIECAQJ2Agq9biYEIECBAgEAuoOBzQwkECBAgQKBOQMHXrcRABAgQIEAgF1DwuaEEAgQIECBQJ6Dg61ZiIAIECBAgkAso+NxQAgECBAgQqBNQ8HUrMRABAgQIEMgFFHxuKIEAAQIECNQJKPi6lRiIAAECBAjkAgo+N5RAgAABAgTqBBR83UoMRIAAAQIEcgEFnxtKIECAAAECdQIKvm4lBiJAgAABArmAgs8NJRAgQIAAgToBBV+3EgMRIECAAIFcQMHnhhIIECBAgECdgIKvW4mBCBAgQIBALqDgc0MJBAgQIECgTkDB163EQAQIECBAIBdQ8LmhBAIECBAgUCeg4OtWYiACBAgQIJALKPjcUAIBAgQIEKgTUPB1KzEQAQIECBDIBRR8biiBAAECBAjUCSj4upUYiAABAgQI5AIKPjeUQIAAAQIE6gQUfN1KDESAAAECBHIBBZ8bSiBAgAABAnUCCr5uJQYiQIAAAQK5gILPDSUQIECAAIE6AQVftxIDESBAgACBXEDB54YSCBAgQIBAnYCCr1uJgQgQIECAQC6g4HNDCQQIECBAoE5AwdetxEAECBAgQCAXUPC5oQQCBAgQIFAnoODrVmIgAgQIECCQCyj43FACAQIECBCoE1DwdSsxEAECBAgQyAUUfG4ogQABAgQI1Ako+LqVGIgAAQIECOQCCj43lECAAAECBOoEFHzdSgxEgAABAgRyAQWfG0ogQIAAAQJ1Agq+biUGIkCAAAECuYCCzw0lECBAgACBOgEFX7cSAxEgQIAAgVxAweeGEggQIECAQJ2Agq9biYEIECBAgEAuoOBzQwkECBAgQKBOQMHXrcRABAgQIEAgF1DwuaEEAgQIECBQJ6Dg61ZiIAIECBAgkAso+NxQAgECBAgQqBNQ8HUrMRABAgQIEMgFFHxuKIEAAQIECNQJKPi6lRiIAAECBAjkAgo+N5RAgAABAgTqBBR83UoMRIAAAQIEcgEFnxtKIECAAAECdQIKvm4lBiJAgAABArmAgs8NJRAgQIAAgToBBV+3EgMRIECAAIFcQMHnhhIIECBAgECdgIKvW4mBCBAgQIBALqDgc0MJBAgQIECgTkDB163EQAQIECBAIBdQ8LmhBAIECBAgUCeg4OtWYiACBAgQIJALKPjcUAIBAgQIEKgTUPB1KzEQAQIECBDIBRR8biiBAAECBAjUCSj4upUYiAABAgQI5AIKPjeUQIAAAQIE6gQUfN1KDESAAAECBHIBBZ8bSiBAgAABAnUCCr5uJQYiQIAAAQK5gILPDSUQIECAAIE6AQVftxIDESBAgACBXEDB54YSCBAgQIBAnYCCr1uJgQgQIECAQC6g4HNDCQQIECBAoE5AwdetxEAECBAgQCAXUPC5oQQCBAgQIFAnoODrVmIgAgQIECCQCyj43FACAQIECBCoE1DwdSsxEAECBAgQyAUUfG4ogQABAgQI1Ako+LqVGIgAAQIECOQCCj43lECAAAECBOoEFHzdSgxEgAABAgRyAQWfG0ogQIAAAQJ1Agq+biUGIkCAAAECuYCCzw0lECBAgACBOgEFX7cSAxEgQIAAgVxAweeGEggQIECAQJ2Agq9biYEIECBAgEAuoOBzQwkECBAgQKBOQMHXrcRABAgQIEAgF1DwuaEEAgQIECBQJ6Dg61ZiIAIECBAgkAso+NxQAgECBAgQqBNQ8HUrMRABAgQIEMgFFHxuKIEAAQIECNQJKPi6lRiIAAECBAjkAgo+N5RAgAABAgTqBBR83UoMRIAAAQIEcgEFnxtKIECAAAECdQIKvm4lBiJAgAABArmAgs8NJRAgQIAAgToBBV+3EgMRIECAAIFcQMHnhhIIECBAgECdgIKvW4mBCBAgQIBALqDgc0MJBAgQIECgTkDB163EQAQIECBAIBdQ8LmhBAIECBAgUCeg4OtWYiACBAgQIJALKPjcUAIBAgQIEKgTUPB1KzEQAQIECBDIBRR8biiBAAECBAjUCSj4upUYiAABAgQI5AIKPjeUQIAAAQIE6gQUfN1KDESAAAECBHIBBZ8bSiBAgAABAnUCCr5uJQYiQIAAAQK5gILPDSUQIECAAIE6AQVftxIDESBAgACBXEDB54YSCBAgQIBAnYCCr1uJgQgQIECAQC6g4HNDCQQIECBAoE5AwdetxEAECBAgQCAXUPC5oQQCBAgQIFAnoODrVmIgAgQIECCQCyj43FACAQIECBCoE1DwdSsxEAECBAgQyAUUfG4ogQABAgQI1Ako+LqVGIgAAQIECOQCCj43lECAAAECBOoEFHzdSgxEgAABAgRyAQWfG0ogQIAAAQJ1Agq+biUGIkCAAAECuYCCzw0lECBAgACBOgEFX7cSAxEgQIAAgVxAweeGEggQIECAQJ2Agq9biYEIECBAgEAuoOBzQwkECBAgQKBOQMHXrcRABAgQIEAgF1DwuaEEAgQIECBQJ6Dg61ZiIAIECBAgkAso+NxQAgECBAgQqBNQ8HUrMRABAgQIEMgFFHxuKIEAAQIECNQJKPi6lRiIAAECBAjkAgo+N5RAgAABAgTqBBR83UoMRIAAAQIEcgEFnxtKIECAAAECdQIKvm4lBiJAgAABArmAgs8NJRAgQIAAgToBBV+3EgMRIECAAIFcQMHnhhIIECBAgECdgIKvW4mBCBAgQIBALqDgc0MJBAgQIECgTkDB163EQAQIECBAIBdQ8LmhBAIECBAgUCeg4OtWYiACBAgQIJALKPjcUAIBAgQIEKgTUPB1KzEQAQIECBDIBRR8biiBAAECBAjUCSj4upUYiAABAgQI5AIKPjeUQIAAAQIE6gQUfN1KDESAAAECBHIBBZ8bSiBAgAABAnUCCr5uJQYiQIAAAQK5gILPDSUQIECAAIE6AQVftxIDESBAgACBXEDB54YSCBAgQIBAnYCCr1uJgQgQIECAQC6g4HNDCQQIECBAoE5AwdetxEAECBAgQCAXUPC5oQQCBAgQIFAnoODrVmIgAgQIECCQCyj43FACAQIECBCoE1DwdSsxEAECBAgQyAUUfG4ogQABAgQI1Ako+LqVGIgAAQIECOQCCj43lECAAAECBOoEFHzdSgxEgAABAgRyAQWfG0ogQIAAAQJ1Agq+biUGIkCAAAECuYCCzw0lECBAgACBOgEFX7cSAxEgQIAAgVxAweeGEggQIECAQJ2Agq9biYEIECBAgEAuoOBzQwkECBAgQKBOQMHXrcRABAgQIEAgF1DwuaEEAgQIECBQJ6Dg61ZiIAIECBAgkAso+NxQAgECBAgQqBNQ8HUrMRABAgQIEMgFFHxuKIEAAQIECNQJKPi6lRiIAAECBAjkAgo+N5RAgAABAgTqBBR83UoMRIAAAQIEcgEFnxtKIECAAAECdQIKvm4lBiJAgAABArmAgs8NJRAgQIAAgToBBV+3EgMRIECAAIFcQMHnhhIIECBAgECdgIKvW4mBCBAgQIBALqDgc0MJBAgQIECgTkDB163EQAQIECBAIBdQ8LmhBAIECBAgUCeg4OtWYiACBAgQIJALKPjcUAIBAgQIEKgTUPB1KzEQAQIECBDIBRR8biiBAAECBAjUCSj4upUYiAABAgQI5AIKPjeUQIAAAQIE6gQUfN1KDESAAAECBHIBBZ8bSiBAgAABAnUCCr5uJQYiQIAAAQK5gILPDSUQIECAAIE6AQVftxIDESBAgACBXEDB54YSCBAgQIBAnYCCr1uJgQgQIECAQC6g4HNDCQQIECBAoE5AwdetxEAECBAgQCAXUPC5oQQCBAgQIFAnoODrVmIgAgQIECCQCyj43FACAQIECBCoE1DwdSsxEAECBAgQyAUUfG4ogQABAgQI1Ako+LqVGIgAAQIECOQCCj43lECAAAECBOoEFHzdSgxEgAABAgRyAQWfG0ogQIAAAQJ1Agq+biUGIkCAAAECuYCCzw0lECBAgACBOgEFX7cSAxEgQIAAgVxAweeGEggQIECAQJ2Agq9biYEIECBAgEAu8G/0oY0MxCrdxAAAAABJRU5ErkJggg==" alt="plot of chunk :combatVennKidney"/> </p>

<pre><code class="r">draw.triple.venn(100, 100, 100, 
                 length(intersect(HPA_k_c,GTEx_k_c)),
                 length(intersect(GTEx_k_c,Atlas_k_c)),
                 length(intersect(HPA_k_c,Atlas_k_c)),
                 length(intersect(intersect(HPA_k_c,GTEx_k_c),Atlas_k_c)),
                 category = c(&quot;HPA&quot;,&quot;GTEx&quot;,&quot;Atlas&quot;), lwd = rep(0, 3), lty = rep(&quot;solid&quot;, 3),
                 fill = c(&quot;steelblue&quot;,&quot;lightgoldenrod&quot;,&quot;darkseagreen&quot;)
)
</code></pre>

<pre><code>## Error: object &#39;GTEx_k_c&#39; not found
</code></pre>

<p>Revisit Anova with log-TMMed and combated values.</p>

<pre><code class="r">m &lt;- melt(f.log.tmm)
</code></pre>

<pre><code>## Error: object &#39;f.log.tmm&#39; not found
</code></pre>

<pre><code class="r">colnames(m) &lt;- c(&quot;gene_ID&quot;,&quot;sample_ID&quot;,&quot;logTMMRPKM&quot;)
</code></pre>

<pre><code>## Error: object &#39;m&#39; not found
</code></pre>

<pre><code class="r">data &lt;- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)
</code></pre>

<pre><code>## Error: object &#39;m&#39; not found
</code></pre>

<pre><code class="r">#subset &lt;- data[sample(1:nrow(data), 1000),]
fit &lt;- lm(logTMMRPKM ~ + prep + layout + study + tissue, data=data)
</code></pre>

<pre><code>## Error: &#39;data&#39; argument is of the wrong type
</code></pre>

<pre><code class="r">b &lt;- anova(fit)
</code></pre>

<pre><code>## Error: object &#39;fit&#39; not found
</code></pre>

<pre><code class="r">barplot(b$&quot;F value&quot;,names.arg=rownames(b),main=&quot;Anova F score, log2-TMM&quot;)
</code></pre>

<pre><code>## Error: object &#39;b&#39; not found
</code></pre>

<pre><code class="r">m &lt;- melt(combat)
</code></pre>

<pre><code>## Error: object &#39;combat&#39; not found
</code></pre>

<pre><code class="r">colnames(m) &lt;- c(&quot;gene_ID&quot;,&quot;sample_ID&quot;,&quot;combatlogTMMRPKM&quot;)
</code></pre>

<pre><code>## Error: object &#39;m&#39; not found
</code></pre>

<pre><code class="r">data &lt;- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)
</code></pre>

<pre><code>## Error: object &#39;m&#39; not found
</code></pre>

<pre><code class="r">#subset &lt;- data[sample(1:nrow(data), 1000),]
fit &lt;- lm(combatlogTMMRPKM ~ prep + layout + study + tissue, data=data)
</code></pre>

<pre><code>## Error: &#39;data&#39; argument is of the wrong type
</code></pre>

<pre><code class="r">d &lt;- anova(fit)
</code></pre>

<pre><code>## Error: object &#39;fit&#39; not found
</code></pre>

<pre><code class="r">barplot(d$&quot;F value&quot;,names.arg=rownames(d),main=&quot;Anova F score, ComBat on voom-TMM&quot;)
</code></pre>

<pre><code>## Error: object &#39;d&#39; not found
</code></pre>

<p>Plot results</p>

<pre><code class="r">pdf(&quot;anova.pdf&quot;)
par(mfrow=c(3,1))
barplot(a$&quot;F value&quot;,names.arg=rownames(a),main=&quot;Anova F score, Raw RPKM&quot;)
</code></pre>

<pre><code>## Error: object &#39;a&#39; not found
</code></pre>

<pre><code class="r">barplot(b$&quot;F value&quot;,names.arg=rownames(b),main=&quot;Anova F score, voom-TMM&quot;)
</code></pre>

<pre><code>## Error: object &#39;b&#39; not found
</code></pre>

<pre><code class="r">barplot(d$&quot;F value&quot;,names.arg=rownames(d),main=&quot;Anova F score, ComBat on voom-TMM&quot;)
</code></pre>

<pre><code>## Error: object &#39;d&#39; not found
</code></pre>

<pre><code class="r">dev.off()
</code></pre>

<pre><code>## pdf 
##   2
</code></pre>

</body>

</html>

