Comparing reprocessed RNA-seq data
========================================================

FASTQ files have been downloaded and TopHat and HTSeq have been used to produce count and FPKM tables tables as described elsewhere. (TODO: include link to Gist)

Prepare by loading libraries and defining functions.

```{r prepare}
library(edgeR)
library(pheatmap)
library(ops)
library(calibrate)

normalize.voom <- function(counts){
  require(limma)
	return(voom(counts)$E)
}
 
cpm.tmm <- function(counts, groups=NA){
	require(edgeR)
	if(is.na(groups)){
		d<-DGEList(counts=counts)
	}
	else{
		d<-DGEList(counts=counts, group=groups)
	}
	d <- calcNormFactors(d, method="TMM") 
	return(cpm(d, normalized.lib.sizes=TRUE))
}

do.SVD = function(m, comp.1=1, comp.2=2){ # returns eig.cell
  s <- svd(m)
	ev <- s$d^2 / sum(s$d^2)
	return(s$u[,c(comp.1, comp.2)])
}

project.SVD <- function(m, eig.cell){
	return(t(m) %*% eig.cell)
}

plot.SVD <- function(m, comp.1=1, comp.2=2, groups=rep("blue", ncol(m)), title=""){
	eig <- do.SVD(m, comp.1, comp.2)
	proj <- project.SVD(m, eig)
	xminv <- min(proj[,1]) # - .2 * abs(min(proj[,1]))
	xmaxv <- max(proj[,1]) # + .2 * abs(max(proj[,1]))
	yminv <- min(proj[,2]) # - .2 * abs(min(proj[,2]))
	ymaxv <- max(proj[,2]) # + .2 * abs(max(proj[,2]))
	plot(proj,pch=20,col="white",xlim=c(xminv,xmaxv),ylim=c(yminv,ymaxv),xaxt='n',yaxt='n',xlab="PC1",ylab="PC2",main=title)
	
	points(proj, col=as.character(groups),pch=20) # , #pch=c(rep(15,3),rep(17,3),rep(19,3),rep(18,3),rep(20,2)), cex=2)
	textxy(proj[,1],proj[,2],labs=colnames(m))
}

loadings.SVD <- function(m, comp=1, gene.ids = rownames(m)){
	s <- svd(m)
	l <- s$u[,comp]
	names(l) <- gene.ids
	l.s <- l[order(l)]
	return(l.s)
}

plot.loadings.SVD <- function(m, comp=1, cutoff=0.1, gene.ids = rownames(m)){
	l <- loadings.SVD(m, comp, gene.ids)
	barplot(l[abs(l)>cutoff],las=2,main=paste("PC", comp, "cutoff", cutoff),cex.names=0.6)
}
```

Read data. The columns in the count table have complicated names which we will replace for clarity.

```{r load-data}
counts <- read.delim("count_table.txt",row.names=1)
colnames(counts)<-c("EoEG_brain_TH","EoEG_heart_TH","EoEG_kidney_TH","Atlas_brain_TH","Atlas_heart_TH","Atlas_kidney_TH","BodyMap_brain_TH","BodyMap_heart_TH","BodyMap_kidney_TH","HPA_brain_TH","HPA_heart_TH","HPA_kidney_TH","Wang_brain_TH","Wang_heart_TH")
colors <- c("blue","red","black","blue","red","black","blue","red","black","blue","red","black","blue","red")
```
Make various preprocessed versions: OPS (optimal power space transform), voom (log2 transform), CPM-TMM (edgeR), log2-CPM-TMM.

```{r transform-counts}
p<-findP(counts)$maxIQR
ops.counts <- counts^p
tmm <- cpm.tmm(counts)
voom <- normalize.voom(counts)
dual <- normalize.voom(tmm)
```

SVD plots on the raw, logged, TMMed and logCPM-TMM preprocessed counts.

```{r plot-svd}
plotPC <- function(matrix,a,b,desc,colors){
eig <- do.SVD(matrix, a, b)
proj <- project.SVD(matrix, eig)
xminv <- min(proj[,1]) - .2 * abs(min(proj[,1]))
xmaxv <- max(proj[,1]) + .2 * abs(max(proj[,1]))
yminv <- min(proj[,2]) - .2 * abs(min(proj[,2]))
ymaxv <- max(proj[,2]) + .2 * abs(max(proj[,2]))
plot(proj,pch=20,xlim=c(xminv,xmaxv),ylim=c(yminv,ymaxv),xaxt='n',yaxt='n',xlab=paste0("PC",a),ylab=paste("PC",b),col=colors,main=desc)
textxy(proj[,1],proj[,2],labs=rownames(proj))
}

plotPC(counts,1,2,"Raw counts PC1,2",colors=colors)
plotPC(counts,2,3,"Raw counts PC2,3",colors=colors)
plotPC(voom,1,2,"Log2 counts PC1,2",colors=colors)
plotPC(voom,2,3,"Log2 counts PC2,3",colors=colors)
plotPC(tmm,1,2,"CPM-TMM counts PC1,2",colors=colors)
plotPC(tmm,2,3,"CPM-TMM counts PC2,3",colors=colors)
plotPC(dual,1,2,"logCPM-TMM PC1,2",colors=colors)
plotPC(dual,2,3,"logCPM-TMM PC2,3",colors=colors)
plotPC(dual,3,4,"logCPM-TMM PC3,4",colors=colors)

p <- prcomp(t(voom))
plot(p$x[,2],p$x[,3],pch=20,col=colors,main="prcomp, PCs 2 vs 3")

```

Examine the most loaded genes in PCs 1-3 in raw counts. 

```{r:loadingsPC1-3Raw}
    par(mfrow=c(1,1))
    symbol <- read.table("ENSG_Symbol.txt",header=TRUE)
    p <- prcomp(t(counts))
      plot(p$x[,1],p$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"))
      textxy(p$x[,1],p$x[,2],labs=rownames(p$x))

      load.pc1 <- p$rotation[,1][order(p$rotation[,1])]
      extreme.pc1 <- c(tail(load.pc1), head(load.pc1))
      symbs <- vector()
      for (n in names(extreme.pc1)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc1, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC1 (raw counts)")

     load.pc2 <- p$rotation[,2][order(p$rotation[,2])]
      extreme.pc2 <- c(tail(load.pc2), head(load.pc2))
      symbs <- vector()
      for (n in names(extreme.pc2)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc2, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC2 (raw counts)")

     load.pc3 <- p$rotation[,3][order(p$rotation[,3])]
      extreme.pc3 <- c(tail(load.pc3), head(load.pc3))
      symbs <- vector()
      for (n in names(extreme.pc3)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc3, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC3 (raw counts)")
```

Examine the most loaded genes in PCs 1-3 in log-CPM. 

```{r:loadingsPC1-3logCPM}
    par(mfrow=c(1,1))
    symbol <- read.table("ENSG_Symbol.txt",header=TRUE)
    p <- prcomp(t(voom))
      plot(p$x[,1],p$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"))
      textxy(p$x[,1],p$x[,2],labs=rownames(p$x))

      load.pc1 <- p$rotation[,1][order(p$rotation[,1])]
      extreme.pc1 <- c(tail(load.pc1), head(load.pc1))
      symbs <- vector()
      for (n in names(extreme.pc1)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc1, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC1 (log-CPM)")

     load.pc2 <- p$rotation[,2][order(p$rotation[,2])]
      extreme.pc2 <- c(tail(load.pc2), head(load.pc2))
      symbs <- vector()
      for (n in names(extreme.pc2)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc2, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC2 (raw counts)")

     load.pc3 <- p$rotation[,3][order(p$rotation[,3])]
      extreme.pc3 <- c(tail(load.pc3), head(load.pc3))
      symbs <- vector()
      for (n in names(extreme.pc3)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc3, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC3 (raw counts)")
```

```{r:combat}
library(sva)
meta <- data.frame(study=c(rep("EoGE",3),rep("Atlas",3),rep("BodyMap",3),rep("HPA",3),rep("AltIso",2)),tissue=c("Brain","Heart","Kidney","Brain","Heart","Kidney","Brain","Heart","Kidney","Brain","Heart","Kidney","Brain","Heart"))
batch <- meta$study
design <- model.matrix(~as.factor(tissue),data=meta)
# Combat fails unless we remove low/unexpressed genes
combat <- ComBat(dat=dual,batch=batch,mod=design,numCovs=NULL,par.prior=TRUE)
plotPC(combat,1,2,"ComBat/dual",colors=colors)
combat <- ComBat(dat=voom,batch=batch,mod=design,numCovs=NULL,par.prior=TRUE)
plotPC(combat,1,2,"ComBat/voom",colors=colors)
```

Examine the most loaded genes in PCs 1-3 in ComBat processed data. 

```{r:loadingsPC1-3ComBat}
    par(mfrow=c(1,1))
    symbol <- read.table("ENSG_Symbol.txt",header=TRUE)
    p <- prcomp(t(combat))
      plot(p$x[,1],p$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"))
      textxy(p$x[,1],p$x[,2],labs=rownames(p$x))

      load.pc1 <- p$rotation[,1][order(p$rotation[,1])]
      extreme.pc1 <- c(tail(load.pc1), head(load.pc1))
      symbs <- vector()
      for (n in names(extreme.pc1)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc1, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC1 (log-CPM)")

     load.pc2 <- p$rotation[,2][order(p$rotation[,2])]
      extreme.pc2 <- c(tail(load.pc2), head(load.pc2))
      symbs <- vector()
      for (n in names(extreme.pc2)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc2, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC2 (raw counts)")

     load.pc3 <- p$rotation[,3][order(p$rotation[,3])]
      extreme.pc3 <- c(tail(load.pc3), head(load.pc3))
      symbs <- vector()
      for (n in names(extreme.pc3)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc3, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC3 (raw counts)")
```

ANOVA as for published RPKMs.

```{r:anova-counts}
library(reshape)
m <- melt(counts)
colnames(m) <- c("sample_ID","count")
meta <- data.frame(tissue=c(rep(c("brain","heart","kidney"),4),"brain","heart"),study=c("EoGE","EoGE","EoGE","Atlas","Atlas","Atlas","BodyMap","BodyMap","BodyMap","HPA","HPA","HPA","AltIso","AltIso"),prep=c(rep("poly-A",3),rep("rRNA-depl",3),rep("poly-A",8)),layout=c(rep("PE",3),rep("SE",3),rep("PE",6),rep("SE",2)))
rownames(meta) <- colnames(counts)
tissue <- rep(meta$tissue, each=nrow(counts))
study <- rep(meta$study, each=nrow(counts))
prep <- rep(meta$prep, each=nrow(counts))
layout <- rep(meta$layout, each=nrow(counts))
data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)

#subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(count ~ prep + layout + study + tissue, data=data)
a <- anova(fit)
```
Revisit Anova with log-TMMed and combated values.

```{r:anova2}
m <- melt(dual)
colnames(m) <- c("gene_ID","sample_ID","logTMM")
data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)

#subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(logTMM ~ + prep + layout + study + tissue, data=data)
b <- anova(fit)

m <- melt(combat)
colnames(m) <- c("gene_ID","sample_ID","combatlogTMM")
data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)

#subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(combatlogTMM ~ prep + layout + study + tissue, data=data)
d <- anova(fit)
```

Plot results
```{r:plot-anova3}
pdf("anova_repr_counts.pdf")
par(mfrow=c(3,1))
barplot(a$"F value",names.arg=rownames(a),main="Anova F score, Raw counts")
barplot(b$"F value",names.arg=rownames(b),main="Anova F score, voom-TMM counts")
barplot(d$"F value",names.arg=rownames(d),main="Anova F score, ComBat on voom-TMM")
dev.off()
```

Calculate RPKM and TPM values. For this, we need an auxiliary file of feature lengths

```{r rpkm-tpm}
#lens <- read.delim("gene_lengths.tsv", header=F, row.names=1)
#r <- rpkm(counts[,1], lens, normalized.lib.sizes=F)
#for (i in 2:ncol(counts)){
#  temp <- rpkm(counts[,i], lens, normalized.lib.sizes=F)
#  r <- cbind(r, temp)
#}
#colnames(r)<-colnames(counts)

#tpm <- function(vec, feat.lens, rlens){
#rg <- vec 
#fl <- feat.lens # Feature lengths
#rl <- rlens
#T <- sum(rg*rl/fl)
#tpm <- rg*rl*1e6/(fl*T)
#return(tpm)
#}

#readlengths <- c(76,76,76,35,35,35,152,152,152)
#featlens <- lens
#d.tpm <- tpm(counts[,1], featlens, readlengths)
#for (i in 2:ncol(counts)){
#  temp <- tpm(counts[,i], featlens, readlengths)
 # d.tpm <- cbind(d.tpm, temp)
#}
#colnames(d.tpm)<-colnames(counts)
```

Cufflinks FPKM based analysis
=============================
Cufflinks produces output which occasionally has duplicate entries for genes. We need to collapse these to a single value. One might take the maximum or the sum of genes with the same ID. We choose the latter option here. 

```{r:read-fpkms}
#fpkm <- read.delim("fpkm_table_tophat.txt", sep="\t")
```

```{r SumTranscripts}
#library(data.table)
#data.dt <- data.table(fpkm)
#setkey(data.dt, ENSEMBL_ID)
#temp <- data.dt[, lapply(.SD, sum), by=ENSEMBL_ID]
#collapsed <- as.data.frame(temp)
#fpkms.summed <- collapsed[,3:ncol(collapsed)] 
#rownames(fpkms.summed) <- collapsed[,1]
#write.table(fpkms.summed, file="fpkm_table_tophat_summed.txt", quote=F, sep="\t")
```


TMM normalization on FPKMs.
```{r:fpkm-tmm}
#sizeF <- calcNormFactors(fpkms.summed, method="TMM") 
#fpkm.tmm <- sizeF * fpkms.summed
#fpkm.voom <- normalize.voom(fpkms.summed)
#fpkm.dual <- normalize.voom(fpkm.tmm)
```

Plot PCA/SVD for FPKMs.

```{r:plot-fpkm-pc}
#plotPC(fpkms.summed,1,2,"Raw FPKM PC1,2")
#plotPC(fpkms.summed,2,3,"Raw FPKM PC2,3")
#plotPC(fpkms.summed,3,4,"Raw FPKM PC3,4")
#plotPC(fpkm.tmm,3,4,"TMM-FPKM PC1,2")
#plotPC(fpkm.tmm,3,4,"TMM-FPKM PC2,3")
#plotPC(fpkm.voom,3,4,"Log2 FPKM PC1,2")
#plotPC(fpkm.voom,3,4,"Log2 FPKM PC2,3")
#plotPC(fpkm.dual,3,4,"Log2-TMM FPKM PC1,2")
#plotPC(fpkm.dual,3,4,"Log2-TMM FPKM PC2,3")
```
ANOVA on Cuff-FPKMs.

```{r:anova-FPKM}
#library(reshape)
#m <- melt(fpkms.summed)
#colnames(m) <- c("sample_ID","FPKM")
#meta <- data.frame(tissue=c(rep(c("brain","heart","kidney"),4),"brain","heart"),study=c("EoGE","EoGE","EoGE","Atlas","Atlas","Atlas","BodyMap","BodyMap","BodyMap","HPA","HPA","HPA","AltIso","AltIso"),prep=c(rep("poly-A",3),rep("rRNA-depl",3),rep("poly-A",8)),layout=c(rep("PE",3),rep("SE",3),rep("PE",6),rep("SE",2)))
#rownames(meta) <- colnames(fpkms.summed)
#tissue <- rep(meta$tissue, each=nrow(fpkms.summed))
#study <- rep(meta$study, each=nrow(fpkms.summed))
#prep <- rep(meta$prep, each=nrow(fpkms.summed))
#layout <- rep(meta$layout, each=nrow(fpkms.summed))
#data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)

#subset <- data[sample(1:nrow(data), 1000),]
#fit <- lm(FPKM ~ prep + layout + study + tissue, data=data)
#a <- anova(fit)
```
Revisit Anova with log-TMMed and combated values.

```{r:anova3}
#m <- melt(fpkm.dual)
#colnames(m) <- c("gene_ID","sample_ID","logTMMFPKM")
#data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)

#subset <- data[sample(1:nrow(data), 1000),]
#fit <- lm(logTMMFPKM ~ + prep + layout + study + tissue, data=data)
#b <- anova(fit)

#batch <- meta$study
#design <- model.matrix(~as.factor(tissue),data=meta)
# Combat fails unless we remove low/unexpressed genes
#combat <- ComBat(dat=fpkm.dual,batch=batch,mod=design,numCovs=NULL,par.prior=TRUE)
#m <- melt(combat)
#colnames(m) <- c("gene_ID","sample_ID","combatlogTMMFPKM")
#data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)

#subset <- data[sample(1:nrow(data), 1000),]
#fit <- lm(combatlogTMMFPKM ~ prep + layout + study + tissue, data=data)
#d <- anova(fit)
```

Plot results
```{r:plot-anova2}
#pdf("anova_repr_FPKM.pdf")
#par(mfrow=c(3,1))
#barplot(a$"F value",names.arg=rownames(a),main="Anova F score, Raw counts")
#barplot(b$"F value",names.arg=rownames(b),main="Anova F score, voom-TMM counts")
#barplot(d$"F value",names.arg=rownames(d),main="Anova F score, ComBat on voom-TMM")
#dev.off()
```

Cufflinks on protein coding genes only:

```{r:cuff-protcod}
#prot <- read.delim("cufflinks_data_protein_coding.txt")
#pnum <- prot[,3:ncol(prot)]
#rownames(pnum) <- prot[,1]
```

Some miscellaneous plots: heatmaps of various transformed counts, RPKMs, TPMs.

Heatmaps with linear corr and Spearman rank corr on raw count, optimal power space transformed counts (OPS), TMM scaled counts, logged counts, and log-TMMed counts.

```{r heatmaps}
#pheatmap(cor(counts),main="Raw counts, Pearson")
#pheatmap(cor(counts,method="spearman"),main="Raw counts, Spearman")
#pheatmap(cor(ops.counts),main= "OPS counts, Pearson")
#pheatmap(cor(tmm),main="CPM-TMM, Pearson")
#pheatmap(cor(tmm, method="spearman"),main="CPM-TMM, Spearman")
#pheatmap(cor(voom),main="Log2 counts, Pearson")
#pheatmap(cor(voom,method="spearman"),main="Log2 counts, Spearman")
#pheatmap(cor(dual),main="log-CPM-TMM")
```

