Walkthrough with all data analyses that comes with the manuscript
Comparing published, reprocessed FPKMs and reprocessed counts
========================================================

Prepare by loading packages etc.
```{r prepare}
library(pheatmap)
library(reshape)
library(gplots)
library(ops)
library(calibrate)
```

Data from four different public sources are downloaded and brain, heart and kidney samples are extracted. 

Start with Human Protein Atlas (HPA):

```{r:HPA download}
#temp <- tempfile()
#download.file(url="http://www.proteinatlas.org/download/rna.csv.zip",destfile=temp)
#hpa <- read.csv(unz(temp, "rna.csv"))
#unlink(temp)

#hpa.heart <- hpa[hpa$Sample=="heart muscle", c("Gene", "Value")]
#hpa.brain <- hpa[hpa$Sample=="cerebral cortex", c("Gene", "Value")]
#hpa.kidney <- hpa[hpa$Sample=="kidney", c("Gene", "Value")]

#hpa.fpkms <- merge(hpa.heart, hpa.brain, by="Gene")
#hpa.fpkms <- merge(hpa.fpkms, hpa.kidney, by="Gene")
#colnames(hpa.fpkms) <- c("ENSG_ID", "HPA_heart", "HPA_brain", "HPA_kidney")
```

Check if the identifiers are unique and write table to file.

```{r:HPA check identifiers}
#length(hpa.fpkms[,1])
#length(unique(hpa.fpkms[,1]))

#write.table(hpa.fpkms,file="hpa_fpkms.txt",quote=F,sep="\t")
```

Next dataset is from the article "Alternative isoform regulation in human tissue transcriptomes.." by Wang et.al
AltIso:

```{r:AltIso download}
#temp <- tempfile()
#download.file(url="http://genes.mit.edu/burgelab/Supplementary/wang_sandberg08/hg18.ensGene.CEs.rpkm.txt",destfile=temp)
#altiso <- read.delim(temp, sep="\t")
#unlink(temp)
```

There is no kidney sample here, so just use heart + brain

```{r}
#altiso.fpkms <- altiso[,c("X.Gene","heart","brain")]
#colnames(altiso.fpkms) <- c("ENSG_ID", "AltIso_heart", "AltIso_brain")
```

Check uniqueness of IDs.

```{r:AltIso check identifiers}
#length(altiso.fpkms[,1])
#length(unique(altiso.fpkms[,1]))

#write.table(altiso.fpkms,file="altiso_fpkms.txt",quote=F,sep="\t")
```

Next dataset is derived from "GTEx": Genotype-Tissue Expression

This is a big download: 337.8 Mb (as of 2014-02-04)
We also add some code to randomly select one sample from each tissue type; there are many biological replicates in this data set.

```{r:GTEx download}
#temp <- tempfile()
#download.file(url="http://www.gtexportal.org/home/rest/file/download?portalFileId=175729&forDownload=true",destfile=temp)
#header_lines <- readLines(temp, n=2)
#gtex <- read.delim(temp, skip=2, sep="\t")
#unlink(temp)

#write.table(gtex, file="gtex_all.txt",   quote=F, sep="\t")

#download.file(url="http://www.gtexportal.org/home/rest/file/download?portalFileId=175707&forDownload=true",destfile="GTEx_description.txt")

#metadata <- read.delim("GTEx_description.txt", sep="\t")
```

The metadata table seems to contain entries that are not in the RPKM table.

```{r}
#samp.id <- gsub('-','.',metadata$SAMPID)
#eligible.samples <- which(samp.id %in% colnames(gtex))
#metadata <- metadata[eligible.samples,]
```

Select random heart, kidney and brain samples.

```{r}
#random.heart <- sample(which(metadata$SMTS=="Heart"), size=1)
#random.heart.samplename <- gsub('-','.',metadata[random.heart, "SAMPID"])
#gtex.heart.fpkm <- as.numeric(gtex[,random.heart.samplename])

#random.brain <- sample(which(metadata$SMTS=="Brain"), size=1)
#random.brain.samplename <- gsub('-','.',metadata[random.brain, "SAMPID"])
#gtex.brain.fpkm <- as.numeric(gtex[,random.brain.samplename])

#random.kidney <- sample(which(metadata$SMTS=="Kidney"), size=1)
#random.kidney.samplename <- gsub('-','.',metadata[random.kidney, "SAMPID"])
#gtex.kidney.fpkm <- as.numeric(gtex[,random.kidney.samplename])
```

Get gene IDs on same format as the other data sets by removing the part after the dot; check ID uniqueness and write to file.

```{r}
#gtex.names <- gtex[,"Name"]
#temp_list <- strsplit(as.character(gtex.names), split="\\.")
#gtex.names.nodot <- unlist(temp_list)[2*(1:length(gtex.names))-1]

#gtex.fpkms <- data.frame(ENSG_ID=gtex.names.nodot, GTEx_heart=gtex.heart.fpkm, GTEx_brain=gtex.brain.fpkm,GTEx_kidney=gtex.kidney.fpkm)

#length(gtex.fpkms[,1])
#length(unique(gtex.fpkms[,1]))

#write.table(gtex.fpkms,file="gtex_fpkms.txt",quote=F,sep="\t")
```

*RNA-seq Atlas*

```{r:Atlas download}
#temp <- tempfile()
#download.file(url="http://medicalgenomics.org/rna_seq_atlas/download?download_revision1=1",destfile=temp)
#atlas <- read.delim(temp, sep="\t")
#unlink(temp)

#atlas.fpkms <- atlas[,c("ensembl_gene_id","heart","hypothalamus","kidney")]
#colnames(atlas.fpkms) <- c("ENSG_ID","Atlas_heart","Atlas_brain","Atlas_kidney")
#write.table(atlas.fpkms,file="atlas_fpkms.txt",quote=F,sep="\t")
```

Combining F/RPKM values from public data sets
---------------------------------------------

We will join the data sets on ENSEMBL ID:s, losing a lot of data in the process - but joining on gene symbols or something else would lead to an even worse loss. 

```{r LoadPackagesAndData}
library(org.Hs.eg.db) # for transferring gene identifiers
library(data.table) # for collapsing transcript RPKMs
library(pheatmap) # for nicer visualization
library(edgeR) # for TMM normalization

#hpa.fpkms <- read.delim("hpa_fpkms.txt")
#altiso.fpkms <- read.delim("altiso_fpkms.txt")
#gtex.fpkms <- read.delim("gtex_fpkms.txt")
#atlas.fpkms <- read.delim("atlas_fpkms.txt")
```

The RNA-seq Atlas data set uses many different identifiers, while the other all use ENSG as the primary identifier

Approach 1: Merge on ENSEMBL genes (ENSG) as given in RNA-seq Atlas. Note that there are repeated ENSG ID:s in RNA-seq Atlas, as opposed to the other data sets, so we need to do something about that. In this case, we just sum the transcripts that belong to each ENSG gene. We use data.table for this.

```{r SumTranscripts}
#data.dt <- data.table(atlas.fpkms)
#setkey(data.dt, ENSG_ID)
#temp <- data.dt[, lapply(.SD, sum), by=ENSG_ID]
#collapsed <- as.data.frame(temp)
#atlas.fpkms.summed <- collapsed[,2:ncol(collapsed)] 
#rownames(atlas.fpkms.summed) <- collapsed[,1]

#atlas.fpkms.summed <- atlas.fpkms.summed[2:nrow(atlas.fpkms.summed),]
```

Finally, combine all the data sets into a data frame.

```{r Combine}
#fpkms <- merge(hpa.fpkms, altiso.fpkms, by="ENSG_ID")
#fpkms <- merge(fpkms, gtex.fpkms, by="ENSG_ID")
#fpkms <- merge(fpkms, atlas.fpkms.summed, by.x="ENSG_ID", by.y=0)
#gene_id <- fpkms[,1]
#f <- fpkms[,2:ncol(fpkms)]
#rownames(f) <- gene_id
```

Check how many ENSG IDs we have left.

```{r}
#dim(f)
```

Approach 2: Try to map Entrez symbols to ENSEMBL to recover more ENSG IDs than already present in the table. 

```{r RemapENSEMBL}
#m <- org.Hs.egENSEMBL
#mapped_genes <- mappedkeys(m)
#ensg.for.entrez <- as.list(m[mapped_genes])
#remapped.ensg <- ensg.for.entrez[as.character(atlas$entrez_gene_id)]

#atlas.fpkms$remapped_ensg <- as.character(remapped.ensg)

# And add expression values
#data.dt <- data.table(atlas.fpkms[,2:ncol(atlas.fpkms)])
#setkey(data.dt, remapped_ensg)
#temp <- data.dt[, lapply(.SD, sum), by=remapped_ensg]
#collapsed <- as.data.frame(temp)
#atlas.fpkms.summed <- collapsed[,2:ncol(collapsed)] 
#rownames(atlas.fpkms.summed) <- collapsed[,1]
```

Combine data sets again

```{r Combine2}
#fpkms <- merge(hpa.fpkms, altiso.fpkms, by="ENSG_ID")
#fpkms <- merge(fpkms, gtex.fpkms, by="ENSG_ID")
#fpkms <- merge(fpkms, atlas.fpkms.summed, by.x="ENSG_ID", by.y=0)
#gene_id <- fpkms[,1]
#f <- fpkms[,2:ncol(fpkms)]
#rownames(f) <- gene_id
#write.table(f, file = 'published_rpkms.txt', quote=F)

#instead of downloading everytime:

published <- read.delim("published_rpkms.txt",sep=" ")
sampleinfo_published <- read.table("sample_info_published.txt",header=TRUE)
dim(published)
```

The published FPKM values are first filtered by removing all lines where FPKM is less or equal to 0.01 in all samples

```{r:filter}

published.nozero <- published[-which(rowSums(published[,])<=0.01),]

```

Heatmap of Spearman correlations between published expression profiles (# genes = 13,323)

```{r:heatmap spearma raw values}

pheatmap(cor(published.nozero, method="spearman")) 

```

The brain samples are in a separate cluster, whereas the heart and kidney ones are intermixed.

Alternatively, one could use Pearson correlation:

```{r:heatmap pearson raw values}

pheatmap(cor(published.nozero))

```

PCA analysis of published FPKM values

```{r:PCA raw FPKMs}

colors <- c(1, 2, 3, 1, 2, 1, 2, 3, 1, 2, 3)

p <- prcomp(t(published.nozero))

plot(p$x[,1],p$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"),main="Published FPKM values \n n=13323")
textxy(p$x[,1],p$x[,2],labs=rownames(p$x))

plot(p$x[,2],p$x[,3],pch=20,col=colors,xlab=paste("PC2"),ylab=paste("PC3"),main="Published FPKM values \n n=13323")
textxy(p$x[,2],p$x[,3],labs=rownames(p$x))
```

We can plot all pairwise combinations of principal components 1 to 5. (not shown in paper):

```{r}

par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
  	if (i<j){ 
		plot(p$x[,i],p$x[,j],pch=20,col=colors,xlab=paste("PC",i),ylab=paste("PC",j),main="Published FPKM values \n n=13323")
		}
	}
}

```

Look a bit closer at PCs 1-3 in prcomp: (not shown in paper)

```{r:prcomp1-3rawpublished}

    symbol <- read.table("ENSG_Symbol.txt",header=TRUE)
    
      plot(p$x[,1],p$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"))
      textxy(p$x[,1],p$x[,2],labs=rownames(p$x))
      load.pc1 <- p$rotation[,1][order(p$rotation[,1])]
      extreme.pc1 <- c(tail(load.pc1), head(load.pc1))
      symbs <- vector()
      for (n in names(extreme.pc1)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc1, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC1 (raw RPKM)")

      load.pc2 <- p$rotation[,2][order(p$rotation[,2])]
      extreme.pc2 <- c(tail(load.pc2), head(load.pc2))
      symbs <- vector()
      for (n in names(extreme.pc2)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc2, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC2 (raw RPKM)")

      load.pc3 <- p$rotation[,3][order(p$rotation[,3])]
      extreme.pc3 <- c(tail(load.pc3), head(load.pc3))
      symbs <- vector()
      for (n in names(extreme.pc3)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc3, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC3 (raw RPKM)")

```

Despite what it might appear like from the plot, PC 1 seems to be about tissue specificity. MYL2, TNNT2, MB, DES, ACTC1, MYH7 are all heart genes. SPP1 and ALDOB are kidney genes and GFAP, CLU, PLP1 are brain genes.

PC 2: Very high positive loadings for FTH1 (highly variable between tissues) and MYL2 (heart). But also high *negative* loading for TNNT2 (heart). 

PC3: The same genes seem to be involved again (MYL2, TNNT2, FTH1 etc.)


Try Anova on a "melted" expression matrix with some metadata:

```{r:anova published FPKMs}

m <- melt(published.nozero)
colnames(m) <- c("sample_ID","RPKM")

meta <- sampleinfo_published[,c("Study","Tissue","Preparation","NumberRaw","Numbermapped","Readtype")]
rownames(meta) <- colnames(published.nozero)
tissue <- rep(meta$Tissue, each=nrow(published.nozero))
study <- rep(meta$Study, each=nrow(published.nozero))
prep <- rep(meta$Preparation, each=nrow(published.nozero))
layout <- rep(meta$Readtype, each=nrow(published.nozero))
raw <- rep(meta$NumberRaw, each=nrow(published.nozero))
mapped <- rep(meta$Numbermapped, each=nrow(published.nozero))
data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout,nraw=raw,nmapped=mapped)
subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(RPKM ~ study + nraw + layout + prep + tissue, data=data)
a <- anova(fit)
maxval = 3000

barplot(a$"F value"[-7],names.arg=rownames(a)[-7],main="Anova F score, Raw RPKM",ylim=c(0,maxval))
print(a)

```

Try log2 tranform the published FPKM values:

```{r:log tranform}

pseudo <- 1
published.log <- log2(published.nozero + pseudo)

```

Heatmap with log2 values:

```{r:heatmap log2 FPKM values}

pheatmap(cor(published.log),method="spearman")

```

PCA analysis of log2 published FPKM values

```{r:PCA log FPKMs}

colors <- c(1, 2, 3, 1, 2, 1, 2, 3, 1, 2, 3)

p.log <- prcomp(t(published.log))

plot(p.log$x[,1],p.log$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"),main="log2 Published FPKM values \n n=13323")
textxy(p.log$x[,1],p.log$x[,2],labs=rownames(p.log$x))

plot(p.log$x[,2],p.log$x[,3],pch=20,col=colors,xlab=paste("PC2"),ylab=paste("PC3"),main="log2 Published FPKM values \n n=13323")
textxy(p.log$x[,2],p.log$x[,3],labs=rownames(p.log$x))

```

We can plot all pairwise combinations of principal components 1 to 5. (not shown in paper):

```{r:PCA log FPKMs pairwise}

par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
    if (i<j){ 
		plot(p.log$x[,i],p.log$x[,j],pch=20,col=colors,xlab=paste("PC",i),ylab=paste("PC",j),main="log2 Published FPKM values \n n=13323")
		}
	}
}

```

Look a bit closer at PCs 1-3 in prcomp:

```{r:prcomp1-3log2}
    symbol <- read.table("ENSG_Symbol.txt",header=TRUE)
    
    plot(p.log$x[,1],p.log$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"))
    textxy(p.log$x[,1],p.log$x[,2],labs=rownames(p$x))
  
      
      load.pc1 <- p.log$rotation[,1][order(p.log$rotation[,1])]
      extreme.pc1 <- c(tail(load.pc1), head(load.pc1))
      symbs <- vector()
      for (n in names(extreme.pc1)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc1, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC1 (log2RPKM)")

      load.pc2 <- p.log$rotation[,2][order(p.log$rotation[,2])]
      extreme.pc2 <- c(tail(load.pc2), head(load.pc2))
      symbs <- vector()
      for (n in names(extreme.pc2)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc2, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC2 (log2RPKM)")

      load.pc3 <- p.log$rotation[,3][order(p$rotation[,3])]
      extreme.pc3 <- c(tail(load.pc3), head(load.pc3))
      symbs <- vector()
      for (n in names(extreme.pc3)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc3, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC3 (log2RPKM)")


```

ENSG00000171560 - FGA - fibrinogen light chain. A bit inconsistent between studies. Mostly kidney specific but not in the Atlas.
ENSG00000148677 - ANKRD1 - definitely heart gene.
ENSG00000135218 - CD36 - heart gene.
ENSG00000057593 - F7 - varies mostly across studies.
ENSG00000118194 - TNNT2 - heart.
ENSG00000188257 - PLA2G2A - heart.
ENSG00000105372 - RPS19 - ribosomal - varies across studies. 
ENSG00000063177 - RPL18 - ribosomal - varies across studies.
ENSG00000100097 - LGALS1 - varies across studies / a bit heart specific
ENSG00000105640 - RPL18A - varies across studies
ENSG00000105583 - varies across studies
ENSG00000087086 - FTL - kidney specific / large variation across studies


PC2:
ENSG00000168314 - MOBP - brain
ENSG00000104833 - TUBB4A - brain
ENSG00000104435 - STMN2 - brain
ENSG00000132639 - SNAP25 - brain
ENSG00000123560 - PLP1 - brain
ENSG00000131095 - GFAP - brain
ENSG00000111245 - MYL1 - heart
ENSG00000114854 - TNNC1 - heart
ENSG00000160808 - MYL3 - heart
ENSG00000198125 - MB - heart
ENSG00000104879 - CKM - heart
ENSG00000159251 - ACTC1 - heart

PC3:
ENSG00000111245 - MYL2 - heart
ENSG00000101608 - MYL12A - heart but varying
ENSG00000175206 - NPPA - heart
ENSG00000175084 - DES - heart
ENSG00000159251 - ACTC1 - heart
ENSG00000129991 - TNNI3 - heart
ENSG00000167996 - FTH1 - varying between tissues, high in kidney and brain
ENSG00000118194 - TNNT2 - heart
ENSG00000125971 - DYNLRB1 - varying
ENSG00000075624 - ACTB - brain
ENSG00000071082 - RPL31 - varying
ENSG00000106211 - HSPB1 - heart & kidney


Try Anova on a "melted" expression matrix with logged values and some metadata:

```{r:anova published log FPKMs}
n <- melt(published.log)
colnames(n) <- c("sample_ID","RPKM")
meta <- sampleinfo_published[,c("Study","Tissue","Preparation","NumberRaw","Numbermapped","Readtype")]
rownames(meta) <- colnames(published.log)
tissue <- rep(meta$Tissue, each=nrow(f.nozero))
study <- rep(meta$Study, each=nrow(f.nozero))
prep <- rep(meta$Preparation, each=nrow(f.nozero))
layout <- rep(meta$Readtype, each=nrow(f.nozero))
raw <- rep(meta$NumberRaw, each=nrow(f.nozero))
mapped <- rep(meta$Numbermapped, each=nrow(f.nozero))
data <- data.frame(n, tissue=tissue, study=study, prep=prep, layout=layout,nraw=raw,nmapped=mapped)
subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(2RPKM ~ study + nraw + layout + prep + tissue, data=data)
b <- anova(fit)
maxval = 3000

barplot(b$"F value"[-7],names.arg=rownames(b)[-7],main="Anova F score, Raw RPKM",ylim=c(0,maxval))
print(b)

```

Combat analysis is performed on logged values (n=13,323)

```{r:combat}
library(sva)

meta <- data.frame(study=c(rep("HPA",3),rep("AltIso",2),rep("GTex",3),rep("Atlas",3)),tissue=c("Heart","Brain","Kidney","Heart","Brain","Heart","Brain","Kidney","Heart","Brain","Kidney"))

batch <- meta$study
design <- model.matrix(~as.factor(tissue),data=meta)

combat <- ComBat(dat=published.log,batch=batch,mod=design,numCovs=NULL,par.prior=TRUE)

write.table(combat, file="published_rpkms_combat_log2.txt", quote=F)

```

Heatmap of Spearman correlations between published expression profiles after combat run (# genes = 13,323)

```{r:heatmap spearma combat values}

pheatmap(cor(combat, method="spearman")) 

```

Alternatively, one could use Pearson correlation:

```{r:heatmap pearson combat values}

pheatmap(cor(combat))

```

PCA analysis of published FPKM values after combat run

```{r:PCA combat values}

colors <- c(1, 2, 3, 1, 2, 1, 2, 3, 1, 2, 3)

p.combat <- prcomp(t(combat))

plot(p.combat$x[,1],p.combat$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"),main="Published FPKM values \n COMBAT \n n=13323")

plot(p.combat$x[,2],p.combat$x[,3],pch=20,col=colors,xlab=paste("PC2"),ylab=paste("PC3"),main="Published FPKM values \n COMBAT \n n=13323")

```

We can plot all pairwise combinations of principal components 1 to 5. (not shown in paper):

```{r:PCA combat values pairwise}

par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
    if (i<j){ 
		plot(p.combat$x[,i],p.combat$x[,j],pch=20,col=colors,xlab=paste("PC",i),ylab=paste("PC",j),main="Published FPKM values \n COMBAT \ n=13323")
		}
	}
}

```

Look a bit closer at PCs 1-3 in prcomp:

```{r:prcomp1-3combat}
    symbol <- read.table("ENSG_Symbol.txt",header=TRUE)
    
    plot(p.combat$x[,1],p.combat$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"))
    textxy(p.combat$x[,1],p.combat$x[,2],labs=rownames(p$x))
  
      
      load.pc1 <- p.combat$rotation[,1][order(p.combat$rotation[,1])]
      extreme.pc1 <- c(tail(load.pc1), head(load.pc1))
      symbs <- vector()
      for (n in names(extreme.pc1)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc1, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC1 (COMBAT)")

      load.pc2 <- p.combat$rotation[,2][order(p.combat$rotation[,2])]
      extreme.pc2 <- c(tail(load.pc2), head(load.pc2))
      symbs <- vector()
      for (n in names(extreme.pc2)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc2, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC2 (COMBAT)")

      load.pc3 <- p.combat$rotation[,3][order(p.combat$rotation[,3])]
      extreme.pc3 <- c(tail(load.pc3), head(load.pc3))
      symbs <- vector()
      for (n in names(extreme.pc3)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc3, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC3 (COMBAT)")
```

Revisit Anova with combated values.

```{r:anova-combat}

m <- melt(combat)
colnames(m) <- c("sample_ID","combat")
data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout,nraw=raw,nmapped=mapped)
fit <- lm(combat ~ study + nraw + layout + prep + tissue, data=data)
c <- anova(fit)

maxval <- 500
barplot(c$"F value"[-7],names.arg=rownames(c)[-7],main="Anova F score, Combat",ylim=c(0,maxval))

print(c)

```

Comparing FPKMs for FASTQ files reprocessed with TopHat and Cufflinks

```{r:download reprocessed cufflinks}

cufflinks <- read.delim("fpkm_table_tophat.txt")

sampleinfo_cufflinks <- read.delim("sample_info_reprocessed.txt")

```

First, we will restrict the data set to only include protein coding genes using the ensembl based R package biomaRt

```{r:filter protein coding}

library(biomaRt)

gene_ids <- as.vector(cufflinks[,1])

ensembl = useMart("ensembl", dataset = "hsapiens_gene_ensembl") #select the ensembl database

gene_type <- getBM(attributes=c("ensembl_gene_id", "gene_biotype"), 
                   filters = "ensembl_gene_id",
                   values=gene_ids,
                   mart=ensembl)

pc <- subset(gene_type[,1],gene_type[,2]=="protein_coding")

cufflinks_pc <- cufflinks[match(pc,cufflinks[,1]),]

```

And let's remove all lines where FPKM is close to zero in all samples before we proceed with this version of the data set:

```{r:remove low expressed}

cufflinks_pc_nozero <- cufflinks_pc[-which(rowSums(cufflinks_pc[,3:16])<=0.01),]

#write.table(cufflinks_pc_nozero, file="cufflinks_pc_nozero.txt", quote=F)
#cufflinks_pc_nozero <- read.table(file="cufflinks_fpkm_proteincoding_nozero.txt")

```

Heatmap of Spearman correlations between reprocessed expression profiles (# genes = 19,475)

```{r:heatmap spearman reprocessed cufflinks}

pheatmap(cor(cufflinks_pc_nozero[,3:16], method="spearman")) 

```

The brain samples are in a separate cluster, whereas the heart and kidney ones are intermixed.

Alternatively, one could use Pearson correlation:

```{r:heatmap pearson reprocessed cufflinks}

pheatmap(cor(cufflinks_pc_nozero[,3:16])) 

```

Let's look at a few SVD plots. 

First we take a look at the "raw" FPKM values for PC 1&2:

```{r:PCA reprocessed cufflinks 1&2}

colors <- c(2,1,3,2,1,3,2,1,3,2,1,3,2,1)

p.cufflinks <- prcomp(t(cufflinks_pc_nozero[,3:16]))

plot(p.cufflinks$x[,1],p.cufflinks$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"),main="Reprocessed FPKM values \n n=19475")
textxy(p.cufflinks$x[,1],p.cufflinks$x[,2],labs=rownames(p.cufflinks$x))


```

and PC 2&3:

```{r PCA reprocessed cufflinks 2&3}

plot(p.cufflinks$x[,2],p.cufflinks$x[,3],pch=20,col=colors,xlab=paste("PC2"),ylab=paste("PC3"),main="Reprocessed FPKM values \n n=19475")
textxy(p.cufflinks$x[,2],p.cufflinks$x[,3],labs=rownames(p.cufflinks$x))

```

We can plot all pairwise combinations of principal components 1 to 5. (not shown in paper)
Start with SVD on the "raw" F/RPKMs.

```{r:PCA reprocessed cufflinks pairwise}

colors <- c(2,1,3,2,1,3,2,1,3,2,1,3,2,1)

par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
    if (i<j){ 
      plot(p.cufflinks$x[,i],p.cufflinks$x[,j],pch=20,col=colors,xlab=paste("PC",i),ylab=paste("PC",j),main="Reprocessed FPKM values \n n=19475")
		}
	}
}

```

Look at PCA loadings for PC1 (raw FPKM)

```{r:prcomp1-3raw}
    par(mfrow=c(1,1))
    symbol <- read.table("ENSG_Symbol.txt",header=TRUE)
    fpkms <- cufflinks_pc_nozero[,3:16]
    rownames(fpkms) <- cufflinks_pc_nozero[,1]
    p <- prcomp(t(fpkms))
      plot(p$x[,1],p$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"))
      textxy(p$x[,1],p$x[,2],labs=rownames(p$x))
      
      load.pc1 <- p$rotation[,1][order(p$rotation[,1])]
      extreme.pc1 <- c(tail(load.pc1), head(load.pc1))
      symbs <- vector()
      for (n in names(extreme.pc1)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc1, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC1 (raw Cufflinks FPKM)")

     load.pc2 <- p$rotation[,2][order(p$rotation[,2])]
      extreme.pc2 <- c(tail(load.pc2), head(load.pc2))
      symbs <- vector()
      for (n in names(extreme.pc2)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc2, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC2 (raw Cufflinks FPKM)")

     load.pc3 <- p$rotation[,3][order(p$rotation[,3])]
      extreme.pc3 <- c(tail(load.pc3), head(load.pc3))
      symbs <- vector()
      for (n in names(extreme.pc3)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc3, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC3 (raw Cufflinks FPKM)")
```

All highly loaded ones are mitochondrial. This is likely not tissue specificity but more about variation between studies.
PC2 seems to be "mitochondrial vs brain" (?)

Anova analysis of different batch factors:

```{r:anova reprocessed cufflinks}

o <- melt(cufflinks_pc_nozero)
colnames(o) <- c("ENSG","Gene","sample_ID","Cuff_FPKM")
meta <- data.frame(tissue=c("brain","heart","kidney","brain","heart","kidney","brain","heart","kidney","brain","heart","kidney","brain","heart"),study=c("EoGE","EoGE","EoGE","Atlas","Atlas","Atlas","BodyMap","BodyMap","BodyMap","HPA","HPA","HPA","AltIso","AltIso"),prep=c(rep("poly-A",3),rep("rRNA-depl",3),rep("poly-A",8)),layout=c(rep("PE",3),rep("SE",3),rep("PE",6),rep("SE",2)))
rownames(meta) <- colnames(f_pc_nozero)[3:16]
tissue <- rep(meta$tissue, each=nrow(cufflinks_pc_nozero))
study <- rep(meta$study, each=nrow(cufflinks_pc_nozero))
prep <- rep(meta$prep, each=nrow(cufflinks_pc_nozero))
layout <- rep(meta$layout, each=nrow(cufflinks_pc_nozero))
data <- data.frame(o, tissue=tissue, study=study, prep=prep, layout=layout)

#subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(Cuff_FPKM ~ prep + layout + study + tissue, data=data)
d <- anova(fit)
maxval = 100
barplot(d$"F value"[-5],names.arg=rownames(d)[-5],main="Anova F score, Cufflinks FPKM",ylim=c(0,maxval))

```

Try log2 tranform the reprocessed FPKM values:

```{r:log tranform reprocessed cufflinks FPKM values}

pseudo <- 1
logs <- log2(cufflinks_pc_nozero[,3:16] + pseudo)
cufflinks.log <- cbind(f_pc_nozero[,1:2],logs) 


```

Heatmap with log2 values:

```{r:heatmap log2 reprocessed cufflinks FPKM values}

pheatmap(cor(cufflinks.log))

```

PCA analysis of log2 reprocessed cufflinks FPKM values

```{r:PCA log2 reprocessed cufflinks FPKM values}

colors <- c(2,1,3,2,1,3,2,1,3,2,1,3,2,1)

p.log.cufflinks <- prcomp(t(cufflinks.log))

plot(p.log.cufflinks$x[,1],p.log.cufflinks$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"),main="log2 reprocessed cufflinks FPKM values \n n=19475")

plot(p.log.cufflinks$x[,2],p.log.cufflinks$x[,3],pch=20,col=colors,xlab=paste("PC2"),ylab=paste("PC3"),main="log2 reprocessed cufflinks FPKM values \n n=19475")

```

We can plot all pairwise combinations of principal components 1 to 5. (not shown in paper):

```{r:PCA log cufflinks pairwise}

par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
    if (i<j){ 
  	plot(p.log.cufflinks$x[,i],p.log.cufflinks$x[,j],pch=20,col=colors,xlab=paste("PC",i),ylab=paste("PC",j),main="log2 reprocessed FPKM values \n n=19475")
		}
	}
}

```

Look a bit closer at PCs 1-3 in prcomp for the logged FPKM values from cufflinks:

```{r:prcomp1-3log reprocessed}
     par(mfrow=c(1,1))
    symbol <- read.table("ENSG_Symbol.txt",header=TRUE)
    f.log <- fpkms.log[,3:16]
    rownames(f.log) <- fpkms.log[,1]
    p <- prcomp(t(f.log))
      plot(p$x[,1],p$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"))
      textxy(p$x[,1],p$x[,2],labs=rownames(p$x))

      load.pc1 <- p$rotation[,1][order(p$rotation[,1])]
      extreme.pc1 <- c(tail(load.pc1), head(load.pc1))
      symbs <- vector()
      for (n in names(extreme.pc1)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc1, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC1 (log Cufflinks FPKM)")

     load.pc2 <- p$rotation[,2][order(p$rotation[,2])]
      extreme.pc2 <- c(tail(load.pc2), head(load.pc2))
      symbs <- vector()
      for (n in names(extreme.pc2)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc2, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC2 (log Cufflinks FPKM)")

     load.pc3 <- p$rotation[,3][order(p$rotation[,3])]
      extreme.pc3 <- c(tail(load.pc3), head(load.pc3))
      symbs <- vector()
      for (n in names(extreme.pc3)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc3, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC3 (log Cufflinks FPKM)")
```

Seems to yield a heart vs. brain separation


```

ENSG00000171560 - FGA - fibrinogen light chain. A bit inconsistent between studies. Mostly kidney specific but not in the Atlas.
ENSG00000148677 - ANKRD1 - definitely heart gene.
ENSG00000135218 - CD36 - heart gene.
ENSG00000057593 - F7 - varies mostly across studies.
ENSG00000118194 - TNNT2 - heart.
ENSG00000188257 - PLA2G2A - heart.
ENSG00000105372 - RPS19 - ribosomal - varies across studies. 
ENSG00000063177 - RPL18 - ribosomal - varies across studies.
ENSG00000100097 - LGALS1 - varies across studies / a bit heart specific
ENSG00000105640 - RPL18A - varies across studies
ENSG00000105583 - varies across studies
ENSG00000087086 - FTL - kidney specific / large variation across studies



Try Anova on a "melted" expression matrix with logged values and some metadata:

```{r:anova published log FPKMs}
n <- melt(published.log)
colnames(n) <- c("sample_ID","RPKM")
meta <- sampleinfo_published[,c("Study","Tissue","Preparation","NumberRaw","Numbermapped","Readtype")]
rownames(meta) <- colnames(published.log)
tissue <- rep(meta$Tissue, each=nrow(f.nozero))
study <- rep(meta$Study, each=nrow(f.nozero))
prep <- rep(meta$Preparation, each=nrow(f.nozero))
layout <- rep(meta$Readtype, each=nrow(f.nozero))
raw <- rep(meta$NumberRaw, each=nrow(f.nozero))
mapped <- rep(meta$Numbermapped, each=nrow(f.nozero))
data <- data.frame(n, tissue=tissue, study=study, prep=prep, layout=layout,nraw=raw,nmapped=mapped)
subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(2RPKM ~ study + nraw + layout + prep + tissue, data=data)
b <- anova(fit)
maxval = 3000

barplot(b$"F value"[-7],names.arg=rownames(b)[-7],main="Anova F score, Raw RPKM",ylim=c(0,maxval))
print(b)

```


```{r:reprocessed }
plotPC(fpkms.log[,3:16], 1, 2, desc="Reprocessed F/RPKM values, log2 \n SVD \n n=19475", colors=colors)

plotPC(fpkms.log[,3:16], 2, 3, desc="Reprocessed FPKM values, log2 \n SVD \n n=19475", colors=colors)

```

Or all combinations for log2-FPKM and regular PCA (prcomp())
```{r}
colors <- c(2,1,3,2,1,3,2,1,3,2,1,3,2,1)

p <- prcomp(t(fpkms.log[,3:16]),scale.=T)

par(mfrow=c(4,4))
for (i in 1:6){
  for(j in 1:6){
    if (i<j){ 
    	plot(p$x[,i],p$x[,j],pch=20,col=colors,xlab=paste("PC", i),ylab=paste("PC", j))
		}
	}
}
```
Loadings for logged values?

```{r:prcomp1-3log}
    par(mfrow=c(1,1))
    symbol <- read.table("ENSG_Symbol.txt",header=TRUE)
    f.log <- fpkms.log[,3:16]
    rownames(f.log) <- fpkms.log[,1]
    p <- prcomp(t(f.log))
      plot(p$x[,1],p$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"))
      textxy(p$x[,1],p$x[,2],labs=rownames(p$x))

      load.pc1 <- p$rotation[,1][order(p$rotation[,1])]
      extreme.pc1 <- c(tail(load.pc1), head(load.pc1))
      symbs <- vector()
      for (n in names(extreme.pc1)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc1, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC1 (log Cufflinks FPKM)")

     load.pc2 <- p$rotation[,2][order(p$rotation[,2])]
      extreme.pc2 <- c(tail(load.pc2), head(load.pc2))
      symbs <- vector()
      for (n in names(extreme.pc2)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc2, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC2 (log Cufflinks FPKM)")

     load.pc3 <- p$rotation[,3][order(p$rotation[,3])]
      extreme.pc3 <- c(tail(load.pc3), head(load.pc3))
      symbs <- vector()
      for (n in names(extreme.pc3)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc3, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC3 (log Cufflinks FPKM)")
```

Seems to yield a heart vs. brain separation

Combat analysis for removal of batch effects (n=19475):

```{r:combat}
library(sva)

meta <- data.frame(study=c(rep("EoGE",3),rep("Atlas",3),rep("BodyMap",3),rep("HPA",3),rep("AltIso",2)),tissue=c("Brain","Heart","Kidney","Brain","Heart","Kidney","Brain","Heart","Kidney","Brain","Heart","Kidney","Brain","Heart"),prep=c(rep("poly-A",3),rep("rRNA-depl",3),rep("poly-A",8)),layout=c(rep("PE",3),rep("SE",3),rep("PE",6),rep("SE",2)))

batch <- meta$study
design <- model.matrix(~as.factor(tissue),data=meta)

combat <- ComBat(dat=fpkms.log[,3:16],batch=batch,mod=design,numCovs=NULL,par.prior=TRUE)
rownames(combat) <- fpkms.log[,1]
write.table(combat, file="reprocessed_rpkms_combat_log2.txt", quote=F)

```

Let's see how the correlation heatmap and PCA plots look after correction for batch effects with combat:

```{r}

pheatmap(cor(combat))
plotPC(combat,1,2,colors=colors,desc="Reprocessed F/RPKM values, ComBat on log2 values \n SVD \n n=19475")
plotPC(combat,2,3,colors=colors,desc="Reprocessed F/RPKM values, ComBat on log2 values \n SVD \n n=19475")

```

Examine loadings of PCs 1-3:

```{r:loadingsComBat}
    par(mfrow=c(1,1))
    symbol <- read.table("ENSG_Symbol.txt",header=TRUE)
    p <- prcomp(t(combat))
      plot(p$x[,1],p$x[,2],pch=20,col=colors,xlab=paste("PC1"),ylab=paste("PC2"))
      textxy(p$x[,1],p$x[,2],labs=rownames(p$x))

      load.pc1 <- p$rotation[,1][order(p$rotation[,1])]
      extreme.pc1 <- c(tail(load.pc1), head(load.pc1))
      symbs <- vector()
      for (n in names(extreme.pc1)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc1, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC1 (ComBat Cufflinks FPKM)")

     load.pc2 <- p$rotation[,2][order(p$rotation[,2])]
      extreme.pc2 <- c(tail(load.pc2), head(load.pc2))
      symbs <- vector()
      for (n in names(extreme.pc2)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc2, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC2 (ComBat Cufflinks FPKM)")

     load.pc3 <- p$rotation[,3][order(p$rotation[,3])]
      extreme.pc3 <- c(tail(load.pc3), head(load.pc3))
      symbs <- vector()
      for (n in names(extreme.pc3)){
        temp <- as.character(symbol[symbol[,1] == n,2])
        if (length(temp)>1) temp <- temp[1]
        symbs <- c(symbs, temp)
        }
      
      barplot(extreme.pc3, names.arg=symbs, las=2, main="Genes w highest absolute loadings in PC3 (log Cufflinks FPKM)")
```

**Figure 4C**

```{r}
barplot(a$"F value"[-5],names.arg=rownames(a)[-5],main="Anova F score, Cufflinks FPKM",ylim=c(0,maxval))
```

**Figure 4D (?)**

ANOVA analyses on logged values:

```{r:anova-log}
m <- melt(fpkms.log[,3:ncol(fpkms.log)])
colnames(m) <- c("sample_ID","log2FPKM")

data <- data.frame(m, tissue=tissue, study=study, prep=prep, layout=layout)
#subset <- data[sample(1:nrow(data), 1000),]
fit <- lm(log2FPKM ~ prep + layout + study + tissue, data=data)
b <- anova(fit)

barplot(b$"F value"[-5],names.arg=rownames(b)[-5],main="Anova F score, log2-RPKM",ylim=c(0,3000))
print(b)
```

